// @sqs/i18n-ui/lib/react/index.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var tslib_1 = require("tslib");
var react_1 = tslib_1.__importDefault(require("react"));
var FormattedDateTime_1 = tslib_1.__importDefault(require("./components/FormattedDateTime"));
var RelativeTime_1 = tslib_1.__importDefault(require("./components/RelativeTime"));
var FormattedCurrency_1 = tslib_1.__importDefault(require("./components/FormattedCurrency"));
var FormattedNumber_1 = tslib_1.__importDefault(require("./components/FormattedNumber"));
var FormattedQuantity_1 = tslib_1.__importDefault(require("./components/FormattedQuantity"));
var FormattedMoney_1 = tslib_1.__importDefault(require("./components/FormattedMoney"));
var T_1 = tslib_1.__importDefault(require("./components/T"));
var Plural_1 = tslib_1.__importDefault(require("./components/Plural"));
var I18nReact = (function () {
    function I18nReact(i18nInstance) {
        var _this = this;
        this.FormattedCurrency = function (props) { return react_1.default.createElement(FormattedCurrency_1.default, tslib_1.__assign({}, props, { formatter: _this.i18nInstance.formatCurrency })); };
        this.FormattedMoney = function (props) { return (react_1.default.createElement(FormattedMoney_1.default, tslib_1.__assign({}, props, { formatter: _this.i18nInstance.formatMoney }))); };
        this.FormattedNumber = function (props) { return react_1.default.createElement(FormattedNumber_1.default, tslib_1.__assign({}, props, { formatter: _this.i18nInstance.formatNumber })); };
        this.FormattedQuantity = function (props) { return react_1.default.createElement(FormattedQuantity_1.default, tslib_1.__assign({}, props, { formatter: _this.i18nInstance.formatQuantity })); };
        this.FormattedDateTime = function (props) { return react_1.default.createElement(FormattedDateTime_1.default, tslib_1.__assign({}, props, { formatter: _this.i18nInstance.formatDateTime })); };
        this.RelativeTime = function (props) { return (react_1.default.createElement(RelativeTime_1.default, tslib_1.__assign({}, props, { formatter: _this.i18nInstance.formatRelativeTime }))); };
        this.T = function (_a) {
            var children = _a.children, project = _a.project, rest = tslib_1.__rest(_a, ["children", "project"]);
            return (react_1.default.createElement(T_1.default, tslib_1.__assign({}, rest, { project: project, formatter: _this.i18nInstance.translate, htmlAttributesTransform: _this.i18nInstance.htmlAttributesTransform }), children));
        };
        this.Plural = function (props) { return (react_1.default.createElement(Plural_1.default, tslib_1.__assign({}, props, { project: props.project, forms: props.forms, n: props.n, formatter: _this.i18nInstance.pluralize, htmlAttributesTransform: _this.i18nInstance.htmlAttributesTransform }))); };
        this.i18nInstance = i18nInstance;
        this.T.displayName = 'TWrapper';
        this.Plural.displayName = 'PluralWrapper';
        this.FormattedCurrency.displayName = 'FormattedCurrencyWrapper';
        this.FormattedMoney.displayName = 'FormattedMoneyWrapper';
        this.FormattedNumber.displayName = 'FormattedNumberWrapper';
        this.FormattedQuantity.displayName = 'FormattedQuantityWrapper';
        this.FormattedDateTime.displayName = 'FormattedDateTimeWrapper';
        this.RelativeTime.displayName = 'RelativeTimeWrapper';
    }
    return I18nReact;
}());
exports.default = I18nReact;
