// @sqs/i18n-ui/lib/icu-message-formatter/icuMessageFormatter.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;
require("core-js/modules/es.object.keys.js");
require("core-js/modules/es.object.to-string.js");
var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));
var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));
var _messageformat = _interopRequireDefault(require("./messageformat"));
var _WrapperFormatter = _interopRequireDefault(require("./WrapperFormatter"));
var _utils = require("./utils");
var IcuMessageFormatter = function () {
  function IcuMessageFormatter(pluralFuncs) {
    var formatters = arguments.length > 1 && arguments[1] !== undefined ? arguments[1] : {};
    (0, _classCallCheck2.default)(this, IcuMessageFormatter);
    this.dictionary = {};
    this.icuInstance = new _messageformat.default(pluralFuncs);
    this.icuInstance.addFormatters(this.wrapFormatters(formatters));
  }
  return (0, _createClass2.default)(IcuMessageFormatter, [{
    key: "wrapFormatters",
    value: function wrapFormatters(formatters) {
      return Object.keys(formatters).reduce(function (collector, formatterName) {
        collector[formatterName] = new _WrapperFormatter.default(formatters[formatterName]);
        return collector;
      }, {});
    }
  }, {
    key: "compileMessage",
    value: function compileMessage(message, messageHash) {
      var func = this.icuInstance.compile(message);
      this.dictionary[messageHash] = func;
      return func;
    }
  }, {
    key: "translate",
    value: function translate(message) {
      var context = arguments.length > 1 && arguments[1] !== undefined ? arguments[1] : {};
      var messageHash = (0, _utils.hashCode)(message);
      var formatter = this.dictionary[messageHash];
      if (!formatter) {
        formatter = this.compileMessage(message, messageHash);
      }
      return formatter(context);
    }
  }]);
}();
var _default = exports.default = IcuMessageFormatter;
module.exports = exports.default;
