// @sqs/i18n-ui/lib/icu-message-formatter/index.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var tslib_1 = require("tslib");
var icuMessageFormatter_1 = tslib_1.__importDefault(require("./icuMessageFormatter"));
var IcuMessageFormatter = (function () {
    function IcuMessageFormatter(i18nInstance) {
        var _this = this;
        this.formatIcuMessage = function (message, context) {
            return _this.icuFormatter.translate(message, context);
        };
        this.icuFormatter = new icuMessageFormatter_1.default({
            pluralGenerator: function (count, isOrdinal) {
                return isOrdinal
                    ? i18nInstance.pluralOrdinal(count)
                    : i18nInstance.plural(count);
            },
        }, {
            formatNumber: i18nInstance.formatNumber,
            formatCurrency: i18nInstance.formatCurrency,
            formatDateTime: i18nInstance.formatDateTime,
            formatRelativeTime: i18nInstance.formatRelativeTime,
        });
    }
    return IcuMessageFormatter;
}());
exports.default = IcuMessageFormatter;
