// @sqs/universal-state/lib/slices/accessPermissions.js
import "core-js/modules/es.array.iterator.js";
import "core-js/modules/es.object.to-string.js";
import "core-js/modules/es.set.js";
import "core-js/modules/web.dom-collections.iterator.js";
import { createSlice, createSelector } from '@reduxjs/toolkit';
import AccessPermissions from '@sqs/enums/AccessPermissions';
import { bootstrap as bootstrapActions } from '../actions';
export var initialState = null;
var slice = createSlice({
  name: 'accessPermissions',
  initialState: initialState,
  reducers: {
    update(_, action) {
      return action.payload;
    }
  },
  extraReducers: builder => {
    builder.addCase(bootstrapActions.hardBootstrap, (_, action) => action.payload.accessPermissions);
    builder.addCase(bootstrapActions.softBootstrap, (state, action) => {
      return state || action.payload.accessPermissions;
    });
  }
});
export var reducer = slice.reducer;
export var actions = {
  updateAccessPermissions: slice.actions.update
};

/**
 * Selectors accept a permission as either an enum key or value.
 * The following are equivalent:
 *
 * selectHasPermission(AccessPermission.CONFIG_SETTINGS)
 * selectHasPermission('CONFIG_SETTINGS')
 */

function hasPermission(permission, permissionSet) {
  var key = typeof permission === 'string' ? AccessPermissions[permission] : permission;
  if (key === undefined) {
    throw new Error(permission ? "Permission '".concat(permission, "' not found.") : 'Permission is undefined.');
  }
  return permissionSet.has(key);
}
var selectPermissions = rootState => rootState.accessPermissions;
var selectPermissionSet = createSelector(selectPermissions, accessPermissions => new Set(accessPermissions));
function selectHasPermission(permission) {
  return createSelector(selectPermissionSet, permissionSet => hasPermission(permission, permissionSet));
}
function selectHasAllPermissions(permissions) {
  return createSelector(selectPermissionSet, permissionSet => permissions.every(permission => hasPermission(permission, permissionSet)));
}
function selectHasAnyPermission(permissions) {
  return createSelector(selectPermissionSet, permissionSet => permissions.some(permission => hasPermission(permission, permissionSet)));
}
var selectIsAdmin = selectHasPermission(AccessPermissions.ADMIN);
export var selectors = {
  selectHasPermission,
  selectHasAllPermissions,
  selectHasAnyPermission,
  selectIsAdmin
};
