// @phensley/cldr-core/lib/schema/code/calendars.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.PERSIAN_INDICES = exports.PERSIAN = exports.JAPANESE_INDICES = exports.JAPANESE = exports.GREGORIAN_INDICES = exports.GREGORIAN = exports.BUDDHIST_INDICES = exports.BUDDHIST = exports.CALENDAR_INDICES = exports.WeekdaysIndex = exports.QuartersIndex = exports.FormatWidthIndex = exports.FieldWidthIndex = exports.EraTypeIndex = exports.DayPeriodIndex = exports.DateTimePatternFieldIndex = void 0;
var instructions_1 = require("../instructions");
var schema_1 = require("../schema");
var formats = function (name, rename, month) {
    return (0, instructions_1.scope)(name, rename, [
        (0, instructions_1.vector)('weekdays', ['field-width', 'weekday']),
        (0, instructions_1.vector)('months', ['field-width', "".concat(month, "-month")]),
        (0, instructions_1.vector)('quarters', ['field-width', 'quarter']),
        (0, instructions_1.vector)('dayPeriods', ['field-width', 'day-period', 'day-period-alt-key']),
    ]);
};
var calendarScope = function (name, scopeName) {
    return (0, instructions_1.scope)(scopeName, scopeName, [
        (0, instructions_1.vector)('eras', ['era-type', "".concat(name, "-era"), 'era-alt-key']),
        formats('format', 'format', name),
        formats('standAlone', 'standAlone', name),
        (0, instructions_1.vector)('availableFormats', ["".concat(name, "-available-format")]),
        (0, instructions_1.vector)('pluralFormats', ['plural-key', "".concat(name, "-plural-format")]),
        (0, instructions_1.vector)('intervalFormats', ['date-time-pattern-field', "".concat(name, "-interval-format")]),
        (0, instructions_1.vector)('dateFormats', ['format-width']),
        (0, instructions_1.vector)('timeFormats', ['format-width']),
        (0, instructions_1.vector)('dateTimeFormats', ['format-width']),
        (0, instructions_1.vector)('dateTimeFormatsAt', ['format-width']),
        (0, instructions_1.field)('intervalFormatFallback'),
    ]);
};
exports.DateTimePatternFieldIndex = new instructions_1.KeyIndexImpl(schema_1.DateTimePatternFieldValues);
exports.DayPeriodIndex = new instructions_1.KeyIndexImpl(schema_1.DayPeriodValues);
exports.EraTypeIndex = new instructions_1.KeyIndexImpl(schema_1.EraWidthValues);
exports.FieldWidthIndex = new instructions_1.KeyIndexImpl(schema_1.FieldWidthValues);
exports.FormatWidthIndex = new instructions_1.KeyIndexImpl(schema_1.FormatWidthValues);
exports.QuartersIndex = new instructions_1.KeyIndexImpl(schema_1.QuarterValues);
exports.WeekdaysIndex = new instructions_1.KeyIndexImpl(schema_1.WeekdayValues);
exports.CALENDAR_INDICES = {
    'date-time-pattern-field': exports.DateTimePatternFieldIndex,
    'day-period': exports.DayPeriodIndex,
    'era-type': exports.EraTypeIndex,
    'field-width': exports.FieldWidthIndex,
    'format-width': exports.FormatWidthIndex,
    quarter: exports.QuartersIndex,
    weekday: exports.WeekdaysIndex,
};
exports.BUDDHIST = calendarScope('buddhist', 'Buddhist');
exports.BUDDHIST_INDICES = {
    'buddhist-era': schema_1.BuddhistEraIndex,
    'buddhist-month': schema_1.GregorianMonthsIndex,
};
exports.GREGORIAN = calendarScope('gregorian', 'Gregorian');
exports.GREGORIAN_INDICES = {
    'gregorian-era': schema_1.GregorianEraIndex,
    'gregorian-month': schema_1.GregorianMonthsIndex,
};
exports.JAPANESE = calendarScope('japanese', 'Japanese');
exports.JAPANESE_INDICES = {
    'japanese-era': schema_1.JapaneseEraIndex,
    'japanese-month': schema_1.GregorianMonthsIndex,
};
exports.PERSIAN = calendarScope('persian', 'Persian');
exports.PERSIAN_INDICES = {
    'persian-era': schema_1.PersianEraIndex,
    'persian-month': schema_1.GregorianMonthsIndex,
};
