// @sqs/enums/AccountLoginOrigin.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;
/* File generated by JavaScriptEnumProcessor */
/* Derived from Java class com.squarespace.v6.utils.enums.AccountLoginOrigin */
/* Java class defined in repository squarespace-v6 */
var AccountLoginOrigin;
(function (AccountLoginOrigin) {
  AccountLoginOrigin["CUSTOMER_NOTIFICATION"] = "customer_notification";
  AccountLoginOrigin["MEMBER_AREA_ACCESS_PAGE"] = "member_area_access_page";
  AccountLoginOrigin["MEMBER_AREA_BLOCK"] = "member_area_block";
  AccountLoginOrigin["MEMBER_IMPORT_LINK"] = "member_import_link";
  AccountLoginOrigin["OTP_MEMBERSHIP_PURCHASE"] = "otp_membership_purchase";
  AccountLoginOrigin["ORDER_CONFIRMATION_PAGE"] = "order_confirmation_page";
  AccountLoginOrigin["ORDER_STATUS_PAGE"] = "order_status_page";
  AccountLoginOrigin["PAYWALL_PAGE"] = "paywall_page";
  AccountLoginOrigin["SIGNUP"] = "signup";
})(AccountLoginOrigin || (AccountLoginOrigin = {}));
var _default = AccountLoginOrigin;
exports["default"] = _default;
module.exports = exports.default;
