// @sqs/enums/AccountFlag.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;
/* File generated by JavaScriptEnumProcessor */
/* Derived from Java class com.squarespace.v6.utils.enums.MemberAccountFlag */
/* Java class defined in repository squarespace-v6 */
var AccountFlag;
(function (AccountFlag) {
  AccountFlag["COMMAND_CENTER"] = "command_center";
  AccountFlag["MEMBER_PRESERVATION_REQUEST"] = "member_preservation_request";
  AccountFlag["MEMBER_EMAIL_VERIFIED_MIGRATION"] = "member_email_verified_flag_migration";
  AccountFlag["MEMBER_STATUS_MIGRATION"] = "member_status_migration";
  AccountFlag["MEMBER_PASSWORD_EXTERNAL_ID_MIGRATION"] = "member_password_external_id_migration";
  AccountFlag["DASHBOARD_CROSS_SELL"] = "dashboard_cross_sell";
  AccountFlag["CUSTOPS_VERIFICATION_LINK"] = "custops_verification_link";
  AccountFlag["SEVEN_ONE_DEFAULT_THEME_REMOVAL_ENABLED_GROUP"] = "seven_one_default_theme_removal_enabled";
  AccountFlag["SEVEN_ONE_DEFAULT_THEME_REMOVAL_DISABLED_GROUP"] = "seven_one_default_theme_removal_disabled";
})(AccountFlag || (AccountFlag = {}));
var _default = AccountFlag;
exports["default"] = _default;
module.exports = exports.default;
