// @sqs/enums/AnalyticsFeature.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;
/* File generated by JavaScriptEnumProcessor */
/* Derived from Java class com.squarespace.v6.utils.enums.census.AnalyticsFeature */
/* Java class defined in repository squarespace-v6 */
var AnalyticsFeature;
(function (AnalyticsFeature) {
  AnalyticsFeature["ACTIVITY_LOG"] = "activityLog";
  AnalyticsFeature["TRAFFIC_OVERVIEW"] = "trafficOverview";
  AnalyticsFeature["SALES_OVERVIEW"] = "salesOverview";
  AnalyticsFeature["DEVICE_TYPE"] = "deviceType";
  AnalyticsFeature["POPULAR_CONTENT"] = "popularContent";
  AnalyticsFeature["TOP_PRODUCTS"] = "topProducts";
  AnalyticsFeature["RSS_SUBSCRIBERS"] = "rssSubscribers";
  AnalyticsFeature["SITE_SEARCH"] = "siteSearch";
  AnalyticsFeature["GOOGLE_SEARCH"] = "googleSearch";
  AnalyticsFeature["OTHER_SEARCH"] = "otherSearch";
  AnalyticsFeature["GEOGRAPHY"] = "geography";
  AnalyticsFeature["CONVERSIONS"] = "conversions";
  AnalyticsFeature["TRAFFIC_SOURCES"] = "trafficSources";
  AnalyticsFeature["PURCHASE_FUNNEL"] = "purchaseFunnel";
  AnalyticsFeature["ABANDONED_CART"] = "abandonedCart";
  AnalyticsFeature["ANOMALY_PUSH_NOTIFICATIONS"] = "anomalyPushNotifications";
  AnalyticsFeature["SALES_BY_PRODUCT"] = "salesByProduct";
})(AnalyticsFeature || (AnalyticsFeature = {}));
var _default = AnalyticsFeature;
exports["default"] = _default;
module.exports = exports.default;
