// @sqs/enums/AnalyticsFeatureSetIdentifier.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;
/* File generated by JavaScriptEnumProcessor */
/* Derived from Java class com.squarespace.v6.utils.enums.census.AnalyticsFeatureSetIdentifier */
/* Java class defined in repository squarespace-v6 */
var AnalyticsFeatureSetIdentifier;
(function (AnalyticsFeatureSetIdentifier) {
  AnalyticsFeatureSetIdentifier["PERSONAL_NON_SELLING"] = "personal-non-selling";
  AnalyticsFeatureSetIdentifier["PERSONAL_SELLING"] = "personal-selling";
  AnalyticsFeatureSetIdentifier["BUSINESS_NON_SELLING"] = "business-non-selling";
  AnalyticsFeatureSetIdentifier["BUSINESS_SELLING"] = "business-selling";
  AnalyticsFeatureSetIdentifier["COMMERCE_BASIC"] = "commerce-basic";
  AnalyticsFeatureSetIdentifier["COMMERCE_ADVANCED"] = "commerce-advanced";
})(AnalyticsFeatureSetIdentifier || (AnalyticsFeatureSetIdentifier = {}));
var _default = AnalyticsFeatureSetIdentifier;
exports["default"] = _default;
module.exports = exports.default;
