// @sqs/enums/ApiKeyScope.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;
/* File generated by JavaScriptEnumProcessor */
/* Derived from Java class com.squarespace.api.apikeys.models.PublicApiKeyScope */
/* Java class defined in repository squarespace-v6 */
var ApiKeyScope;
(function (ApiKeyScope) {
  ApiKeyScope["DISCOUNT"] = "DISCOUNT";
  ApiKeyScope["DISCOUNT_READONLY"] = "DISCOUNT_READONLY";
  ApiKeyScope["FORM"] = "FORM";
  ApiKeyScope["FORM_READONLY"] = "FORM_READONLY";
  ApiKeyScope["INVENTORY"] = "INVENTORY";
  ApiKeyScope["INVENTORY_READONLY"] = "INVENTORY_READONLY";
  ApiKeyScope["ORDER"] = "ORDER";
  ApiKeyScope["ORDER_READONLY"] = "ORDER_READONLY";
  ApiKeyScope["PRODUCT"] = "PRODUCT";
  ApiKeyScope["PRODUCT_READONLY"] = "PRODUCT_READONLY";
  ApiKeyScope["PROFILE"] = "PROFILE";
  ApiKeyScope["PROFILE_READONLY"] = "PROFILE_READONLY";
  ApiKeyScope["TRANSACTION"] = "TRANSACTION";
  ApiKeyScope["TRANSACTION_READONLY"] = "TRANSACTION_READONLY";
  ApiKeyScope["UI_TOOLKIT"] = "UI_TOOLKIT";
})(ApiKeyScope || (ApiKeyScope = {}));
var _default = ApiKeyScope;
exports["default"] = _default;
module.exports = exports.default;
