// @sqs/enums/BillingProductPurchaseType.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;
/* File generated by JavaScriptEnumProcessor */
/* Derived from Java class com.squarespace.v6.utils.enums.BillingProductPurchaseType */
/* Java class defined in repository squarespace-v6 */
var BillingProductPurchaseType;
(function (BillingProductPurchaseType) {
  BillingProductPurchaseType["WEBSITE"] = "Website";
  BillingProductPurchaseType["DOMAIN"] = "Domain";
  BillingProductPurchaseType["GETTY"] = "Getty";
  BillingProductPurchaseType["GOOGLE_APPS"] = "G Suite";
  BillingProductPurchaseType["CAMPAIGNS"] = "Campaigns";
  BillingProductPurchaseType["SCHEDULING"] = "Scheduling";
  BillingProductPurchaseType["MEMBER_AREAS"] = "Member Areas";
  BillingProductPurchaseType["SHIPPING_LABEL"] = "Shipping Label";
  BillingProductPurchaseType["BUNDLE"] = "Multiple subscriptions";
  BillingProductPurchaseType["WEBSITE_BUNDLE"] = "Website + multiple subscriptions";
  BillingProductPurchaseType["SQUARESPACE_BUNDLE"] = "Squarespace Bundle";
})(BillingProductPurchaseType || (BillingProductPurchaseType = {}));
var _default = BillingProductPurchaseType;
exports["default"] = _default;
module.exports = exports.default;
