// @sqs/enums/BillingRequestState.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;
/* File generated by JavaScriptEnumProcessor */
/* Derived from Java class com.squarespace.v6.model.billing.billingrequests.BillingRequestState */
/* Java class defined in repository squarespace-v6 */
var BillingRequestState;
(function (BillingRequestState) {
  BillingRequestState["PENDING"] = "pending";
  BillingRequestState["VALIDATED"] = "validated";
  BillingRequestState["AWAITING_ASYNC_PAYMENT"] = "awaiting-async-payment";
  BillingRequestState["REQUIRES_ACTION"] = "requires-action";
  BillingRequestState["REQUIRES_ACTION_PASSED"] = "requires-action-passed";
  BillingRequestState["FULFILLING"] = "fulfilling";
  BillingRequestState["FULFILLED"] = "fulfilled";
  BillingRequestState["COMPLETED"] = "completed";
  BillingRequestState["FAILED"] = "failed";
})(BillingRequestState || (BillingRequestState = {}));
var _default = BillingRequestState;
exports["default"] = _default;
module.exports = exports.default;
