// @sqs/enums/BlendMode.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;
/* File generated by JavaScriptEnumProcessor */
/* Derived from Java class com.squarespace.v6.utils.enums.blocks.BlendMode */
/* Java class defined in repository cms-legacy-data */
var BlendMode;
(function (BlendMode) {
  BlendMode["NORMAL"] = "normal";
  BlendMode["MULTIPLY"] = "multiply";
  BlendMode["SCREEN"] = "screen";
  BlendMode["OVERLAY"] = "overlay";
  BlendMode["DARKEN"] = "darken";
  BlendMode["LIGHTEN"] = "lighten";
  BlendMode["COLOR_DODGE"] = "color-dodge";
  BlendMode["COLOR_BURN"] = "color-burn";
  BlendMode["HARD_LIGHT"] = "hard-light";
  BlendMode["SOFT_LIGHT"] = "soft-light";
  BlendMode["DIFFERENCE"] = "difference";
  BlendMode["EXCLUSION"] = "exclusion";
  BlendMode["HUE"] = "hue";
  BlendMode["SATURATION"] = "saturation";
  BlendMode["COLOR"] = "color";
  BlendMode["LUMINOSITY"] = "luminosity";
})(BlendMode || (BlendMode = {}));
var _default = BlendMode;
exports["default"] = _default;
module.exports = exports.default;
