// @sqs/enums/BulkProductManagementErrorType.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;
/* File generated by JavaScriptEnumProcessor */
/* Derived from Java class com.squarespace.v6.model.BulkProductManagementErrorType */
/* Java class defined in repository squarespace-v6 */
var BulkProductManagementErrorType;
(function (BulkProductManagementErrorType) {
  BulkProductManagementErrorType["INVALID_CELL"] = "INVALID_CELL";
  BulkProductManagementErrorType["MISSING_COLUMN"] = "MISSING_COLUMN";
  BulkProductManagementErrorType["PRODUCT_NOT_FOUND"] = "PRODUCT_NOT_FOUND";
  BulkProductManagementErrorType["PRODUCT_PAGE_NOT_FOUND"] = "PRODUCT_PAGE_NOT_FOUND";
  BulkProductManagementErrorType["VARIANT_NOT_FOUND"] = "VARIANT_NOT_FOUND";
  BulkProductManagementErrorType["UNSUPPORTED_PRODUCT_TYPE"] = "UNSUPPORTED_PRODUCT_TYPE";
  BulkProductManagementErrorType["PRODUCT_TYPE_UNMATCHED"] = "PRODUCT_TYPE_UNMATCHED";
  BulkProductManagementErrorType["CATEGORY_ASSIGNMENT_FAILURE"] = "CATEGORY_ASSIGNMENT_FAILURE";
  BulkProductManagementErrorType["IMAGE_IMPORT_FAILURE"] = "IMAGE_IMPORT_FAILURE";
  BulkProductManagementErrorType["INVALID_PRODUCT"] = "INVALID_PRODUCT";
  BulkProductManagementErrorType["DUPLICATE_VARIANT_ID"] = "DUPLICATE_VARIANT_ID";
  BulkProductManagementErrorType["DUPLICATE_URL_ID"] = "DUPLICATE_URL_ID";
  BulkProductManagementErrorType["PRODUCT_PAGE_SIZE_LIMIT_REACHED"] = "PRODUCT_PAGE_SIZE_LIMIT_REACHED";
  BulkProductManagementErrorType["UNABLE_TO_PARSE_FILE"] = "UNABLE_TO_PARSE_FILE";
  BulkProductManagementErrorType["UNKNOWN_ERROR"] = "UNKNOWN_ERROR";
})(BulkProductManagementErrorType || (BulkProductManagementErrorType = {}));
var _default = BulkProductManagementErrorType;
exports["default"] = _default;
module.exports = exports.default;
