// @sqs/enums/ButtonPackId.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;
/* File generated by JavaScriptEnumProcessor */
/* Derived from Java class com.squarespace.v6.utils.enums.ButtonPackId */
/* Java class defined in repository squarespace-v6 */
var ButtonPackId;
(function (ButtonPackId) {
  ButtonPackId["BUTTON_PACK_01"] = 1;
  ButtonPackId["BUTTON_PACK_02"] = 2;
  ButtonPackId["BUTTON_PACK_03"] = 3;
  ButtonPackId["BUTTON_PACK_04"] = 4;
  ButtonPackId["BUTTON_PACK_05"] = 5;
  ButtonPackId["BUTTON_PACK_06"] = 6;
  ButtonPackId["BUTTON_PACK_07"] = 7;
  ButtonPackId["BUTTON_PACK_08"] = 8;
  ButtonPackId["BUTTON_PACK_09"] = 9;
  ButtonPackId["BUTTON_PACK_10"] = 10;
  ButtonPackId["BUTTON_PACK_11"] = 11;
  ButtonPackId["BUTTON_PACK_12"] = 12;
  ButtonPackId["BUTTON_PACK_13"] = 13;
  ButtonPackId["BUTTON_PACK_14"] = 14;
  ButtonPackId["BUTTON_PACK_15"] = 15;
  ButtonPackId["BUTTON_PACK_16"] = 16;
  ButtonPackId["BUTTON_PACK_17"] = 17;
  ButtonPackId["BUTTON_PACK_18"] = 18;
  ButtonPackId["BUTTON_PACK_19"] = 19;
  ButtonPackId["BUTTON_PACK_20"] = 20;
  ButtonPackId["BUTTON_PACK_21"] = 21;
  ButtonPackId["BUTTON_PACK_22"] = 22;
  ButtonPackId["BUTTON_PACK_23"] = 23;
  ButtonPackId["BUTTON_PACK_24"] = 24;
  ButtonPackId["BUTTON_PACK_25"] = 25;
  ButtonPackId["BUTTON_PACK_26"] = 26;
  ButtonPackId["BUTTON_PACK_27"] = 27;
  ButtonPackId["BUTTON_PACK_28"] = 28;
  ButtonPackId["BUTTON_PACK_29"] = 29;
  ButtonPackId["BUTTON_PACK_30"] = 30;
  ButtonPackId["BUTTON_PACK_31"] = 31;
  ButtonPackId["BUTTON_PACK_32"] = 32;
  ButtonPackId["BUTTON_PACK_33"] = 33;
  ButtonPackId["BUTTON_PACK_34"] = 34;
  ButtonPackId["BUTTON_PACK_35"] = 35;
  ButtonPackId["BUTTON_PACK_36"] = 36;
  ButtonPackId["BUTTON_PACK_37"] = 37;
  ButtonPackId["BUTTON_PACK_38"] = 38;
  ButtonPackId["BUTTON_PACK_39"] = 39;
  ButtonPackId["BUTTON_PACK_40"] = 40;
  ButtonPackId["BUTTON_PACK_41"] = 41;
  ButtonPackId["BUTTON_PACK_42"] = 42;
  ButtonPackId["BUTTON_PACK_43"] = 43;
  ButtonPackId["BUTTON_PACK_44"] = 44;
  ButtonPackId["BUTTON_PACK_45"] = 45;
  ButtonPackId["BUTTON_PACK_46"] = 46;
  ButtonPackId["BUTTON_PACK_47"] = 47;
  ButtonPackId["BUTTON_PACK_48"] = 48;
  ButtonPackId["BUTTON_PACK_49"] = 49;
  ButtonPackId["BUTTON_PACK_50"] = 50;
  ButtonPackId["BUTTON_PACK_51"] = 51;
  ButtonPackId["BUTTON_PACK_52"] = 52;
  ButtonPackId["BUTTON_PACK_53"] = 53;
  ButtonPackId["BUTTON_PACK_54"] = 54;
  ButtonPackId["BUTTON_PACK_55"] = 55;
  ButtonPackId["BUTTON_PACK_56"] = 56;
})(ButtonPackId || (ButtonPackId = {}));
var _default = ButtonPackId;
exports["default"] = _default;
module.exports = exports.default;
