// @sqs/enums/CartShippingOptionsStatus.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;
/* File generated by JavaScriptEnumProcessor */
/* Derived from Java class com.squarespace.v6.business.commerce.cart.shipping.CartShippingOptionsStatus */
/* Java class defined in repository squarespace-v6 */
var CartShippingOptionsStatus;
(function (CartShippingOptionsStatus) {
  CartShippingOptionsStatus["UNKNOWN"] = "UNKNOWN";
  CartShippingOptionsStatus["SHIPPING_NOT_REQUIRED"] = "SHIPPING_NOT_REQUIRED";
  CartShippingOptionsStatus["APPLICABLE_SHIPPING_OPTIONS"] = "APPLICABLE_SHIPPING_OPTIONS";
  CartShippingOptionsStatus["NO_APPLICABLE_SHIPPING_OPTIONS"] = "NO_APPLICABLE_SHIPPING_OPTIONS";
  CartShippingOptionsStatus["INCOMPLETE_ADDRESS"] = "INCOMPLETE ADDRESS";
  CartShippingOptionsStatus["INVALID_US_ZIP_CODE"] = "INVALID_US_ZIP_CODE";
  CartShippingOptionsStatus["POSTAL_CODE_NOT_APPLICABLE"] = "POSTAL_CODE_NOT_APPLICABLE";
})(CartShippingOptionsStatus || (CartShippingOptionsStatus = {}));
var _default = CartShippingOptionsStatus;
exports["default"] = _default;
module.exports = exports.default;
