// @sqs/enums/ContractOperationIneligibilityReason.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;
/* File generated by JavaScriptEnumProcessor */
/* Derived from Java class com.squarespace.api.billing.models.bundling.ContractOperationIneligibilityReason */
/* Java class defined in repository squarespace-v6 */
var ContractOperationIneligibilityReason;
(function (ContractOperationIneligibilityReason) {
  ContractOperationIneligibilityReason["RENEWAL_IN_PROGRESS"] = "RENEWAL_IN_PROGRESS";
  ContractOperationIneligibilityReason["INSUFFICIENT_TIME_BEFORE_RENEWAL"] = "INSUFFICIENT_TIME_BEFORE_RENEWAL";
  ContractOperationIneligibilityReason["PAYMENT_PROCESSING"] = "PAYMENT_PROCESSING";
  ContractOperationIneligibilityReason["SUBSCRIPTION_PAST_DUE"] = "SUBSCRIPTION_PAST_DUE";
  ContractOperationIneligibilityReason["INELIGIBLE_STATE"] = "INELIGIBLE_STATE";
  ContractOperationIneligibilityReason["INSUFFICIENT_TIME_IN_REACTIVATION_WINDOW"] = "INSUFFICIENT_TIME_IN_REACTIVATION_WINDOW";
  ContractOperationIneligibilityReason["NOT_SUPPORTED"] = "NOT_SUPPORTED";
  ContractOperationIneligibilityReason["RENEWAL_PREPAID"] = "RENEWAL_PREPAID";
  ContractOperationIneligibilityReason["NO_SCHEDULED_PLAN_CHANGE"] = "NO_SCHEDULED_PLAN_CHANGE";
  ContractOperationIneligibilityReason["IN_MOVE_TERM"] = "IN_MOVE_TERM";
  ContractOperationIneligibilityReason["IN_MIGRATION_TERM"] = "IN_MIGRATION_TERM";
  ContractOperationIneligibilityReason["DOMAIN_NOT_VERIFIED"] = "DOMAIN_NOT_VERIFIED";
  ContractOperationIneligibilityReason["SUBSCRIPTION_SUSPENDED"] = "SUBSCRIPTION_SUSPENDED";
  ContractOperationIneligibilityReason["UNACCEPTED_TOS"] = "UNACCEPTED_TOS";
  ContractOperationIneligibilityReason["INELIGIBLE_PAYMENT_PLAN"] = "INELIGIBLE_PAYMENT_PLAN";
  ContractOperationIneligibilityReason["NO_HIGHER_TIER_PLANS_AVAILABLE"] = "NO_HIGHER_TIER_PLANS_AVAILABLE";
})(ContractOperationIneligibilityReason || (ContractOperationIneligibilityReason = {}));
var _default = ContractOperationIneligibilityReason;
exports["default"] = _default;
module.exports = exports.default;
