// @sqs/enums/ContractOperationType.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;
/* File generated by JavaScriptEnumProcessor */
/* Derived from Java class com.squarespace.api.billing.models.bundling.ContractOperationType */
/* Java class defined in repository squarespace-v6 */
var ContractOperationType;
(function (ContractOperationType) {
  ContractOperationType["QUANTITY_CHANGE_REMOVE"] = "QUANTITY_CHANGE_REMOVE";
  ContractOperationType["QUANTITY_CHANGE_ADD"] = "QUANTITY_CHANGE_ADD";
  ContractOperationType["PLAN_CHANGE"] = "PLAN_CHANGE";
  ContractOperationType["UNSCHEDULE_PLAN_CHANGE"] = "UNSCHEDULE_PLAN_CHANGE";
  ContractOperationType["MOVE"] = "MOVE";
  ContractOperationType["CANCEL"] = "CANCEL";
  ContractOperationType["AUTO_RENEW"] = "AUTO_RENEW";
  ContractOperationType["REACTIVATION"] = "REACTIVATION";
  ContractOperationType["TRANSFER"] = "TRANSFER";
  ContractOperationType["TERM_EXTENSION"] = "TERM_EXTENSION";
})(ContractOperationType || (ContractOperationType = {}));
var _default = ContractOperationType;
exports["default"] = _default;
module.exports = exports.default;
