// @sqs/enums/CountryWithoutPostalCode.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;
/* File generated by JavaScriptEnumProcessor */
/* Derived from Java class com.squarespace.domains.commons.contacts.validation.CountryWithoutPostalCode */
/* Java class defined in repository domains-commons */
var CountryWithoutPostalCode;
(function (CountryWithoutPostalCode) {
  CountryWithoutPostalCode["AO"] = "AO";
  CountryWithoutPostalCode["AG"] = "AG";
  CountryWithoutPostalCode["AW"] = "AW";
  CountryWithoutPostalCode["AZ"] = "AZ";
  CountryWithoutPostalCode["BS"] = "BS";
  CountryWithoutPostalCode["BZ"] = "BZ";
  CountryWithoutPostalCode["BJ"] = "BJ";
  CountryWithoutPostalCode["BW"] = "BW";
  CountryWithoutPostalCode["BQ"] = "BQ";
  CountryWithoutPostalCode["BF"] = "BF";
  CountryWithoutPostalCode["BI"] = "BI";
  CountryWithoutPostalCode["CM"] = "CM";
  CountryWithoutPostalCode["CF"] = "CF";
  CountryWithoutPostalCode["KM"] = "KM";
  CountryWithoutPostalCode["CG"] = "CG";
  CountryWithoutPostalCode["CD"] = "CD";
  CountryWithoutPostalCode["CK"] = "CK";
  CountryWithoutPostalCode["CI"] = "CI";
  CountryWithoutPostalCode["CW"] = "CW";
  CountryWithoutPostalCode["DJ"] = "DJ";
  CountryWithoutPostalCode["CO"] = "CO";
  CountryWithoutPostalCode["DM"] = "DM";
  CountryWithoutPostalCode["GQ"] = "GQ";
  CountryWithoutPostalCode["ER"] = "ER";
  CountryWithoutPostalCode["FJ"] = "FJ";
  CountryWithoutPostalCode["TF"] = "TF";
  CountryWithoutPostalCode["GM"] = "GM";
  CountryWithoutPostalCode["GH"] = "GH";
  CountryWithoutPostalCode["GD"] = "GD";
  CountryWithoutPostalCode["GY"] = "GY";
  CountryWithoutPostalCode["HK"] = "HK";
  CountryWithoutPostalCode["JM"] = "JM";
  CountryWithoutPostalCode["KI"] = "KI";
  CountryWithoutPostalCode["LY"] = "LY";
  CountryWithoutPostalCode["MO"] = "MO";
  CountryWithoutPostalCode["MW"] = "MW";
  CountryWithoutPostalCode["ML"] = "ML";
  CountryWithoutPostalCode["MR"] = "MR";
  CountryWithoutPostalCode["NR"] = "NR";
  CountryWithoutPostalCode["NU"] = "NU";
  CountryWithoutPostalCode["KP"] = "KP";
  CountryWithoutPostalCode["QA"] = "QA";
  CountryWithoutPostalCode["RW"] = "RW";
  CountryWithoutPostalCode["KN"] = "KN";
  CountryWithoutPostalCode["ST"] = "ST";
  CountryWithoutPostalCode["SC"] = "SC";
  CountryWithoutPostalCode["SL"] = "SL";
  CountryWithoutPostalCode["SB"] = "SB";
  CountryWithoutPostalCode["SR"] = "SR";
  CountryWithoutPostalCode["SY"] = "SY";
  CountryWithoutPostalCode["TL"] = "TL";
  CountryWithoutPostalCode["TG"] = "TG";
  CountryWithoutPostalCode["TK"] = "TK";
  CountryWithoutPostalCode["TO"] = "TO";
  CountryWithoutPostalCode["TV"] = "TV";
  CountryWithoutPostalCode["UG"] = "UG";
  CountryWithoutPostalCode["AE"] = "AE";
  CountryWithoutPostalCode["VU"] = "VU";
  CountryWithoutPostalCode["YE"] = "YE";
  CountryWithoutPostalCode["ZW"] = "ZW";
  CountryWithoutPostalCode["TZ"] = "TZ";
  CountryWithoutPostalCode["GN"] = "GN";
  CountryWithoutPostalCode["SV"] = "SV";
  CountryWithoutPostalCode["IL"] = "IL";
  CountryWithoutPostalCode["PE"] = "PE";
  CountryWithoutPostalCode["CL"] = "CL";
  CountryWithoutPostalCode["PH"] = "PH";
  CountryWithoutPostalCode["VN"] = "VN";
  CountryWithoutPostalCode["ID"] = "ID";
  CountryWithoutPostalCode["CY"] = "CY";
  CountryWithoutPostalCode["TH"] = "TH";
  CountryWithoutPostalCode["KZ"] = "KZ";
  CountryWithoutPostalCode["BG"] = "BG";
  CountryWithoutPostalCode["TN"] = "TN";
  CountryWithoutPostalCode["GT"] = "GT";
  CountryWithoutPostalCode["DZ"] = "DZ";
  CountryWithoutPostalCode["AR"] = "AR";
  CountryWithoutPostalCode["IE"] = "IE";
  CountryWithoutPostalCode["LC"] = "LC";
  CountryWithoutPostalCode["SA"] = "SA";
  CountryWithoutPostalCode["ME"] = "ME";
  CountryWithoutPostalCode["AM"] = "AM";
  CountryWithoutPostalCode["AD"] = "AD";
  CountryWithoutPostalCode["KH"] = "KH";
  CountryWithoutPostalCode["BO"] = "BO";
  CountryWithoutPostalCode["EC"] = "EC";
  CountryWithoutPostalCode["EG"] = "EG";
  CountryWithoutPostalCode["KE"] = "KE";
  CountryWithoutPostalCode["OM"] = "OM";
  CountryWithoutPostalCode["PA"] = "PA";
  CountryWithoutPostalCode["PK"] = "PK";
  CountryWithoutPostalCode["SI"] = "SI";
  CountryWithoutPostalCode["SS"] = "SS";
  CountryWithoutPostalCode["UM"] = "UM";
  CountryWithoutPostalCode["VG"] = "VG";
})(CountryWithoutPostalCode || (CountryWithoutPostalCode = {}));
var _default = CountryWithoutPostalCode;
exports["default"] = _default;
module.exports = exports.default;
