// @sqs/enums/DiscountValidationErrorType.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;
/* File generated by JavaScriptEnumProcessor */
/* Derived from Java class com.squarespace.v6.business.commerce.discount.DiscountValidationErrorType */
/* Java class defined in repository squarespace-v6 */
var DiscountValidationErrorType;
(function (DiscountValidationErrorType) {
  DiscountValidationErrorType["AT_LEAST_ONE_CATEGORY"] = "AT_LEAST_ONE_CATEGORY";
  DiscountValidationErrorType["AUTO_TRIGGER_NOT_SUPPORTED"] = "AUTO_TRIGGER_NOT_SUPPORTED";
  DiscountValidationErrorType["BUY_X_GET_Y_CRITERIA_APPLIES_TO_REQUIRED"] = "BUY_X_GET_Y_CRITERIA_APPLIES_TO_REQUIRED";
  DiscountValidationErrorType["BUY_X_GET_Y_CRITERIA_AT_LEAST_ONE_CATEGORY"] = "BUY_X_GET_Y_CRITERIA_AT_LEAST_ONE_CATEGORY";
  DiscountValidationErrorType["BUY_X_GET_Y_CRITERIA_AT_LEAST_ONE_PRODUCT"] = "BUY_X_GET_Y_CRITERIA_AT_LEAST_ONE_PRODUCT";
  DiscountValidationErrorType["BUY_X_GET_Y_CRITERIA_NON_EXISTING_CATEGORY"] = "BUY_X_GET_Y_CRITERIA_NON_EXISTING_CATEGORY";
  DiscountValidationErrorType["BUY_X_GET_Y_CRITERIA_NON_EXISTING_PRODUCT"] = "BUY_X_GET_Y_CRITERIA_NON_EXISTING_PRODUCT";
  DiscountValidationErrorType["BUY_X_GET_Y_CRITERIA_REQUIRED_AMOUNT_GREATER_THAN_ZERO"] = "BUY_X_GET_Y_CRITERIA_REQUIRED_AMOUNT_GREATER_THAN_ZERO";
  DiscountValidationErrorType["BUY_X_GET_Y_INVALID_CRITERIA_TEMPLATE_COMBINATION"] = "BUY_X_GET_Y_INVALID_CRITERIA_TEMPLATE_COMBINATION";
  DiscountValidationErrorType["BUY_X_GET_Y_NOT_SUPPORTED"] = "BUY_X_GET_Y_NOT_SUPPORTED";
  DiscountValidationErrorType["BUY_X_GET_Y_REWARD_APPLIES_TO_REQUIRED"] = "BUY_X_GET_Y_REWARD_APPLIES_TO_REQUIRED";
  DiscountValidationErrorType["BUY_X_GET_Y_REWARD_AT_LEAST_ONE_CATEGORY"] = "BUY_X_GET_Y_REWARD_AT_LEAST_ONE_CATEGORY";
  DiscountValidationErrorType["BUY_X_GET_Y_REWARD_AT_LEAST_ONE_PRODUCT"] = "BUY_X_GET_Y_REWARD_AT_LEAST_ONE_PRODUCT";
  DiscountValidationErrorType["BUY_X_GET_Y_REWARD_NON_EXISTING_CATEGORY"] = "BUY_X_GET_Y_REWARD_NON_EXISTING_CATEGORY";
  DiscountValidationErrorType["BUY_X_GET_Y_REWARD_NON_EXISTING_PRODUCT"] = "BUY_X_GET_Y_REWARD_NON_EXISTING_PRODUCT";
  DiscountValidationErrorType["BUY_X_GET_Y_REWARD_REQUIRED_AMOUNT_GREATER_THAN_ZERO"] = "BUY_X_GET_Y_REWARD_REQUIRED_AMOUNT_GREATER_THAN_ZERO";
  DiscountValidationErrorType["CRITERIA_REQUIRED"] = "CRITERIA_REQUIRED";
  DiscountValidationErrorType["DISCOUNT_DISABLED"] = "DISCOUNT_DISABLED";
  DiscountValidationErrorType["DISCOUNT_LIMIT_GREATER_THAN_ZERO"] = "DISCOUNT_LIMIT_GREATER_THAN_ZERO";
  DiscountValidationErrorType["FIXED_AMOUNT_GREATER_THAN_ZERO"] = "FIXED_AMOUNT_GREATER_THAN_ZERO";
  DiscountValidationErrorType["INVALID_SUBSCRIPTION_OPTIONS"] = "INVALID_SUBSCRIPTION_OPTIONS";
  DiscountValidationErrorType["LIMITED_USE_NOT_SUPPPORTED"] = "LIMITED_USE_NOT_SUPPPORTED";
  DiscountValidationErrorType["MAX_ALLOWED_AUTO_TRIGGER_REACHED"] = "MAX_ALLOWED_AUTO_TRIGGER_REACHED";
  DiscountValidationErrorType["MAX_APPLICATIONS_PER_CART_NON_NEGATIVE"] = "MAX_APPLICATIONS_PER_CART_NON_NEGATIVE";
  DiscountValidationErrorType["MINIMUM_CART_TOTAL_GREATER_THAN_ZERO"] = "MINIMUM_CART_TOTAL_GREATER_THAN_ZERO";
  DiscountValidationErrorType["NAME_REQUIRED"] = "NAME_REQUIRED";
  DiscountValidationErrorType["NON_EXISTING_CATEGORY"] = "NON_EXISTING_CATEGORY";
  DiscountValidationErrorType["NON_EXISTING_PRICING_OPTION"] = "NON_EXISTING_PRICING_OPTION";
  DiscountValidationErrorType["NON_EXISTING_PRODUCT"] = "NON_EXISTING_PRODUCT";
  DiscountValidationErrorType["NON_EXISTING_SHIPPING_OPTION"] = "NON_EXISTING_SHIPPING_OPTION";
  DiscountValidationErrorType["PERCENTAGE_INVALID"] = "PERCENTAGE_INVALID";
  DiscountValidationErrorType["PRODUCT_ID_REQUIRED"] = "PRODUCT_ID_REQUIRED";
  DiscountValidationErrorType["PRODUCT_QUANTITY_REQUIRED"] = "PRODUCT_QUANTITY_REQUIRED";
  DiscountValidationErrorType["PROMO_CODE_ALREADY_EXISTS"] = "PROMO_CODE_ALREADY_EXISTS";
  DiscountValidationErrorType["PROMO_CODE_REQUIRED"] = "PROMO_CODE_REQUIRED";
  DiscountValidationErrorType["PROMO_CODE_RESERVED"] = "PROMO_CODE_RESERVED";
  DiscountValidationErrorType["START_DATE_REQUIRED"] = "START_DATE_REQUIRED";
  DiscountValidationErrorType["TEMPLATE_REQUIRED"] = "TEMPLATE_REQUIRED";
  DiscountValidationErrorType["TRIGGER_REQUIRED"] = "TRIGGER_REQUIRED";
  DiscountValidationErrorType["UNKNOWN"] = "UNKNOWN";
})(DiscountValidationErrorType || (DiscountValidationErrorType = {}));
var _default = DiscountValidationErrorType;
exports["default"] = _default;
module.exports = exports.default;
