// @sqs/enums/DisputeReason.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;
/* File generated by JavaScriptEnumProcessor */
/* Derived from Java class com.squarespace.api.admin.billing.payments.DisputeReason */
/* Java class defined in repository squarespace-v6 */
var DisputeReason;
(function (DisputeReason) {
  DisputeReason["CANNOT_PROCESS"] = "cannot_process";
  DisputeReason["CREDIT_NOT_PROCESSED"] = "credit_not_processed";
  DisputeReason["DUPLICATE"] = "duplicate";
  DisputeReason["FRAUDULENT"] = "fraudulent";
  DisputeReason["GENERAL"] = "general";
  DisputeReason["INCORRECT_ACCOUNT_DETAILS"] = "incorrect_account_details";
  DisputeReason["INCORRECT_AMOUNT"] = "incorrect_amount";
  DisputeReason["PRODUCT_NOT_RECEIVED"] = "product_not_received";
  DisputeReason["PRODUCT_UNACCEPTABLE"] = "product_unacceptable";
  DisputeReason["SUBSCRIPTION_CANCELED"] = "subscription_canceled";
  DisputeReason["UNRECOGNIZED"] = "unrecognized";
})(DisputeReason || (DisputeReason = {}));
var _default = DisputeReason;
exports["default"] = _default;
module.exports = exports.default;
