// @sqs/enums/DnssecAlgorithm.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;
/* File generated by JavaScriptEnumProcessor */
/* Derived from Java class com.squarespace.v6.business.components.domains.nameservers.models.DnssecAlgorithm */
/* Java class defined in repository squarespace-v6 */
var DnssecAlgorithm;
(function (DnssecAlgorithm) {
  DnssecAlgorithm["RSA_MD5"] = "RSA/MD5";
  DnssecAlgorithm["DIFFIE_HELLMAN"] = "DIFFIE-HELLMAN";
  DnssecAlgorithm["DSA_SHA1"] = "DSA/SHA1";
  DnssecAlgorithm["RSA_SHA_1"] = "RSA/SHA-1";
  DnssecAlgorithm["DSA_NSEC3_SHA1"] = "DSA-NSEC3-SHA1";
  DnssecAlgorithm["RSASHA1_NSEC3_SHA1"] = "RSASHA1-NSEC3-SHA1";
  DnssecAlgorithm["RSA_SHA_256"] = "RSA/SHA-256";
  DnssecAlgorithm["RSA_SHA_512"] = "RSA/SHA-512";
  DnssecAlgorithm["GOST_R_34_10_2001"] = "GOST R 34.10-2001";
  DnssecAlgorithm["ECDSA_CURVE_P_256_WITH_SHA_256"] = "ECDSA CURVE P-256 WITH SHA-256";
  DnssecAlgorithm["ECDSA_CURVE_P_384_WITH_SHA_384"] = "ECDSA CURVE P-384 WITH SHA-384";
  DnssecAlgorithm["PRIVATE_ALGORITHMS_DOMAIN_NAME"] = "PRIVATE ALGORITHMS - DOMAIN NAME";
  DnssecAlgorithm["PRIVATE_ALGORITHMS_OID"] = "PRIVATE ALGORITHMS - OID";
  DnssecAlgorithm["INDIRECT"] = "INDIRECT";
  DnssecAlgorithm["ED_25519"] = "ED 25519";
  DnssecAlgorithm["ED_448"] = "ED 448";
  DnssecAlgorithm["ECC"] = "ECC";
})(DnssecAlgorithm || (DnssecAlgorithm = {}));
var _default = DnssecAlgorithm;
exports["default"] = _default;
module.exports = exports.default;
