// @sqs/enums/DomainMoveIneligibilityReason.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;
/* File generated by JavaScriptEnumProcessor */
/* Derived from Java class com.squarespace.business.domains.move.DomainMoveIneligibilityReason */
/* Java class defined in repository squarespace-v6 */
var DomainMoveIneligibilityReason;
(function (DomainMoveIneligibilityReason) {
  DomainMoveIneligibilityReason["NOT_MANAGED"] = "NOT_MANAGED";
  DomainMoveIneligibilityReason["NO_PERMISSIONS_ON_SOURCE_WEBSITE"] = "NO_PERMISSIONS_ON_SOURCE_WEBSITE";
  DomainMoveIneligibilityReason["INVALID_PERMISSIONS"] = "INVALID_PERMISSIONS";
  DomainMoveIneligibilityReason["MID_GOOGLE_DOMAINS_MIGRATION"] = "MID_GOOGLE_DOMAINS_MIGRATION";
  DomainMoveIneligibilityReason["NO_PAYMENT_METHOD_ON_DESTINATION_WEBSITE"] = "NO_PAYMENT_METHOD_ON_DESTINATION_WEBSITE";
})(DomainMoveIneligibilityReason || (DomainMoveIneligibilityReason = {}));
var _default = DomainMoveIneligibilityReason;
exports["default"] = _default;
module.exports = exports.default;
