// @sqs/enums/DomainPlanLevel.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;
/* File generated by JavaScriptEnumProcessor */
/* Derived from Java class com.squarespace.v6.utils.enums.DomainPlanLevel */
/* Java class defined in repository squarespace-v6 */
var DomainPlanLevel;
(function (DomainPlanLevel) {
  DomainPlanLevel["FREE"] = "forever-free";
  DomainPlanLevel["BASIC"] = "basic";
  DomainPlanLevel["BASIC_DISCOUNT"] = "basic_discount";
  DomainPlanLevel["PREMIUM_I"] = "premium_i";
  DomainPlanLevel["PREMIUM_II"] = "premium_ii";
  DomainPlanLevel["PREMIUM_III"] = "premium_iii";
  DomainPlanLevel["PREMIUM_IV"] = "premium_iv";
  DomainPlanLevel["PREMIUM_V"] = "premium_v";
  DomainPlanLevel["PREMIUM_VI"] = "premium_vi";
  DomainPlanLevel["PREMIUM_VII"] = "premium_vii";
  DomainPlanLevel["PREMIUM_VIII"] = "premium_viii";
  DomainPlanLevel["PREMIUM_IX"] = "premium_ix";
  DomainPlanLevel["PREMIUM_X"] = "premium_x";
  DomainPlanLevel["PREMIUM_XI"] = "premium_xi";
  DomainPlanLevel["DOMAIN_COM"] = "domain_com";
  DomainPlanLevel["DOMAIN_NET"] = "domain_net";
  DomainPlanLevel["DOMAIN_ORG"] = "domain_org";
})(DomainPlanLevel || (DomainPlanLevel = {}));
var _default = DomainPlanLevel;
exports["default"] = _default;
module.exports = exports.default;
