// @sqs/enums/DomainTransferRequestError.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;
/* File generated by JavaScriptEnumProcessor */
/* Derived from Java class com.squarespace.v6.model.domains.DomainTransferRequestError */
/* Java class defined in repository squarespace-v6 */
var DomainTransferRequestError;
(function (DomainTransferRequestError) {
  DomainTransferRequestError["UNKNOWN_FAILURE"] = 1;
  DomainTransferRequestError["UNKNOWN_BILLING_FAILURE"] = 2;
  DomainTransferRequestError["BILLING_CARD_DECLINED_FAILURE"] = 3;
  DomainTransferRequestError["CONTACT_FAILURE"] = 4;
  DomainTransferRequestError["INELIGIBLE_TRANSFER_FAILURE"] = 5;
  DomainTransferRequestError["INVALID_NAMESERVER_FAILURE"] = 6;
  DomainTransferRequestError["TRANSFER_EXISTS_FAILURE"] = 7;
  DomainTransferRequestError["TRANSFER_PROVIDER_FAILURE"] = 8;
  DomainTransferRequestError["TRANSFER_DNS_FAILURE"] = 9;
  DomainTransferRequestError["INVALID_AUTH_CODE_FAILURE"] = 10;
  DomainTransferRequestError["MISSING_INFO_FAILURE"] = 11;
  DomainTransferRequestError["LOCKED_DOMAIN_FAILURE"] = 12;
  DomainTransferRequestError["TRANSFER_WITHIN_EXPIRATION_PERIOD_FAILURE"] = 13;
  DomainTransferRequestError["DNS_ZONE_COLLISION_FAILURE"] = 14;
})(DomainTransferRequestError || (DomainTransferRequestError = {}));
var _default = DomainTransferRequestError;
exports["default"] = _default;
module.exports = exports.default;
