// @sqs/enums/EmailCategory.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;
/* File generated by JavaScriptEnumProcessor */
/* Derived from Java class com.squarespace.enums.EmailCategory */
/* Java class defined in repository squarespace-v6 */
var EmailCategory;
(function (EmailCategory) {
  EmailCategory["ACCOUNT"] = "ACCOUNT";
  EmailCategory["BILLING"] = "BILLING";
  EmailCategory["COMMERCE_SHOPPER"] = "COMMERCE_SHOPPER";
  EmailCategory["COMMERCE_MERCHANT"] = "COMMERCE_MERCHANT";
  EmailCategory["DOMAINS"] = "DOMAINS";
  EmailCategory["REGISTRAR"] = "REGISTRAR";
  EmailCategory["FRAUD_DETECTION"] = "FRAUD_DETECTION";
  EmailCategory["GOOGLE_WORKSPACE"] = "GOOGLE_WORKSPACE";
  EmailCategory["G_SUITE"] = "G_SUITE";
  EmailCategory["CAMPAIGNS"] = "CAMPAIGNS";
  EmailCategory["DIGITAL_PRODUCTS"] = "DIGITAL_PRODUCTS";
  EmailCategory["SCHEDULING"] = "SCHEDULING";
  EmailCategory["ONBOARDING"] = "ONBOARDING";
  EmailCategory["SITE_USER"] = "SITE_USER";
  EmailCategory["RETIRED"] = "RETIRED";
  EmailCategory["UNKNOWN"] = "UNKNOWN";
  EmailCategory["CONTENT_MEMBERSHIPS"] = "CONTENT_MEMBERSHIPS";
})(EmailCategory || (EmailCategory = {}));
var _default = EmailCategory;
exports["default"] = _default;
module.exports = exports.default;
