// @sqs/enums/FlattenedShippingOptionType.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;
/* File generated by JavaScriptEnumProcessor */
/* Derived from Java class com.squarespace.api.commerce.discounts.flattened.FlattenedShippingOptionType */
/* Java class defined in repository squarespace-v6 */
var FlattenedShippingOptionType;
(function (FlattenedShippingOptionType) {
  FlattenedShippingOptionType["ALL_OPTIONS"] = "ALL_OPTIONS";
  FlattenedShippingOptionType["NON_CARRIER"] = "NON_CARRIER";
  FlattenedShippingOptionType["ALL_SERVICES"] = "ALL_SERVICES";
  FlattenedShippingOptionType["SERVICE"] = "SERVICE";
  FlattenedShippingOptionType["ALL_SHIPPING_OPTIONS"] = "ALL_SHIPPING_OPTIONS";
  FlattenedShippingOptionType["ALL_LOCAL_DELIVERY_OPTIONS"] = "ALL_LOCAL_DELIVERY_OPTIONS";
  FlattenedShippingOptionType["LOCAL_DELIVERY"] = "LOCAL_DELIVERY";
})(FlattenedShippingOptionType || (FlattenedShippingOptionType = {}));
var _default = FlattenedShippingOptionType;
exports["default"] = _default;
module.exports = exports.default;
