// @sqs/enums/FormPackId.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;
/* File generated by JavaScriptEnumProcessor */
/* Derived from Java class com.squarespace.v6.utils.enums.FormPackId */
/* Java class defined in repository squarespace-v6 */
var FormPackId;
(function (FormPackId) {
  FormPackId["FORM_PACK_01"] = 1;
  FormPackId["FORM_PACK_02"] = 2;
  FormPackId["FORM_PACK_03"] = 3;
  FormPackId["FORM_PACK_04"] = 4;
  FormPackId["FORM_PACK_05"] = 5;
  FormPackId["FORM_PACK_06"] = 6;
  FormPackId["FORM_PACK_07"] = 7;
  FormPackId["FORM_PACK_08"] = 8;
  FormPackId["FORM_PACK_09"] = 9;
  FormPackId["FORM_PACK_10"] = 10;
  FormPackId["FORM_PACK_11"] = 11;
  FormPackId["FORM_PACK_12"] = 12;
  FormPackId["FORM_PACK_13"] = 13;
  FormPackId["FORM_PACK_14"] = 14;
  FormPackId["FORM_PACK_15"] = 15;
  FormPackId["FORM_PACK_16"] = 16;
  FormPackId["FORM_PACK_17"] = 17;
  FormPackId["FORM_PACK_18"] = 18;
  FormPackId["FORM_PACK_19"] = 19;
  FormPackId["FORM_PACK_20"] = 20;
  FormPackId["FORM_PACK_21"] = 21;
  FormPackId["FORM_PACK_22"] = 22;
  FormPackId["FORM_PACK_23"] = 23;
  FormPackId["FORM_PACK_24"] = 24;
  FormPackId["FORM_PACK_25"] = 25;
  FormPackId["FORM_PACK_26"] = 26;
  FormPackId["FORM_PACK_27"] = 27;
  FormPackId["FORM_PACK_28"] = 28;
  FormPackId["FORM_PACK_29"] = 29;
  FormPackId["FORM_PACK_30"] = 30;
  FormPackId["FORM_PACK_31"] = 31;
  FormPackId["FORM_PACK_32"] = 32;
  FormPackId["FORM_PACK_33"] = 33;
  FormPackId["FORM_PACK_34"] = 34;
  FormPackId["FORM_PACK_35"] = 35;
  FormPackId["FORM_PACK_36"] = 36;
  FormPackId["FORM_PACK_37"] = 37;
  FormPackId["FORM_PACK_38"] = 38;
  FormPackId["FORM_PACK_39"] = 39;
  FormPackId["FORM_PACK_40"] = 40;
  FormPackId["FORM_PACK_41"] = 41;
  FormPackId["FORM_PACK_42"] = 42;
})(FormPackId || (FormPackId = {}));
var _default = FormPackId;
exports["default"] = _default;
module.exports = exports.default;
