// @sqs/enums/FormSubmissionMessageTextStyle.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;
/* File generated by JavaScriptEnumProcessor */
/* Derived from Java class com.squarespace.v6.utils.enums.blocks.FormSubmissionMessageTextStyle */
/* Java class defined in repository cms-legacy-data */
var FormSubmissionMessageTextStyle;
(function (FormSubmissionMessageTextStyle) {
  FormSubmissionMessageTextStyle["HEADING_ONE"] = "heading-1";
  FormSubmissionMessageTextStyle["HEADING_TWO"] = "heading-2";
  FormSubmissionMessageTextStyle["HEADING_THREE"] = "heading-3";
  FormSubmissionMessageTextStyle["HEADING_FOUR"] = "heading-4";
  FormSubmissionMessageTextStyle["PARAGRAPH_ONE"] = "paragraph-1";
  FormSubmissionMessageTextStyle["PARAGRAPH_TWO"] = "paragraph-2";
  FormSubmissionMessageTextStyle["PARAGRAPH_THREE"] = "paragraph-3";
})(FormSubmissionMessageTextStyle || (FormSubmissionMessageTextStyle = {}));
var _default = FormSubmissionMessageTextStyle;
exports["default"] = _default;
module.exports = exports.default;
