// @sqs/enums/FraudRule.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;
/* File generated by JavaScriptEnumProcessor */
/* Derived from Java class com.squarespace.v6.model.frauddetection.FraudRule */
/* Java class defined in repository squarespace-v6 */
var FraudRule;
(function (FraudRule) {
  FraudRule["PREVIOUSLY_MATCHED_FRAUDULENT_DOMAIN_NAME"] = "PREVIOUSLY_MATCHED_FRAUDULENT_DOMAIN_NAME";
  FraudRule["PREVIOUSLY_MATCHED_FRAUDULENT_PURCHASER_IP_ADDRESS"] = "PREVIOUSLY_MATCHED_FRAUDULENT_PURCHASER_IP_ADDRESS";
  FraudRule["PREVIOUSLY_MATCHED_FRAUDULENT_DOMAIN_REGISTRANT_EMAIL_ADDRESS"] = "PREVIOUSLY_MATCHED_FRAUDULENT_DOMAIN_REGISTRANT_EMAIL_ADDRESS";
  FraudRule["PREVIOUSLY_MATCHED_FRAUDULENT_DOMAIN_REGISTRANT_ADDRESS"] = "PREVIOUSLY_MATCHED_FRAUDULENT_DOMAIN_REGISTRANT_ADDRESS";
  FraudRule["IN_TOP_5_HIGH_FRAUD_REGION"] = "IN_TOP_5_HIGH_FRAUD_REGION";
  FraudRule["IN_TOP_5_HIGH_FRAUD_COUNTRY"] = "IN_TOP_5_HIGH_FRAUD_COUNTRY";
  FraudRule["MASS_DOMAIN_PURCHASE_IN_SINGLE_BULK_PURCHASE"] = "MASS_DOMAIN_PURCHASE_IN_SINGLE_BULK_PURCHASE";
  FraudRule["PAST_PRICE_THRESHOLD_FOR_SINGLE_BULK_PURCHASE"] = "PAST_PRICE_THRESHOLD_FOR_SINGLE_BULK_PURCHASE";
  FraudRule["MASS_NUMBER_OF_DOMAINS_PURCHASED_IN_PAST_DAY_BY_WEBSITE"] = "MASS_NUMBER_OF_DOMAINS_PURCHASED_IN_PAST_DAY_BY_WEBSITE";
  FraudRule["MASS_NUMBER_OF_DOMAINS_PURCHASED_IN_PAST_DAY_BY_MEMBER_ACCOUNT"] = "MASS_NUMBER_OF_DOMAINS_PURCHASED_IN_PAST_DAY_BY_MEMBER_ACCOUNT";
  FraudRule["PAST_PRICE_THRESHOLD_IN_PAST_DAY_BY_WEBSITE"] = "PAST_PRICE_THRESHOLD_IN_PAST_DAY_BY_WEBSITE";
  FraudRule["PAST_PRICE_THRESHOLD_IN_PAST_DAY_BY_MEMBER_ACCOUNT"] = "PAST_PRICE_THRESHOLD_IN_PAST_DAY_BY_MEMBER_ACCOUNT";
  FraudRule["DOMAIN_NAMES_HAVE_SIMPLE_NUMBER_SUFFIXES"] = "DOMAIN_NAMES_HAVE_SIMPLE_NUMBER_SUFFIXES";
  FraudRule["DOMAIN_NAMES_HAVE_NO_DICTIONARY_WORDS"] = "DOMAIN_NAMES_HAVE_NO_DICTIONARY_WORDS";
})(FraudRule || (FraudRule = {}));
var _default = FraudRule;
exports["default"] = _default;
module.exports = exports.default;
