// @sqs/enums/GoogleAppsSubscriptionRequestStatus.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;
/* File generated by JavaScriptEnumProcessor */
/* Derived from Java class com.squarespace.v6.model.googleapps.GoogleAppsSubscriptionRequestStatus */
/* Java class defined in repository squarespace-v6 */
var GoogleAppsSubscriptionRequestStatus;
(function (GoogleAppsSubscriptionRequestStatus) {
  GoogleAppsSubscriptionRequestStatus["UNINITIATED"] = "UNINITIATED";
  GoogleAppsSubscriptionRequestStatus["PENDING"] = "PENDING";
  GoogleAppsSubscriptionRequestStatus["NOT_STARTED"] = "NOT_STARTED";
  GoogleAppsSubscriptionRequestStatus["IN_PROGRESS"] = "IN_PROGRESS";
  GoogleAppsSubscriptionRequestStatus["ILLEGAL_STATE"] = "ILLEGAL_STATE";
  GoogleAppsSubscriptionRequestStatus["SUCCESS"] = "SUCCESS";
  GoogleAppsSubscriptionRequestStatus["FAILED"] = "FAILED";
  GoogleAppsSubscriptionRequestStatus["DISMISSED"] = "DISMISSED";
  GoogleAppsSubscriptionRequestStatus["ASYNC_PAYMENT_PROCESSING"] = "ASYNC_PAYMENT_PROCESSING";
  GoogleAppsSubscriptionRequestStatus["ASYNC_PAYMENT_SUCCEEDED"] = "ASYNC_PAYMENT_SUCCEEDED";
  GoogleAppsSubscriptionRequestStatus["ASYNC_PAYMENT_FAILED"] = "ASYNC_PAYMENT_FAILED";
  GoogleAppsSubscriptionRequestStatus["BILLING_REQUEST"] = "BILLING_REQUEST";
})(GoogleAppsSubscriptionRequestStatus || (GoogleAppsSubscriptionRequestStatus = {}));
var _default = GoogleAppsSubscriptionRequestStatus;
exports["default"] = _default;
module.exports = exports.default;
