// @sqs/enums/GSuiteSubscriptionProcessingFailureReasons.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;
/* File generated by JavaScriptEnumProcessor */
/* Derived from Java class com.squarespace.v6.model.googleapps.GoogleAppsSignupFailureReason */
/* Java class defined in repository squarespace-v6 */
var GSuiteSubscriptionProcessingFailureReasons;
(function (GSuiteSubscriptionProcessingFailureReasons) {
  GSuiteSubscriptionProcessingFailureReasons["RECORD_IS_NOT_MEANT_FOR_ASYNC_FLOW"] = "RECORD_IS_NOT_MEANT_FOR_ASYNC_FLOW";
  GSuiteSubscriptionProcessingFailureReasons["UNKNOWN_FAILURE"] = "UNKNOWN_FAILURE";
  GSuiteSubscriptionProcessingFailureReasons["WEBSITE_NOT_FOUND"] = "WEBSITE_NOT_FOUND";
  GSuiteSubscriptionProcessingFailureReasons["BILLING_MIGRATION_IN_PROGRESS"] = "BILLING_MIGRATION_IN_PROGRESS";
  GSuiteSubscriptionProcessingFailureReasons["CANNOT_BE_RELINKED"] = "CANNOT_BE_RELINKED";
  GSuiteSubscriptionProcessingFailureReasons["DOMAIN_NOT_OWNED_BY_WEBSITE"] = "DOMAIN_NOT_OWNED_BY_WEBSITE";
  GSuiteSubscriptionProcessingFailureReasons["DOMAIN_IS_MISCONFIGURED"] = "DOMAIN_IS_MISCONFIGURED";
  GSuiteSubscriptionProcessingFailureReasons["USER_EMAIL_IS_INVALID"] = "USER_EMAIL_IS_INVALID";
  GSuiteSubscriptionProcessingFailureReasons["PLAN_IS_INVALID"] = "PLAN_IS_INVALID";
  GSuiteSubscriptionProcessingFailureReasons["INVALID_POSTAL_CODE"] = "INVALID_POSTAL_CODE";
  GSuiteSubscriptionProcessingFailureReasons["DUPLICATE_USER_DETECTED"] = "DUPLICATE_USER_DETECTED";
  GSuiteSubscriptionProcessingFailureReasons["EMAIL_ADDRESS_IS_ON_WRONG_DOMAIN"] = "EMAIL_ADDRESS_IS_ON_WRONG_DOMAIN";
  GSuiteSubscriptionProcessingFailureReasons["EMAIL_ADDRESS_IS_ALREADY_USED_BY_GOOGLE_GROUP_IN_GSUITE_ACCOUNT"] = "EMAIL_ADDRESS_IS_ALREADY_USED_BY_GOOGLE_GROUP_IN_GSUITE_ACCOUNT";
  GSuiteSubscriptionProcessingFailureReasons["UNKNOWN_MX_RECORD_SETUP_FAILURE"] = "UNKNOWN_MX_RECORD_SETUP_FAILURE";
  GSuiteSubscriptionProcessingFailureReasons["GSUITE_ADMIN_USERNAME_ALREADY_IN_USE"] = "GSUITE_ADMIN_USERNAME_ALREADY_IN_USE";
  GSuiteSubscriptionProcessingFailureReasons["SUBSCRIPTION_EXISTS_ON_DOMAIN"] = "SUBSCRIPTION_EXISTS_ON_DOMAIN";
  GSuiteSubscriptionProcessingFailureReasons["UNKNOWN_DOMAIN_OWNERSHIP_RESOLVING_ERROR"] = "UNKNOWN_DOMAIN_OWNERSHIP_RESOLVING_ERROR";
  GSuiteSubscriptionProcessingFailureReasons["CREDIT_CARD_DECLINED"] = "CREDIT_CARD_DECLINED";
  GSuiteSubscriptionProcessingFailureReasons["BILLING_CHARGE_FAILED"] = "BILLING_CHARGE_FAILED";
  GSuiteSubscriptionProcessingFailureReasons["GOOGLE_SUBSCRIPTION_CREATION_FULFILLMENT_FAILED"] = "GOOGLE_SUBSCRIPTION_CREATION_FULFILLMENT_FAILED";
  GSuiteSubscriptionProcessingFailureReasons["GOOGLE_SUBSCRIPTION_CREATION_INTERRUPTED"] = "GOOGLE_SUBSCRIPTION_CREATION_INTERRUPTED";
  GSuiteSubscriptionProcessingFailureReasons["AUTO_RENEW_TOGGLE_FAILED"] = "AUTO_RENEW_TOGGLE_FAILED";
  GSuiteSubscriptionProcessingFailureReasons["MANUALLY_FAILED_BY_ADMIN"] = "MANUALLY_FAILED_BY_ADMIN";
  GSuiteSubscriptionProcessingFailureReasons["ASYNC_PAYMENT_FAILED"] = "ASYNC_PAYMENT_FAILED";
  GSuiteSubscriptionProcessingFailureReasons["ASYNC_PAYMENT_OVERDUE"] = "ASYNC_PAYMENT_OVERDUE";
})(GSuiteSubscriptionProcessingFailureReasons || (GSuiteSubscriptionProcessingFailureReasons = {}));
var _default = GSuiteSubscriptionProcessingFailureReasons;
exports["default"] = _default;
module.exports = exports.default;
