// @sqs/enums/GSuiteSubscriptionStep.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;
/* File generated by JavaScriptEnumProcessor */
/* Derived from Java class com.squarespace.v6.model.googleapps.GSuiteSubscriptionStep */
/* Java class defined in repository squarespace-v6 */
var GSuiteSubscriptionStep;
(function (GSuiteSubscriptionStep) {
  GSuiteSubscriptionStep["VALIDATE_REQUEST"] = "VALIDATE_REQUEST";
  GSuiteSubscriptionStep["RESOLVE_DOMAIN_OWNERSHIP"] = "RESOLVE_DOMAIN_OWNERSHIP";
  GSuiteSubscriptionStep["CREATE_GOOGLE_CUSTOMER"] = "CREATE_GOOGLE_CUSTOMER";
  GSuiteSubscriptionStep["CREATE_GOOGLE_ADMIN"] = "CREATE_GOOGLE_ADMIN";
  GSuiteSubscriptionStep["CREATE_MX_RECORDS"] = "CREATE_MX_RECORDS";
  GSuiteSubscriptionStep["CREATE_SUBSCRIPTIONS"] = "CREATE_SUBSCRIPTIONS";
  GSuiteSubscriptionStep["MARK_AS_FAILURE"] = "MARK_AS_FAILURE";
  GSuiteSubscriptionStep["MARK_AS_SUCCESS"] = "MARK_AS_SUCCESS";
  GSuiteSubscriptionStep["RELINK_SUBSCRIPTIONS"] = "RELINK_SUBSCRIPTIONS";
})(GSuiteSubscriptionStep || (GSuiteSubscriptionStep = {}));
var _default = GSuiteSubscriptionStep;
exports["default"] = _default;
module.exports = exports.default;
