// @sqs/enums/MemberAccountRole.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;
/* File generated by JavaScriptEnumProcessor */
/* Derived from Java class com.squarespace.v6.model.roles.MemberAccountRole */
/* Java class defined in repository squarespace-v6 */
var MemberAccountRole;
(function (MemberAccountRole) {
  MemberAccountRole["SUPERADMIN"] = 1;
  MemberAccountRole["BILLING"] = 2;
  MemberAccountRole["SUPPORT"] = 3;
  MemberAccountRole["DEV"] = 4;
  MemberAccountRole["MARKETING"] = 5;
  MemberAccountRole["SUPPORT_SUPERVISOR"] = 6;
  MemberAccountRole["INTERNAL_REPORTING"] = 7;
  MemberAccountRole["EXTERNAL_BETA_ACCESS"] = 8;
  MemberAccountRole["ADVISOR_BILLING"] = 9;
  MemberAccountRole["TRUST_AND_SAFETY"] = 10;
  MemberAccountRole["SQSP_PUBLIC_IMPERSONATION"] = 11;
  MemberAccountRole["CAMPAIGNS_EMAIL_REVIEWER"] = 12;
  MemberAccountRole["ENTERPRISE_ADMIN"] = 13;
  MemberAccountRole["FRAUD_DETECTION_REVIEWER"] = 14;
  MemberAccountRole["INTERNAL_CAMPAIGNS_SENDER"] = 15;
  MemberAccountRole["SITEL_USER"] = 16;
  MemberAccountRole["LEGAL"] = 17;
  MemberAccountRole["PRODUCT_SUPPORT"] = 18;
  MemberAccountRole["EXTENSIONS_MANAGER"] = 19;
  MemberAccountRole["ROLE_GRANTOR"] = 20;
  MemberAccountRole["CUSTOPS_BPO_INTOUCHCX"] = 21;
  MemberAccountRole["IMPERSONATOR"] = 22;
  MemberAccountRole["SQSP_PAYMENTS_RISK_OPS"] = 23;
  MemberAccountRole["ADMIN_APP_SITE_SEARCH"] = 24;
  MemberAccountRole["ADMIN_APP_MARKETING_COUPONS"] = 25;
  MemberAccountRole["ADMIN_APP_ADMIN_ROLES"] = 26;
  MemberAccountRole["ADMIN_APP_ENTERPRISE_ORGANIZATIONS"] = 27;
  MemberAccountRole["ADMIN_APP_BILLING_ACCOUNTS"] = 28;
  MemberAccountRole["ADMIN_APP_LOOKUP_LOGO"] = 29;
  MemberAccountRole["ADMIN_APP_EMAIL_TROUBLESHOOTING"] = 30;
  MemberAccountRole["ADMIN_APP_EMAIL_TEMPLATE_PREVIEW"] = 31;
  MemberAccountRole["ADMIN_APP_SUBSCRIPTION_EVENTS"] = 32;
  MemberAccountRole["ADMIN_APP_MASS_REMOVAL"] = 33;
  MemberAccountRole["ADMIN_APP_CAMPAIGN_SUPPORT"] = 34;
  MemberAccountRole["ADMIN_APP_BLOCK_REDUCE"] = 35;
  MemberAccountRole["ADMIN_APP_COPY_COLLECTION"] = 36;
  MemberAccountRole["ADMIN_APP_CATEGORY_CLEANUP"] = 37;
  MemberAccountRole["ADMIN_APP_FRAUD_TRANSACTIONS"] = 38;
  MemberAccountRole["ADMIN_APP_FULLSTORY_LOOKUP"] = 39;
  MemberAccountRole["ADMIN_APP_ACCOUNT_SEARCH"] = 40;
  MemberAccountRole["ADMIN_APP_ACCOUNT_HISTORY"] = 41;
})(MemberAccountRole || (MemberAccountRole = {}));
var _default = MemberAccountRole;
exports["default"] = _default;
module.exports = exports.default;
