// @sqs/enums/OrderOperationErrorType.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;
/* File generated by JavaScriptEnumProcessor */
/* Derived from Java class com.squarespace.api.content.typesafe.models.v1.OrderOperationErrorType */
/* Java class defined in repository squarespace-v6 */
var OrderOperationErrorType;
(function (OrderOperationErrorType) {
  OrderOperationErrorType["CONCURRENT_MODIFICATION"] = "CONCURRENT_MODIFICATION";
  OrderOperationErrorType["REQUEST_TIMED_OUT"] = "REQUEST_TIMED_OUT";
  OrderOperationErrorType["HALTED_WORKFLOW"] = "HALTED_WORKFLOW";
  OrderOperationErrorType["INSUFFICIENT_FUNDS"] = "INSUFFICIENT_FUNDS";
  OrderOperationErrorType["TRANSACTION_REFUSED"] = "TRANSACTION_REFUSED";
  OrderOperationErrorType["MERCHANT_CONFIGURATION_ERROR"] = "MERCHANT_CONFIGURATION_ERROR";
  OrderOperationErrorType["UNKNOWN"] = "UNKNOWN";
})(OrderOperationErrorType || (OrderOperationErrorType = {}));
var _default = OrderOperationErrorType;
exports["default"] = _default;
module.exports = exports.default;
