// @sqs/enums/PageCategories.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;
/* File generated by JavaScriptEnumProcessor */
/* Derived from Java class com.squarespace.v6.utils.enums.PageCategories */
/* Java class defined in repository squarespace-v6 */
var PageCategories;
(function (PageCategories) {
  PageCategories["ABOUT"] = "ABOUT";
  PageCategories["APPOINTMENTS"] = "APPOINTMENTS";
  PageCategories["BLOG"] = "BLOG";
  PageCategories["CONTACT"] = "CONTACT";
  PageCategories["DESIGNER_LAYOUTS"] = "DESIGNER_LAYOUTS";
  PageCategories["EVENTS"] = "EVENTS";
  PageCategories["FAQS"] = "FAQS";
  PageCategories["GALLERY"] = "GALLERY";
  PageCategories["GENERAL"] = "GENERAL";
  PageCategories["MEMBER_AREAS"] = "MEMBER_AREAS";
  PageCategories["MENU"] = "MENU";
  PageCategories["MUSIC"] = "MUSIC";
  PageCategories["PORTFOLIO"] = "PORTFOLIO";
  PageCategories["PRIVACY_POLICY"] = "PRIVACY_POLICY";
  PageCategories["PROJECT"] = "PROJECT";
  PageCategories["REVIEWS"] = "REVIEWS";
  PageCategories["RSVP"] = "RSVP";
  PageCategories["SERVICES"] = "SERVICES";
  PageCategories["STAGING"] = "STAGING";
  PageCategories["STORE"] = "STORE";
  PageCategories["TEAM"] = "TEAM";
  PageCategories["TOS"] = "TOS";
  PageCategories["LESSONS"] = "LESSONS";
  PageCategories["COURSE"] = "COURSE";
  PageCategories["DIGITAL_PRODUCTS"] = "DIGITAL_PRODUCTS";
  PageCategories["MEMBER_SITE_HOMEPAGE"] = "MEMBER_SITE_HOMEPAGE";
  PageCategories["GIVING_PAGES"] = "GIVING_PAGES";
})(PageCategories || (PageCategories = {}));
var _default = PageCategories;
exports["default"] = _default;
module.exports = exports.default;
