// @sqs/enums/ParcelTemplateType.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;
/* File generated by JavaScriptEnumProcessor */
/* Derived from Java class com.squarespace.business.commerce.shipping.labels.model.ParcelTemplateType */
/* Java class defined in repository squarespace-v6 */
var ParcelTemplateType;
(function (ParcelTemplateType) {
  ParcelTemplateType["UPS_BOX_10KG"] = "UPS_BOX_10KG";
  ParcelTemplateType["UPS_BOX_25KG"] = "UPS_BOX_25KG";
  ParcelTemplateType["UPS_EXPRESS_BOX"] = "UPS_EXPRESS_BOX";
  ParcelTemplateType["UPS_EXPRESS_BOX_LARGE"] = "UPS_EXPRESS_BOX_LARGE";
  ParcelTemplateType["UPS_EXPRESS_BOX_MEDIUM"] = "UPS_EXPRESS_BOX_MEDIUM";
  ParcelTemplateType["UPS_EXPRESS_BOX_SMALL"] = "UPS_EXPRESS_BOX_SMALL";
  ParcelTemplateType["UPS_EXPRESS_ENVELOPE"] = "UPS_EXPRESS_ENVELOPE";
  ParcelTemplateType["UPS_EXPRESS_HARD_PAK"] = "UPS_EXPRESS_HARD_PAK";
  ParcelTemplateType["UPS_EXPRESS_LEGAL_ENVELOPE"] = "UPS_EXPRESS_LEGAL_ENVELOPE";
  ParcelTemplateType["UPS_EXPRESS_PAK"] = "UPS_EXPRESS_PAK";
  ParcelTemplateType["UPS_EXPRESS_TUBE"] = "UPS_EXPRESS_TUBE";
  ParcelTemplateType["UPS_LABORATORY_PAK"] = "UPS_LABORATORY_PAK";
  ParcelTemplateType["UPS_MI_BPM"] = "UPS_MI_BPM";
  ParcelTemplateType["UPS_MI_BPM_FLAT"] = "UPS_MI_BPM_FLAT";
  ParcelTemplateType["UPS_MI_BPM_PARCEL"] = "UPS_MI_BPM_PARCEL";
  ParcelTemplateType["UPS_MI_FIRST_CLASS"] = "UPS_MI_FIRST_CLASS";
  ParcelTemplateType["UPS_MI_FLAT"] = "UPS_MI_FLAT";
  ParcelTemplateType["UPS_MI_IRREGULAR"] = "UPS_MI_IRREGULAR";
  ParcelTemplateType["UPS_MI_MACHINABLE"] = "UPS_MI_MACHINABLE";
  ParcelTemplateType["UPS_MI_MEDIA_MAIL"] = "UPS_MI_MEDIA_MAIL";
  ParcelTemplateType["UPS_MI_PARCEL_POST"] = "UPS_MI_PARCEL_POST";
  ParcelTemplateType["UPS_MI_PRIORITY"] = "UPS_MI_PRIORITY";
  ParcelTemplateType["UPS_MI_STANDARD_FLAT"] = "UPS_MI_STANDARD_FLAT";
  ParcelTemplateType["UPS_PAD_PAK"] = "UPS_PAD_PAK";
  ParcelTemplateType["UPS_PALLET"] = "UPS_PALLET";
  ParcelTemplateType["USPS_FLATRATEENVELOPE"] = "USPS_FLATRATEENVELOPE";
  ParcelTemplateType["USPS_FLATRATELEGALENVELOPE"] = "USPS_FLATRATELEGALENVELOPE";
  ParcelTemplateType["USPS_FLATRATEPADDEDENVELOPE"] = "USPS_FLATRATEPADDEDENVELOPE";
  ParcelTemplateType["USPS_FLATRATEWINDOWENVELOPE"] = "USPS_FLATRATEWINDOWENVELOPE";
  ParcelTemplateType["USPS_IRREGULARPARCEL"] = "USPS_IRREGULARPARCEL";
  ParcelTemplateType["USPS_LARGEFLATRATEBOARDGAMEBOX"] = "USPS_LARGEFLATRATEBOARDGAMEBOX";
  ParcelTemplateType["USPS_LARGEFLATRATEBOX"] = "USPS_LARGEFLATRATEBOX";
  ParcelTemplateType["USPS_APOFLATRATEBOX"] = "USPS_APOFLATRATEBOX";
  ParcelTemplateType["USPS_LARGEVIDEOFLATRATEBOX"] = "USPS_LARGEVIDEOFLATRATEBOX";
  ParcelTemplateType["USPS_MEDIUMFLATRATEBOX1"] = "USPS_MEDIUMFLATRATEBOX1";
  ParcelTemplateType["USPS_MEDIUMFLATRATEBOX2"] = "USPS_MEDIUMFLATRATEBOX2";
  ParcelTemplateType["USPS_REGIONALRATEBOXA1"] = "USPS_REGIONALRATEBOXA1";
  ParcelTemplateType["USPS_REGIONALRATEBOXA2"] = "USPS_REGIONALRATEBOXA2";
  ParcelTemplateType["USPS_REGIONALRATEBOXB1"] = "USPS_REGIONALRATEBOXB1";
  ParcelTemplateType["USPS_REGIONALRATEBOXB2"] = "USPS_REGIONALRATEBOXB2";
  ParcelTemplateType["USPS_SMALLFLATRATEBOX"] = "USPS_SMALLFLATRATEBOX";
  ParcelTemplateType["USPS_SMALLFLATRATEENVELOPE"] = "USPS_SMALLFLATRATEENVELOPE";
})(ParcelTemplateType || (ParcelTemplateType = {}));
var _default = ParcelTemplateType;
exports["default"] = _default;
module.exports = exports.default;
