// @sqs/enums/PaymentError.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;
/* File generated by JavaScriptEnumProcessor */
/* Derived from Java class com.squarespace.enums.PaymentError */
/* Java class defined in repository squarespace-v6 */
var PaymentError;
(function (PaymentError) {
  PaymentError["CARD_DECLINED"] = "card_declined";
  PaymentError["EXPIRED_CARD"] = "expired_card";
  PaymentError["INCORRECT_POSTAL"] = "incorrect_postal";
  PaymentError["INCORRECT_CVV"] = "incorrect_cvc";
  PaymentError["INCORRECT_NUMBER"] = "incorrect_number";
  PaymentError["CARD_MISSING"] = "card_missing";
  PaymentError["PROCESSING_ERROR"] = "processing_error";
  PaymentError["STRONG_AUTHENTICATION_ERROR"] = "strong_authentication_error";
  PaymentError["SUSPECTED_FRAUD"] = "suspected_fraud";
  PaymentError["EXCEEDED_MAXIMUM_DECLINES"] = "exceeded_maximum_declines";
  PaymentError["INVALID_OWNER_NAME"] = "invalid_owner_name";
  PaymentError["INVALID_BANK_ACCOUNT_IBAN"] = "invalid_bank_account_iban";
  PaymentError["BANK_ACCOUNT_DECLINED"] = "bank_account_declined";
  PaymentError["PAYMENT_METHOD_NOT_AVAILABLE"] = "payment_method_not_available";
  PaymentError["PAYMENT_METHOD_MANDATE_NOT_FOUND"] = "payment_method_mandate_not_found";
  PaymentError["INTENT_ATTEMPT_EXPIRED"] = "intent_attempt_expired";
  PaymentError["CUSTOMER_DECLINED"] = "customer_declined";
  PaymentError["PROVIDER_TIMEOUT"] = "provider_timeout";
  PaymentError["UNKNOWN"] = "unknown";
})(PaymentError || (PaymentError = {}));
var _default = PaymentError;
exports["default"] = _default;
module.exports = exports.default;
