// @sqs/enums/PaymentStatus.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;
/* File generated by JavaScriptEnumProcessor */
/* Derived from Java class com.squarespace.api.admin.billing.payments.PaymentStatus */
/* Java class defined in repository squarespace-v6 */
var PaymentStatus;
(function (PaymentStatus) {
  PaymentStatus["PENDING"] = "pending";
  PaymentStatus["PROCESSING"] = "processing";
  PaymentStatus["REQUIRES_ACTION"] = "requires_action";
  PaymentStatus["SUCCEEDED"] = "succeeded";
  PaymentStatus["FAILED"] = "failed";
  PaymentStatus["CANCELED"] = "canceled";
  PaymentStatus["DISPUTE_OPEN"] = "dispute_open";
  PaymentStatus["DISPUTE_OPEN_INQUIRY"] = "dispute_open_inquiry";
  PaymentStatus["DISPUTE_CLOSED_INQUIRY"] = "dispute_closed_inquiry";
  PaymentStatus["DISPUTE_WON"] = "dispute_won";
  PaymentStatus["DISPUTE_LOST"] = "dispute_lost";
})(PaymentStatus || (PaymentStatus = {}));
var _default = PaymentStatus;
exports["default"] = _default;
module.exports = exports.default;
