// @sqs/enums/PlanLine.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;
/* File generated by JavaScriptEnumProcessor */
/* Derived from Java class com.squarespace.api.billing.enums.PlanLine */
/* Java class defined in repository squarespace-v6 */
var PlanLine;
(function (PlanLine) {
  PlanLine["SQUARESPACE_PRESENCE"] = "squarespace-presence";
  PlanLine["SQUARESPACE_COMMERCE"] = "squarespace-commerce";
  PlanLine["DOMAIN"] = "domain";
  PlanLine["RESOLD_DOMAIN"] = "resold-domain";
  PlanLine["SCHEDULING"] = "scheduling";
  PlanLine["WEBSITE_CLASSIC"] = "website-classic";
  PlanLine["WEBSITE_COMMERCE"] = "website-commerce";
  PlanLine["WEBSITE_ENTERPRISE"] = "website-enterprise";
  PlanLine["WEBSITE_PARKING_PAGE"] = "website-parking-page";
  PlanLine["WEBSITE_COVER_PAGE"] = "website-cover-page";
  PlanLine["MEMBER_AREAS"] = "member-areas";
  PlanLine["CAMPAIGNS"] = "campaigns";
  PlanLine["GOOGLE_WORKSPACE"] = "google-apps";
  PlanLine["SHIPPING_LABEL"] = "shipping-label";
  PlanLine["GETTY_IMAGE"] = "getty-image";
})(PlanLine || (PlanLine = {}));
var _default = PlanLine;
exports["default"] = _default;
module.exports = exports.default;
