// @sqs/enums/ProductApiErrorType.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;
/* File generated by JavaScriptEnumProcessor */
/* Derived from Java class com.squarespace.api.content.typesafe.models.v1.ProductApiErrorType */
/* Java class defined in repository squarespace-v6 */
var ProductApiErrorType;
(function (ProductApiErrorType) {
  ProductApiErrorType["SKU_NOT_FOUND"] = "SKU_NOT_FOUND";
  ProductApiErrorType["PRODUCT_NOT_FOUND"] = "PRODUCT_NOT_FOUND";
  ProductApiErrorType["MISSING_PRODUCT_TYPE"] = "MISSING_PRODUCT_TYPE";
  ProductApiErrorType["ILLEGAL_DATA_FOR_PRODUCT_CREATION"] = "ILLEGAL_DATA_FOR_PRODUCT_CREATION";
  ProductApiErrorType["INVALID_DATA_FOR_PRODUCT_TYPE"] = "INVALID_DATA_FOR_PRODUCT_TYPE";
  ProductApiErrorType["NEGATIVE_STOCK_LEVEL"] = "NEGATIVE_STOCK_LEVEL";
  ProductApiErrorType["PAGE_SIZE_TOO_BIG"] = "PAGE_SIZE_TOO_BIG";
  ProductApiErrorType["INVALID_SKU_FOR_PRODUCT_TYPE"] = "INVALID_SKU_FOR_PRODUCT_TYPE";
  ProductApiErrorType["SUBSCRIPTION_PLAN_NOT_FOUND"] = "SUBSCRIPTION_PLAN_NOT_FOUND";
  ProductApiErrorType["VARIANT_NOT_FOUND"] = "VARIANT_NOT_FOUND";
  ProductApiErrorType["INVALID_URL_ID"] = "INVALID_URL_ID";
  ProductApiErrorType["MALFORMED_URL_ID"] = "MALFORMED_URL_ID";
  ProductApiErrorType["IMAGE_UPLOAD_SIZE_TOO_LARGE"] = "IMAGE_UPLOAD_SIZE_TOO_LARGE";
  ProductApiErrorType["UNKNOWN_ERROR"] = "UNKNOWN_ERROR";
  ProductApiErrorType["TAX_CATEGORY_ERROR"] = "TAX_CATEGORY_ERROR";
  ProductApiErrorType["VARIANT_ID_NOT_FOUND"] = "VARIANT_ID_NOT_FOUND";
})(ProductApiErrorType || (ProductApiErrorType = {}));
var _default = ProductApiErrorType;
exports["default"] = _default;
module.exports = exports.default;
