// @sqs/enums/PromotionType.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;
/* File generated by JavaScriptEnumProcessor */
/* Derived from Java class com.squarespace.v6.utils.enums.PromotionType */
/* Java class defined in repository squarespace-v6 */
var PromotionType;
(function (PromotionType) {
  PromotionType["CIRCLE"] = "CIRCLE";
  PromotionType["EDUCATION"] = "EDUCATION";
  PromotionType["FREE_ANNUAL_DOMAIN"] = "FREE_ANNUAL_DOMAIN";
  PromotionType["FREE_ANNUAL_GOOGLE_APPS"] = "FREE_ANNUAL_GOOGLE_APPS";
  PromotionType["FREE_MONTHLY_GOOGLE_APPS"] = "FREE_MONTHLY_GOOGLE_APPS";
  PromotionType["FRIENDS_AND_FAMILY"] = "FRIENDS_AND_FAMILY";
  PromotionType["MARKETING"] = "MARKETING";
  PromotionType["REFERRAL_CODE"] = "REFERRAL_CODE";
  PromotionType["REFERRAL_DISCOUNT"] = "REFERRAL_DISCOUNT";
  PromotionType["DISCOUNT_10_PERCENT"] = "DISCOUNT_10_PERCENT";
  PromotionType["DISCOUNT_20_PERCENT"] = "DISCOUNT_20_PERCENT";
  PromotionType["DISCOUNT_30_PERCENT"] = "DISCOUNT_30_PERCENT";
  PromotionType["FIRST_YEAR_DISCOUNT"] = "FIRST_YEAR_DISCOUNT";
  PromotionType["CYBER_MONDAY_NET_DOMAIN_DISCOUNT"] = "CYBER_MONDAY_NET_DOMAIN_DISCOUNT";
})(PromotionType || (PromotionType = {}));
var _default = PromotionType;
exports["default"] = _default;
module.exports = exports.default;
