// @sqs/enums/RegistryAccount.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;
/* File generated by JavaScriptEnumProcessor */
/* Derived from Java class com.squarespace.domains.commons.models.RegistryAccount */
/* Java class defined in repository domains-commons */
var RegistryAccount;
(function (RegistryAccount) {
  RegistryAccount["SDL_CENTRALNIC"] = "SDL_CENTRALNIC";
  RegistryAccount["SDL_GOOGLE"] = "SDL_GOOGLE";
  RegistryAccount["SDL_IDENTITY_DIGITAL"] = "SDL_IDENTITY_DIGITAL";
  RegistryAccount["SDL_NOMINET"] = "SDL_NOMINET";
  RegistryAccount["SDL_PIR"] = "SDL_PIR";
  RegistryAccount["SDL_VERISIGN"] = "SDL_VERISIGN";
  RegistryAccount["SDL_TUCOWS_REGISTRY_MUSIC"] = "SDL_TUCOWS_REGISTRY_MUSIC";
  RegistryAccount["SDL2_CENTRALNIC"] = "SDL2_CENTRALNIC";
  RegistryAccount["SDL2_FURY"] = "SDL2_FURY";
  RegistryAccount["SDL2_GOOGLE"] = "SDL2_GOOGLE";
  RegistryAccount["SDL2_GODADDY"] = "SDL2_GODADDY";
  RegistryAccount["SDL2_IDENTITY_DIGITAL"] = "SDL2_IDENTITY_DIGITAL";
  RegistryAccount["SDL2_PIR"] = "SDL2_PIR";
  RegistryAccount["SDL2_TUCOWS_REGISTRY_BAR"] = "SDL2_TUCOWS_REGISTRY_BAR";
  RegistryAccount["SDL2_TUCOWS_REGISTRY_CLICK"] = "SDL2_TUCOWS_REGISTRY_CLICK";
  RegistryAccount["SDL2_TUCOWS_REGISTRY_CLOUD"] = "SDL2_TUCOWS_REGISTRY_CLOUD";
  RegistryAccount["SDL2_TUCOWS_REGISTRY_LOVE"] = "SDL2_TUCOWS_REGISTRY_LOVE";
  RegistryAccount["SDL2_VERISIGN"] = "SDL2_VERISIGN";
})(RegistryAccount || (RegistryAccount = {}));
var _default = RegistryAccount;
exports["default"] = _default;
module.exports = exports.default;
