// @sqs/enums/RegistrationFailureReason.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;
/* File generated by JavaScriptEnumProcessor */
/* Derived from Java class com.squarespace.v6.utils.enums.RegistrationFailureReason */
/* Java class defined in repository squarespace-v6 */
var RegistrationFailureReason;
(function (RegistrationFailureReason) {
  RegistrationFailureReason["UNKNOWN_FAILURE"] = 1;
  RegistrationFailureReason["BILLING_CARD_DECLINED_FAILURE"] = 2;
  RegistrationFailureReason["CONTACT_FAILURE"] = 3;
  RegistrationFailureReason["DOMAIN_UNAVAILABLE_FAILURE"] = 4;
  RegistrationFailureReason["DOMAIN_EXISTS_FAILURE"] = 5;
  RegistrationFailureReason["INVALID_NAME_FAILURE"] = 6;
  RegistrationFailureReason["SUSPICIOUS_DOMAIN_BLOCKED"] = 7;
  RegistrationFailureReason["UNSUPPORTED_TLD"] = 8;
  RegistrationFailureReason["INCORRECT_PLAN"] = 9;
  RegistrationFailureReason["INVALID_USER_CONTACT_INFORMATION"] = 10;
  RegistrationFailureReason["INVALID_USER_POSTAL_CODE"] = 11;
  RegistrationFailureReason["INVALID_USER_COUNTRY"] = 12;
  RegistrationFailureReason["INVALID_USER_STATE"] = 13;
  RegistrationFailureReason["INVALID_USER_PHONE_NUMBER"] = 14;
  RegistrationFailureReason["INVALID_USER_EMAIL"] = 15;
  RegistrationFailureReason["UNSUPPORTED_IDN_LANGUAGE_TAG"] = 16;
  RegistrationFailureReason["REGISTRAR_IS_DOWN"] = 17;
  RegistrationFailureReason["MANUALLY_MARKED_AS_FAILURE"] = 18;
  RegistrationFailureReason["PAYMENT_AUTHENTICATION_FAILURE"] = 19;
  RegistrationFailureReason["OVERDUE_AUTHENTICATION_FAILURE"] = 20;
  RegistrationFailureReason["COM_AU_ENTITY_NAME_MISMATCH"] = 21;
  RegistrationFailureReason["COM_AU_NO_RECORDS_FOUND_FOR_ELIGIBILITY_ID"] = 22;
  RegistrationFailureReason["COM_AU_ELIGIBILITY_ID_TYPE_MISMATCH_EXPECTED_ABN"] = 23;
  RegistrationFailureReason["COM_AU_ELIGIBILITY_ID_TYPE_MISMATCH_EXPECTED_ACN"] = 24;
  RegistrationFailureReason["COM_AU_ABN_NOT_ACTIVE"] = 25;
  RegistrationFailureReason["PROCESSING_PAYMENT_FAILURE"] = 26;
  RegistrationFailureReason["PROCESSING_PAYMENT_OVERDUE"] = 27;
  RegistrationFailureReason["DNS_ZONE_COLLISION_FAILURE"] = 28;
  RegistrationFailureReason["DNS_ZONE_CREATION_FAILURE"] = 29;
  RegistrationFailureReason["PROCESSING_PAYMENT_GENERAL_CANCEL"] = 30;
  RegistrationFailureReason["AU_INCORRECT_PRIORITY_INFO_FAILURE"] = 31;
  RegistrationFailureReason["DOMAIN_BATCH_REGISTRATION_REQUEST_NOT_FOUND"] = 32;
  RegistrationFailureReason["INVALID_TERM"] = 33;
  RegistrationFailureReason["DOMAIN_ALREADY_PENDING_REGISTRATION"] = 34;
})(RegistrationFailureReason || (RegistrationFailureReason = {}));
var _default = RegistrationFailureReason;
exports["default"] = _default;
module.exports = exports.default;
