// @sqs/enums/SectionCategories.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;
/* File generated by JavaScriptEnumProcessor */
/* Derived from Java class com.squarespace.v6.utils.enums.SectionCategories */
/* Java class defined in repository squarespace-v6 */
var SectionCategories;
(function (SectionCategories) {
  SectionCategories["ABOUT"] = "ABOUT";
  SectionCategories["APPOINTMENTS"] = "APPOINTMENTS";
  SectionCategories["BANNER"] = "BANNER";
  SectionCategories["CALENDAR"] = "CALENDAR";
  SectionCategories["CHART"] = "CHART";
  SectionCategories["CONTACT"] = "CONTACT";
  SectionCategories["DONATION"] = "DONATION";
  SectionCategories["EVENTS"] = "EVENTS";
  SectionCategories["FAQS"] = "FAQS";
  SectionCategories["FOOTER"] = "FOOTER";
  SectionCategories["FORM"] = "FORM";
  SectionCategories["GALLERY"] = "GALLERY";
  SectionCategories["HEADLINES"] = "HEADLINES";
  SectionCategories["IMAGES"] = "IMAGES";
  SectionCategories["INTRO"] = "INTRO";
  SectionCategories["LIST"] = "LIST";
  SectionCategories["LOCATION"] = "LOCATION";
  SectionCategories["LOGO_WALL"] = "LOGO_WALL";
  SectionCategories["MEMBERS"] = "MEMBERS";
  SectionCategories["MENU"] = "MENU";
  SectionCategories["MUSIC"] = "MUSIC";
  SectionCategories["NEWSLETTER"] = "NEWSLETTER";
  SectionCategories["PEOPLE"] = "PEOPLE";
  SectionCategories["PORTFOLIOS"] = "PORTFOLIOS";
  SectionCategories["PRICING"] = "PRICING";
  SectionCategories["PRODUCTS"] = "PRODUCTS";
  SectionCategories["QUOTE"] = "QUOTE";
  SectionCategories["REGISTRY"] = "REGISTRY";
  SectionCategories["RESERVATIONS"] = "RESERVATIONS";
  SectionCategories["RESUME"] = "RESUME";
  SectionCategories["SAVED_SECTIONS"] = "SAVED_SECTIONS";
  SectionCategories["SERVICES"] = "SERVICES";
  SectionCategories["SOCIAL"] = "SOCIAL";
  SectionCategories["STAGING"] = "STAGING";
  SectionCategories["SUMMARY"] = "SUMMARY";
  SectionCategories["TESTIMONIALS"] = "TESTIMONIALS";
  SectionCategories["TEXT"] = "TEXT";
  SectionCategories["TOUR_DATES"] = "TOUR_DATES";
  SectionCategories["VIDEO"] = "VIDEO";
  SectionCategories["DIGITAL_PRODUCTS"] = "DIGITAL_PRODUCTS";
})(SectionCategories || (SectionCategories = {}));
var _default = SectionCategories;
exports["default"] = _default;
module.exports = exports.default;
