// @sqs/enums/SevenOneMigrationHintType.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;
/* File generated by JavaScriptEnumProcessor */
/* Derived from Java class com.squarespace.v6.model.sevenonemigration.SevenOneMigrationHintType */
/* Java class defined in repository squarespace-v6 */
var SevenOneMigrationHintType;
(function (SevenOneMigrationHintType) {
  SevenOneMigrationHintType["UNSUPPORTED_HOMEPAGE"] = "UNSUPPORTED_HOMEPAGE";
  SevenOneMigrationHintType["UNSUPPORTED_COLLECTION"] = "UNSUPPORTED_COLLECTION";
  SevenOneMigrationHintType["REPLACED_HOMEPAGE"] = "REPLACED_HOMEPAGE";
  SevenOneMigrationHintType["GALLERY_COLLECTION"] = "GALLERY_COLLECTION";
  SevenOneMigrationHintType["INDEX_STACKED_COLLECTION"] = "INDEX_STACKED_COLLECTION";
  SevenOneMigrationHintType["COVER_PAGE"] = "COVER_PAGE";
  SevenOneMigrationHintType["INDEX_GRID_OR_SLIDESHOW_COLLECTION"] = "INDEX_GRID_OR_SLIDESHOW_COLLECTION";
  SevenOneMigrationHintType["PROJECT_PAGE"] = "PROJECT_PAGE";
  SevenOneMigrationHintType["TOO_MANY_PAGES"] = "TOO_MANY_PAGES";
})(SevenOneMigrationHintType || (SevenOneMigrationHintType = {}));
var _default = SevenOneMigrationHintType;
exports["default"] = _default;
module.exports = exports.default;
