// @sqs/enums/ShapeBlockShapes.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;
/* File generated by JavaScriptEnumProcessor */
/* Derived from Java class com.squarespace.v6.utils.enums.ShapeBlockShape */
/* Java class defined in repository cms-legacy-data */
var ShapeBlockShapes;
(function (ShapeBlockShapes) {
  ShapeBlockShapes["ARCHED_WINDOW"] = "arched-window";
  ShapeBlockShapes["AWARD"] = "award";
  ShapeBlockShapes["CIRCLE"] = "circle";
  ShapeBlockShapes["DIAMOND"] = "diamond";
  ShapeBlockShapes["EGG"] = "egg";
  ShapeBlockShapes["EYE"] = "eye";
  ShapeBlockShapes["EIGHT_SIDED_SHIELD"] = "eight-sided-shield";
  ShapeBlockShapes["FLOWER"] = "flower";
  ShapeBlockShapes["FOUR_LEAF_CLOVER"] = "four-leaf-clover";
  ShapeBlockShapes["FOUR_PETAL_FLOWER"] = "four-petal-flower";
  ShapeBlockShapes["FOUR_POINT_STAR"] = "four-point-star";
  ShapeBlockShapes["GEAR"] = "gear";
  ShapeBlockShapes["HORIZONTAL_RIGHT_LEANING_OVAL"] = "horizontal-right-leaning-oval";
  ShapeBlockShapes["HORIZONTAL_STACKED_CIRCLES"] = "horizontal-stacked-circles";
  ShapeBlockShapes["HOURGLASS"] = "hourglass";
  ShapeBlockShapes["KEYSTONE"] = "keystone";
  ShapeBlockShapes["NARROW_CORSET"] = "narrow-corset";
  ShapeBlockShapes["NARROW_POW"] = "narrow-pow";
  ShapeBlockShapes["NARROW_RIBBON"] = "narrow-ribbon";
  ShapeBlockShapes["NARROW_VERTICAL_STACKED_PILLS"] = "narrow-vertical-stacked-pills";
  ShapeBlockShapes["OCTAGON"] = "octagon";
  ShapeBlockShapes["OPEN_BOOK"] = "open-book";
  ShapeBlockShapes["PARALLELOGRAM"] = "parallelogram";
  ShapeBlockShapes["PENTAGON"] = "pentagon";
  ShapeBlockShapes["POINTED_STAR"] = "pointed-star";
  ShapeBlockShapes["POINTED_WINDOW"] = "pointed-window";
  ShapeBlockShapes["RABBET"] = "rabbet";
  ShapeBlockShapes["RECTANGLE"] = "rectangle";
  ShapeBlockShapes["ROUNDED_ARCHED_WINDOW"] = "rounded-arched-window";
  ShapeBlockShapes["ROUNDED_DIAMOND"] = "rounded-diamond";
  ShapeBlockShapes["ROUNDED_EIGHT_POINT_STAR"] = "rounded-eight-point-star";
  ShapeBlockShapes["ROUNDED_FOUR_POINT_STAR"] = "rounded-four-point-star";
  ShapeBlockShapes["ROUNDED_PENTAGON"] = "rounded-pentagon";
  ShapeBlockShapes["ROUNDED_RIBBON"] = "rounded-ribbon";
  ShapeBlockShapes["ROUNDED_STAR"] = "rounded-star";
  ShapeBlockShapes["ROUNDED_TEN_POINT_STAR"] = "rounded-ten-point-star";
  ShapeBlockShapes["STEPPED_CROSS"] = "stepped-cross";
  ShapeBlockShapes["TRIANGLE"] = "triangle";
  ShapeBlockShapes["TV"] = "tv";
  ShapeBlockShapes["VERTICAL_PARALLELOGRAM"] = "vertical-parallelogram";
  ShapeBlockShapes["WIDE_HORIZONTAL_STACKED_PILLS"] = "wide-horizontal-stacked-pills";
  ShapeBlockShapes["WIDE_VERTICAL_STACKED_PILLS"] = "wide-vertical-stacked-pills";
})(ShapeBlockShapes || (ShapeBlockShapes = {}));
var _default = ShapeBlockShapes;
exports["default"] = _default;
module.exports = exports.default;
