// @sqs/enums/ShippingOptionServiceName.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;
/* File generated by JavaScriptEnumProcessor */
/* Derived from Java class com.squarespace.v6.business.commerce.shipping.ShippingOptionServiceName */
/* Java class defined in repository squarespace-v6 */
var ShippingOptionServiceName;
(function (ShippingOptionServiceName) {
  ShippingOptionServiceName["FEDEX_GROUND"] = "FedEx Ground\xAE";
  ShippingOptionServiceName["FEDEX_GROUND_HOME_DELIVERY"] = "FedEx Ground Home Delivery\xAE";
  ShippingOptionServiceName["FEDEX_PRIORITY_OVERNIGHT"] = "FedEx Priority Overnight\xAE";
  ShippingOptionServiceName["FEDEX_STANDARD_OVERNIGHT"] = "FedEx Standard Overnight\xAE";
  ShippingOptionServiceName["FEDEX_2_DAY"] = "FedEx 2Day\xAE";
  ShippingOptionServiceName["FEDEX_2_DAY_AM"] = "FedEx 2Day\xAE A.M.";
  ShippingOptionServiceName["FEDEX_EXPRESS_SAVER"] = "FedEx Express Saver\xAE";
  ShippingOptionServiceName["FEDEX_FIRST_OVERNIGHT"] = "FedEx First Overnight\xAE";
  ShippingOptionServiceName["FEDEX_SAME_DAY"] = "FedEx SameDay\xAE";
  ShippingOptionServiceName["FEDEX_SAME_DAY_CITY"] = "FedEx SameDay\xAE City";
  ShippingOptionServiceName["UPS_GROUND"] = "UPS Ground";
  ShippingOptionServiceName["UPS_NEXT_DAY_AIR"] = "UPS Next Day Air\xAE";
  ShippingOptionServiceName["UPS_NEXT_DAY_AIR_SAVER"] = "UPS Next Day Air\xAE Saver";
  ShippingOptionServiceName["UPS_NEXT_DAY_AIR_EARLY_AM"] = "UPS Next Day Air\xAE Early A.M.\xAE";
  ShippingOptionServiceName["UPS_TWO_DAY_AIR"] = "UPS 2nd Day Air\xAE";
  ShippingOptionServiceName["UPS_TWO_DAY_AIR_AM"] = "UPS 2nd Day Air A.M.";
  ShippingOptionServiceName["UPS_THREE_DAY_SELECT"] = "UPS 3 Day Select\xAE";
  ShippingOptionServiceName["USPS_PRIORITY"] = "USPS\xAE Priority Mail\xAE";
  ShippingOptionServiceName["USPS_EXPRESS"] = "USPS\xAE Priority Mail Express\xAE";
  ShippingOptionServiceName["USPS_STANDARD"] = "USPS\xAE Ground Advantage\u2122";
})(ShippingOptionServiceName || (ShippingOptionServiceName = {}));
var _default = ShippingOptionServiceName;
exports["default"] = _default;
module.exports = exports.default;
