// @sqs/enums/StripePaymentTokenType.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;
/* File generated by JavaScriptEnumProcessor */
/* Derived from Java class com.squarespace.business.commerce.payments.stripe.StripePaymentTokenType */
/* Java class defined in repository squarespace-v6 */
var StripePaymentTokenType;
(function (StripePaymentTokenType) {
  StripePaymentTokenType["LEGACY_TOKEN"] = "LEGACY_TOKEN";
  StripePaymentTokenType["PAYMENT_METHOD_ID"] = "PAYMENT_METHOD_ID";
  StripePaymentTokenType["PAYMENT_INTENT_ID"] = "PAYMENT_INTENT_ID";
  StripePaymentTokenType["AFTERPAY_PAYMENT_INTENT_ID"] = "AFTERPAY_PAYMENT_INTENT_ID";
  StripePaymentTokenType["KLARNA_PAYMENT_INTENT_ID"] = "KLARNA_PAYMENT_INTENT_ID";
  StripePaymentTokenType["AFTERPAY_CLEARPAY"] = "AFTERPAY_CLEARPAY";
  StripePaymentTokenType["KLARNA"] = "KLARNA";
})(StripePaymentTokenType || (StripePaymentTokenType = {}));
var _default = StripePaymentTokenType;
exports["default"] = _default;
module.exports = exports.default;
