// @sqs/enums/StripeExtraAuthenticationErrorType.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;
/* File generated by JavaScriptEnumProcessor */
/* Derived from Java class com.squarespace.business.commerce.payments.stripe.StripeExtraAuthenticationErrorType */
/* Java class defined in repository squarespace-v6 */
var StripeExtraAuthenticationErrorType;
(function (StripeExtraAuthenticationErrorType) {
  StripeExtraAuthenticationErrorType["PAYMENT_AUTHENTICATION"] = "payment-authentication";
  StripeExtraAuthenticationErrorType["SETUP_AUTHENTICATION"] = "setup-authentication";
  StripeExtraAuthenticationErrorType["AFTERPAY_AUTHENTICATION"] = "afterpay-authentication";
  StripeExtraAuthenticationErrorType["KLARNA_AUTHENTICATION"] = "klarna-authentication";
  StripeExtraAuthenticationErrorType["UNKNOWN"] = "unknown";
})(StripeExtraAuthenticationErrorType || (StripeExtraAuthenticationErrorType = {}));
var _default = StripeExtraAuthenticationErrorType;
exports["default"] = _default;
module.exports = exports.default;
