// @sqs/enums/SubmitOrderFailureType.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;
/* File generated by JavaScriptEnumProcessor */
/* Derived from Java class com.squarespace.v6.business.commerce.order.operation.submit.SubmitOrderFailureType */
/* Java class defined in repository squarespace-v6 */
var SubmitOrderFailureType;
(function (SubmitOrderFailureType) {
  SubmitOrderFailureType["INVALID_ORDER_SUBMIT_REQUEST"] = "INVALID_ORDER_SUBMIT_REQUEST";
  SubmitOrderFailureType["INVALID_EMAIL_ADDRESS"] = "INVALID_EMAIL_ADDRESS";
  SubmitOrderFailureType["MISSING_EMAIL_ADDRESS"] = "MISSING_EMAIL_ADDRESS";
  SubmitOrderFailureType["INVALID_SHIPPING_ADDRESS"] = "INVALID_SHIPPING_ADDRESS";
  SubmitOrderFailureType["MISSING_SHIPPING_ADDRESS"] = "MISSING_SHIPPING_ADDRESS";
  SubmitOrderFailureType["INVALID_BILLING_ADDRESS"] = "INVALID_BILLING_ADDRESS";
  SubmitOrderFailureType["MISSING_BILLING_ADDRESS"] = "MISSING_BILLING_ADDRESS";
  SubmitOrderFailureType["INVALID_IDEMPOTENCY_KEY"] = "INVALID_IDEMPOTENCY_KEY";
  SubmitOrderFailureType["INVALID_PAYMENT_TOKEN"] = "INVALID_PAYMENT_TOKEN";
  SubmitOrderFailureType["INVALID_DONATION_AMOUNT"] = "INVALID_DONATION_AMOUNT";
  SubmitOrderFailureType["INVALID_GIFT_CARD_TOTAL"] = "INVALID_GIFT_CARD_TOTAL";
  SubmitOrderFailureType["INVALID_SHIPPING_OPTION"] = "INVALID_SHIPPING_OPTION";
  SubmitOrderFailureType["SHIPPING_OPTION_NOT_SELECTED"] = "SHIPPING_OPTION_NOT_SELECTED";
  SubmitOrderFailureType["SHIPPING_OPTION_MISSING"] = "SHIPPING_OPTION_MISSING";
  SubmitOrderFailureType["SHIPPING_OPTION_NOT_APPLICABLE"] = "SHIPPING_OPTION_NOT_APPLICABLE";
  SubmitOrderFailureType["SHIPPING_OPTION_NOT_AVAILABLE_FOR_ADDRESS"] = "SHIPPING_OPTION_NOT_AVAILABLE_FOR_ADDRESS";
  SubmitOrderFailureType["SHIPPING_CALCULATION_ERROR"] = "SHIPPING_CALCULATION_ERROR";
  SubmitOrderFailureType["INVALID_DISCOUNT"] = "INVALID_DISCOUNT";
  SubmitOrderFailureType["STALE_DISCOUNT_STATE_CHANGED"] = "STALE_DISCOUNT_STATE_CHANGED";
  SubmitOrderFailureType["SITE_USER_ACCOUNT_ERROR"] = "SITE_USER_ACCOUNT_ERROR";
  SubmitOrderFailureType["SITE_USER_ACCOUNT_ALREADY_EXISTS"] = "SITE_USER_ACCOUNT_ALREADY_EXISTS";
  SubmitOrderFailureType["SITE_USER_ACCOUNTS_NOT_ALLOWED"] = "SITE_USER_ACCOUNTS_NOT_ALLOWED";
  SubmitOrderFailureType["ACCOUNT_REQUIRED"] = "ACCOUNT_REQUIRED";
  SubmitOrderFailureType["INVALID_PASSWORD"] = "INVALID_PASSWORD";
  SubmitOrderFailureType["SUBSCRIPTION_SERVICE_ERROR"] = "SUBSCRIPTION_SERVICE_ERROR";
  SubmitOrderFailureType["SUBSCRIPTION_SERVICE_UNAVAILABLE"] = "SUBSCRIPTION_SERVICE_UNAVAILABLE";
  SubmitOrderFailureType["CREATE_SUBSCRIPTION_ERROR"] = "CREATE_SUBSCRIPTION_ERROR";
  SubmitOrderFailureType["INVALID_SUBSCRIPTION_PRODUCT"] = "INVALID_SUBSCRIPTION_PRODUCT";
  SubmitOrderFailureType["MISSING_PAYMENT_METHOD"] = "MISSING_PAYMENT_METHOD";
  SubmitOrderFailureType["COMMERCE_CART_PRICING_SERVICE_ERROR"] = "COMMERCE_CART_PRICING_SERVICE_ERROR";
  SubmitOrderFailureType["COMMERCE_CART_PRICING_SERVICE_UNAVAILABLE"] = "COMMERCE_CART_PRICING_SERVICE_UNAVAILABLE";
  SubmitOrderFailureType["COMMERCE_CART_SERVICE_ERROR"] = "COMMERCE_CART_SERVICE_ERROR";
  SubmitOrderFailureType["COMMERCE_CART_SERVICE_UNAVAILABLE"] = "COMMERCE_CART_SERVICE_UNAVAILABLE";
  SubmitOrderFailureType["GIFT_CARD_SERVICE_ERROR"] = "GIFT_CARD_SERVICE_ERROR";
  SubmitOrderFailureType["GIFT_CARD_SERVICE_UNAVAILABLE"] = "GIFT_CARD_SERVICE_UNAVAILABLE";
  SubmitOrderFailureType["COMMERCE_TAX_SERVICE_ERROR"] = "COMMERCE_TAX_SERVICE_ERROR";
  SubmitOrderFailureType["COMMERCE_TAX_SERVICE_UNAVAILABLE"] = "COMMERCE_TAX_SERVICE_UNAVAILABLE";
  SubmitOrderFailureType["COMMERCE_SHIPPING_SERVICE_ERROR"] = "COMMERCE_SHIPPING_SERVICE_ERROR";
  SubmitOrderFailureType["COMMERCE_SHIPPING_SERVICE_UNAVAILABLE"] = "COMMERCE_SHIPPING_SERVICE_UNAVAILABLE";
  SubmitOrderFailureType["COMMERCE_ORDERS_SERVICE_ERROR"] = "COMMERCE_ORDERS_SERVICE_ERROR";
  SubmitOrderFailureType["COMMERCE_ORDERS_SERVICE_UNAVAILABLE"] = "COMMERCE_ORDERS_SERVICE_UNAVAILABLE";
  SubmitOrderFailureType["COMMERCE_ORDERS_SERVICE_TIMEOUT"] = "COMMERCE_ORDERS_SERVICE_TIMEOUT";
  SubmitOrderFailureType["COMMERCE_CART_PERSISTENCE_SERVICE_ERROR"] = "COMMERCE_CART_PERSISTENCE_SERVICE_ERROR";
  SubmitOrderFailureType["COMMERCE_CART_PERSISTENCE_SERVICE_UNAVAILABLE"] = "COMMERCE_CART_PERSISTENCE_SERVICE_UNAVAILABLE";
  SubmitOrderFailureType["COMMERCE_DISCOUNT_SERVICE_ERROR"] = "COMMERCE_DISCOUNT_SERVICE_ERROR";
  SubmitOrderFailureType["COMMERCE_DISCOUNT_SERVICE_UNAVAILABLE"] = "COMMERCE_DISCOUNT_SERVICE_UNAVAILABLE";
  SubmitOrderFailureType["GATEWAY_AUTHENTICATION_ERROR"] = "GATEWAY_AUTHENTICATION_ERROR";
  SubmitOrderFailureType["PAYMENT_GATEWAY_AUTHENTICATION_ERROR"] = "PAYMENT_GATEWAY_AUTHENTICATION_ERROR";
  SubmitOrderFailureType["NO_SUCH_PAYMENT_TOKEN"] = "NO_SUCH_PAYMENT_TOKEN";
  SubmitOrderFailureType["LOCAL_AUTHENTICATION_ERROR"] = "LOCAL_AUTHENTICATION_ERROR";
  SubmitOrderFailureType["STALE_USER_SESSION"] = "STALE_USER_SESSION";
  SubmitOrderFailureType["RECOVERABLE_PAYMENT_ERROR"] = "RECOVERABLE_PAYMENT_ERROR";
  SubmitOrderFailureType["GIFT_CARD_PAYMENT_ERROR"] = "GIFT_CARD_PAYMENT_ERROR";
  SubmitOrderFailureType["INVALID_PAYMENT_AMOUNT"] = "INVALID_PAYMENT_AMOUNT";
  SubmitOrderFailureType["PAYMENT_DECLINED"] = "PAYMENT_DECLINED";
  SubmitOrderFailureType["PAYMENT_METHOD_NOT_FOUND"] = "PAYMENT_METHOD_NOT_FOUND";
  SubmitOrderFailureType["EXTRA_PAYMENT_AUTHENTICATION_REQUIRED"] = "EXTRA_PAYMENT_AUTHENTICATION_REQUIRED";
  SubmitOrderFailureType["PAYMENT_PROCESSING_UNAVAILABLE"] = "PAYMENT_PROCESSING_UNAVAILABLE";
  SubmitOrderFailureType["RESTRICTED_OR_CLOSED_PAYMENT_ACCOUNT"] = "RESTRICTED_OR_CLOSED_PAYMENT_ACCOUNT";
  SubmitOrderFailureType["UNSUPPORTED_CURRENCY"] = "UNSUPPORTED_CURRENCY";
  SubmitOrderFailureType["PAYMENT_GATEWAY_TIMEOUT"] = "PAYMENT_GATEWAY_TIMEOUT";
  SubmitOrderFailureType["SAVED_PAYMENT_ACCOUNTS_UNAVAILABLE"] = "SAVED_PAYMENT_ACCOUNTS_UNAVAILABLE";
  SubmitOrderFailureType["CUSTOMER_DOES_NOT_HAVE_LINKED_SOURCE"] = "CUSTOMER_DOES_NOT_HAVE_LINKED_SOURCE";
  SubmitOrderFailureType["PAYMENTS_NOT_ALLOWED"] = "PAYMENTS_NOT_ALLOWED";
  SubmitOrderFailureType["STORE_LIVE_IN_TRIAL_MODE"] = "STORE_LIVE_IN_TRIAL_MODE";
  SubmitOrderFailureType["STORE_LIVE_COMMERCE_DISABLED"] = "STORE_LIVE_COMMERCE_DISABLED";
  SubmitOrderFailureType["STORE_LIVE_SUBSCRIPTIONS_DISABLED"] = "STORE_LIVE_SUBSCRIPTIONS_DISABLED";
  SubmitOrderFailureType["CASH_NOT_SUPPORTED"] = "CASH_NOT_SUPPORTED";
  SubmitOrderFailureType["SQUARE_NOT_SUPPORTED"] = "SQUARE_NOT_SUPPORTED";
  SubmitOrderFailureType["SQUARE_NOT_SUPPORTED_IN_TEST_MODE"] = "SQUARE_NOT_SUPPORTED_IN_TEST_MODE";
  SubmitOrderFailureType["SQSP_PAYMENTS_NOT_SUPPORTED_IN_TEST_MODE"] = "SQSP_PAYMENTS_NOT_SUPPORTED_IN_TEST_MODE";
  SubmitOrderFailureType["DONATION_BELOW_MINIMUM_ALLOWED"] = "DONATION_BELOW_MINIMUM_ALLOWED";
  SubmitOrderFailureType["DONATIONS_DISABLED"] = "DONATIONS_DISABLED";
  SubmitOrderFailureType["GIFT_CARD_FEATURES_DISABLED"] = "GIFT_CARD_FEATURES_DISABLED";
  SubmitOrderFailureType["MEMBER_AREAS_DISABLED"] = "MEMBER_AREAS_DISABLED";
  SubmitOrderFailureType["AFTERPAY_NOT_SUPPORTED"] = "AFTERPAY_NOT_SUPPORTED";
  SubmitOrderFailureType["CONSISTENCY_ERROR"] = "CONSISTENCY_ERROR";
  SubmitOrderFailureType["CART_ALREADY_PROCESSING"] = "CART_ALREADY_PROCESSING";
  SubmitOrderFailureType["CART_ALREADY_PURCHASED"] = "CART_ALREADY_PURCHASED";
  SubmitOrderFailureType["CART_EMPTY"] = "CART_EMPTY";
  SubmitOrderFailureType["CART_ITEM_OUT_OF_STOCK"] = "CART_ITEM_OUT_OF_STOCK";
  SubmitOrderFailureType["CART_ITEM_QUANTITY_CONSTRAINT_VIOLATION"] = "CART_ITEM_QUANTITY_CONSTRAINT_VIOLATION";
  SubmitOrderFailureType["CART_MISSING"] = "CART_MISSING";
  SubmitOrderFailureType["CART_TOTAL_OUT_OF_SYNC"] = "CART_TOTAL_OUT_OF_SYNC";
  SubmitOrderFailureType["GIFT_CARD_FEATURE_DISABLED"] = "GIFT_CARD_FEATURE_DISABLED";
  SubmitOrderFailureType["GIFT_CARD_PRODUCT_MISSING"] = "GIFT_CARD_PRODUCT_MISSING";
  SubmitOrderFailureType["GIFT_CARD_PRICE_OUT_OF_SYNC"] = "GIFT_CARD_PRICE_OUT_OF_SYNC";
  SubmitOrderFailureType["PAYMENT_TOKEN_ALREADY_USED"] = "PAYMENT_TOKEN_ALREADY_USED";
  SubmitOrderFailureType["BEFORE_MEMBER_AREA_ENROLLMENT_DATES_PERIOD"] = "BEFORE_MEMBER_AREA_ENROLLMENT_DATES_PERIOD";
  SubmitOrderFailureType["AFTER_MEMBER_AREA_ENROLLMENT_DATES_PERIOD"] = "AFTER_MEMBER_AREA_ENROLLMENT_DATES_PERIOD";
  SubmitOrderFailureType["GATEWAY_ERROR"] = "GATEWAY_ERROR";
  SubmitOrderFailureType["UNKNOWN_STRIPE_GATEWAY_ERROR"] = "UNKNOWN_STRIPE_GATEWAY_ERROR";
  SubmitOrderFailureType["UNKNOWN_PAYPAL_GATEWAY_ERROR"] = "UNKNOWN_PAYPAL_GATEWAY_ERROR";
  SubmitOrderFailureType["UNKNOWN_SQUARE_GATEWAY_ERROR"] = "UNKNOWN_SQUARE_GATEWAY_ERROR";
  SubmitOrderFailureType["HALTED_WORKFLOW"] = "HALTED_WORKFLOW";
  SubmitOrderFailureType["INTERNAL_SERVER_ERROR"] = "INTERNAL_SERVER_ERROR";
  SubmitOrderFailureType["GIFT_CARD_CHECKOUT_UNAVAILABLE"] = "GIFT_CARD_CHECKOUT_UNAVAILABLE";
  SubmitOrderFailureType["INVENTORY_DEDUCTION_FAILURE"] = "INVENTORY_DEDUCTION_FAILURE";
  SubmitOrderFailureType["SAVE_ORDER_FAILURE"] = "SAVE_ORDER_FAILURE";
  SubmitOrderFailureType["UNKNOWN_PAYMENT_ERROR"] = "UNKNOWN_PAYMENT_ERROR";
  SubmitOrderFailureType["UNKNOWN"] = "UNKNOWN";
  SubmitOrderFailureType["SITE_USER_CREATION_FAILED"] = "SITE_USER_CREATION_FAILED";
  SubmitOrderFailureType["SITE_USER_ACCOUNT_CREATION_FAILED"] = "SITE_USER_ACCOUNT_CREATION_FAILED";
  SubmitOrderFailureType["PRODUCT_RETRIEVAL_FAILURE"] = "PRODUCT_RETRIEVAL_FAILURE";
  SubmitOrderFailureType["UNEXPECTED_NUMBER_OF_DONATION_PAYMENTS"] = "UNEXPECTED_NUMBER_OF_DONATION_PAYMENTS";
  SubmitOrderFailureType["RECAPTCHA_VERIFICATION_FAILED"] = "RECAPTCHA_VERIFICATION_FAILED";
  SubmitOrderFailureType["MINIMUM_ORDER_SUBTOTAL_NOT_MET"] = "MINIMUM_ORDER_SUBTOTAL_NOT_MET";
})(SubmitOrderFailureType || (SubmitOrderFailureType = {}));
var _default = SubmitOrderFailureType;
exports["default"] = _default;
module.exports = exports.default;
