// @sqs/enums/TaxJurisdictionType.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;
/* File generated by JavaScriptEnumProcessor */
/* Derived from Java class com.squarespace.v6.utils.enums.TaxJurisdictionType */
/* Java class defined in repository squarespace-v6 */
var TaxJurisdictionType;
(function (TaxJurisdictionType) {
  TaxJurisdictionType["APO"] = "apo";
  TaxJurisdictionType["BOROUGH"] = "borough";
  TaxJurisdictionType["CITY"] = "city";
  TaxJurisdictionType["COUNTRY"] = "country";
  TaxJurisdictionType["COUNTY"] = "county";
  TaxJurisdictionType["DISTRICT"] = "district";
  TaxJurisdictionType["FPO"] = "fpo";
  TaxJurisdictionType["LOCAL_IMPROVEMENT_DISTRICT"] = "local_improvement_district";
  TaxJurisdictionType["PARISH"] = "parish";
  TaxJurisdictionType["PROVINCE"] = "province";
  TaxJurisdictionType["SPECIAL_PURPOSE_DISTRICT"] = "special_purpose_district";
  TaxJurisdictionType["STATE"] = "state";
  TaxJurisdictionType["TERRITORY"] = "territory";
  TaxJurisdictionType["TOWNSHIP"] = "township";
  TaxJurisdictionType["TRADE_BLOCK"] = "trade_block";
  TaxJurisdictionType["TRANSIT_DISTRICT"] = "transit_district";
})(TaxJurisdictionType || (TaxJurisdictionType = {}));
var _default = TaxJurisdictionType;
exports["default"] = _default;
module.exports = exports.default;
