// @sqs/enums/Team.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;
/* File generated by JavaScriptEnumProcessor */
/* Derived from Java class com.squarespace.v6.ownership.Team */
/* Java class defined in repository squarespace-v6 */
var Team;
(function (Team) {
  Team["OWNERLESS"] = "OWNERLESS";
  Team["COMMERCE_MERCH"] = "COMMERCE_MERCH";
  Team["COMMERCE_INVENTORY"] = "COMMERCE_INVENTORY";
  Team["COMMERCE_ACTIVATION"] = "COMMERCE_ACTIVATION";
  Team["COMMERCE_CHANNELS"] = "COMMERCE_CHANNELS";
  Team["COMMERCE_DIGITAL_PRODUCTS_AND_DONATIONS"] = "COMMERCE_DIGITAL_PRODUCTS_AND_DONATIONS";
  Team["COMMERCE_PRINT_ON_DEMAND"] = "COMMERCE_PRINT_ON_DEMAND";
  Team["COMMERCE_ONLINE_MERCH"] = "COMMERCE_ONLINE_MERCH";
  Team["COMMERCE_PROJECT_SUCCESS"] = "COMMERCE_PROJECT_SUCCESS";
  Team["COMMERCE_INVOICING_TOOLS"] = "COMMERCE_INVOICING_TOOLS";
  Team["COMMERCE_MERCHANDISING"] = "COMMERCE_MERCHANDISING";
  Team["COMMERCE_MONETIZATION"] = "COMMERCE_MONETIZATION";
  Team["COMMERCE_TX"] = "COMMERCE_TX";
  Team["COMMERCE_ORDERS"] = "COMMERCE_ORDERS";
  Team["COMMERCE_ENGAGEMENT"] = "COMMERCE_ENGAGEMENT";
  Team["COMMERCE_CORE"] = "COMMERCE_CORE";
  Team["COMMERCE_SELLER_OPERATIONS"] = "COMMERCE_SELLER_OPERATIONS";
  Team["COMMERCE_CHECKOUT"] = "COMMERCE_CHECKOUT";
  Team["COMMERCE_MEMBERSHIPS"] = "COMMERCE_MEMBERSHIPS";
  Team["CAMPAIGNS_COMPOSER"] = "CAMPAIGNS_COMPOSER";
  Team["CAMPAIGNS_DELIVERY"] = "CAMPAIGNS_DELIVERY";
  Team["DISTRIBUTION"] = "DISTRIBUTION";
  Team["ANALYTICS"] = "ANALYTICS";
  Team["CRM_PRODUCT"] = "CRM_PRODUCT";
  Team["INTERFACE_ARCHITECTURE"] = "INTERFACE_ARCHITECTURE";
  Team["I18N_PLATFORM"] = "I18N_PLATFORM";
  Team["CORE_UI"] = "CORE_UI";
  Team["FORMS_FIELDS"] = "FORMS_FIELDS";
  Team["CONTENT_DATA"] = "CONTENT_DATA";
  Team["COMMUNICATION_PLATFORM"] = "COMMUNICATION_PLATFORM";
  Team["MACHINE_LEARNING"] = "MACHINE_LEARNING";
  Team["TEMPLATES"] = "TEMPLATES";
  Team["CONTENT_EDITING"] = "CONTENT_EDITING";
  Team["CONTENT_RENDERING"] = "CONTENT_RENDERING";
  Team["WEBSITE_CONTENT_COMPONENTS"] = "WEBSITE_CONTENT_COMPONENTS";
  Team["WEB_CONCEPTS"] = "WEB_CONCEPTS";
  Team["WEB_PERFORMANCE"] = "WEB_PERFORMANCE";
  Team["CORE_CMS"] = "CORE_CMS";
  Team["ASSET_MANAGEMENT"] = "ASSET_MANAGEMENT";
  Team["ACCOUNTS"] = "ACCOUNTS";
  Team["DOMAINS"] = "DOMAINS";
  Team["G_SUITE"] = "G_SUITE";
  Team["EMAIL_PRODUCT"] = "EMAIL_PRODUCT";
  Team["DOMAIN_REGISTRAR"] = "DOMAIN_REGISTRAR";
  Team["CORE_MOBILE"] = "CORE_MOBILE";
  Team["DISCOVERY"] = "DISCOVERY";
  Team["CMS_RENDERING"] = "CMS_RENDERING";
  Team["PHENSLEY"] = "PHENSLEY";
  Team["BRAND_SYSTEM"] = "BRAND_SYSTEM";
  Team["EDITING_SYSTEM"] = "EDITING_SYSTEM";
  Team["DIGITAL_PRODUCT_ACTIVATION"] = "DIGITAL_PRODUCT_ACTIVATION";
  Team["DIGITAL_PRODUCT_MONETIZATION"] = "DIGITAL_PRODUCT_MONETIZATION";
  Team["DIGITAL_PRODUCT_CONTENT"] = "DIGITAL_PRODUCT_CONTENT";
  Team["DIGITAL_PRODUCT_VIDEO_ASSETS"] = "DIGITAL_PRODUCT_VIDEO_ASSETS";
  Team["WEBSITE_LAYOUT_AND_STYLING"] = "WEBSITE_LAYOUT_AND_STYLING";
  Team["WEBSITE_EDITOR"] = "WEBSITE_EDITOR";
  Team["WEBSITES_CONTENT"] = "WEBSITES_CONTENT";
  Team["WEBSITE_CHARACTER"] = "WEBSITE_CHARACTER";
  Team["WEBSITE_PLATFORM_API_APPLICATION"] = "WEBSITE_PLATFORM_API_APPLICATION";
  Team["WEBSITE_PLATFORM_RENDERING_EXTENSIBILITY"] = "WEBSITE_PLATFORM_RENDERING_EXTENSIBILITY";
  Team["CORE_LAYOUT"] = "CORE_LAYOUT";
  Team["STYLING_SYSTEM"] = "STYLING_SYSTEM";
  Team["I18N_PRODUCT"] = "I18N_PRODUCT";
  Team["CONTEXTUAL_LAYOUT"] = "CONTEXTUAL_LAYOUT";
  Team["WEBSITES"] = "WEBSITES";
  Team["FRONTSITE"] = "FRONTSITE";
  Team["CONVERSION"] = "CONVERSION";
  Team["TRIAL_EXPERIENCE"] = "TRIAL_EXPERIENCE";
  Team["BUSINESS_CONCIERGE"] = "BUSINESS_CONCIERGE";
  Team["CUSTOMER_LIFECYCLE"] = "CUSTOMER_LIFECYCLE";
  Team["BILLING"] = "BILLING";
  Team["GROWTH_TOOLS"] = "GROWTH_TOOLS";
  Team["CUSTOM_STARTING_POINTS"] = "CUSTOM_STARTING_POINTS";
  Team["CORE_SERVICES"] = "CORE_SERVICES";
  Team["SITE_INFRASTRUCTURE"] = "SITE_INFRASTRUCTURE";
  Team["ENTERPRISE"] = "ENTERPRISE";
  Team["MEDIA_INFRASTRUCTURE"] = "MEDIA_INFRASTRUCTURE";
  Team["CHECKOUT_PAYMENTS"] = "CHECKOUT_PAYMENTS";
  Team["CUSTOPS_ENGINEERING"] = "CUSTOPS_ENGINEERING";
  Team["INTERNAL_PRODUCTS"] = "INTERNAL_PRODUCTS";
  Team["MEDIA_PLATFORM"] = "MEDIA_PLATFORM";
  Team["ACTIVATION"] = "ACTIVATION";
  Team["CROSS_PRODUCT_SUCCESS"] = "CROSS_PRODUCT_SUCCESS";
  Team["USER_PLATFORM"] = "USER_PLATFORM";
  Team["IDENTITY_PLATFORM"] = "IDENTITY_PLATFORM";
  Team["API_AND_DEV_PLATFORM"] = "API_AND_DEV_PLATFORM";
  Team["PRICING_MERCHANDISING"] = "PRICING_MERCHANDISING";
  Team["SCHEDULING_PLATFORM"] = "SCHEDULING_PLATFORM";
  Team["PROVISIONING"] = "PROVISIONING";
  Team["I18N_ENG"] = "I18N_ENG";
  Team["CONFIG_FRONTEND_ENG"] = "CONFIG_FRONTEND_ENG";
  Team["PROTECTION_PLATFORM"] = "PROTECTION_PLATFORM";
  Team["ACTIVATION_FUNNEL"] = "ACTIVATION_FUNNEL";
})(Team || (Team = {}));
var _default = Team;
exports["default"] = _default;
module.exports = exports.default;
