// @sqs/enums/TLD.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;
/* File generated by JavaScriptEnumProcessor */
/* Derived from Java class com.squarespace.domains.commons.models.TLD */
/* Java class defined in repository domains-commons */
var TLD;
(function (TLD) {
  TLD["ACADEMY"] = "academy";
  TLD["ACCOUNTANT"] = "accountant";
  TLD["ACCOUNTANTS"] = "accountants";
  TLD["ACTOR"] = "actor";
  TLD["AGENCY"] = "agency";
  TLD["AIRFORCE"] = "airforce";
  TLD["AMSTERDAM"] = "amsterdam";
  TLD["APARTMENTS"] = "apartments";
  TLD["APP"] = "app";
  TLD["ARCHI"] = "archi";
  TLD["ARMY"] = "army";
  TLD["ART"] = "art";
  TLD["ASSOCIATES"] = "associates";
  TLD["ATTORNEY"] = "attorney";
  TLD["AU"] = "au";
  TLD["AUCTION"] = "auction";
  TLD["AUDIO"] = "audio";
  TLD["AUTOS"] = "autos";
  TLD["BABY"] = "baby";
  TLD["BAND"] = "band";
  TLD["BAR"] = "bar";
  TLD["BARGAINS"] = "bargains";
  TLD["BEAUTY"] = "beauty";
  TLD["BEER"] = "beer";
  TLD["BEST"] = "best";
  TLD["BID"] = "bid";
  TLD["BIKE"] = "bike";
  TLD["BINGO"] = "bingo";
  TLD["BIO"] = "bio";
  TLD["BIZ"] = "biz";
  TLD["BLACK"] = "black";
  TLD["BLOG"] = "blog";
  TLD["BLUE"] = "blue";
  TLD["BOATS"] = "boats";
  TLD["BOND"] = "bond";
  TLD["BOO"] = "boo";
  TLD["BOUTIQUE"] = "boutique";
  TLD["BUILDERS"] = "builders";
  TLD["BUSINESS"] = "business";
  TLD["BUZZ"] = "buzz";
  TLD["CA"] = "ca";
  TLD["CAB"] = "cab";
  TLD["CAFE"] = "cafe";
  TLD["CAM"] = "cam";
  TLD["CAMERA"] = "camera";
  TLD["CAMP"] = "camp";
  TLD["CAPETOWN"] = "capetown";
  TLD["CAPITAL"] = "capital";
  TLD["CARDS"] = "cards";
  TLD["CARE"] = "care";
  TLD["CAREERS"] = "careers";
  TLD["CASA"] = "casa";
  TLD["CASH"] = "cash";
  TLD["CASINO"] = "casino";
  TLD["CATERING"] = "catering";
  TLD["CC"] = "cc";
  TLD["CENTER"] = "center";
  TLD["CEO"] = "ceo";
  TLD["CHANNEL"] = "channel";
  TLD["CHAT"] = "chat";
  TLD["CHARITY"] = "charity";
  TLD["CHEAP"] = "cheap";
  TLD["CHURCH"] = "church";
  TLD["CITY"] = "city";
  TLD["CLAIMS"] = "claims";
  TLD["CLEANING"] = "cleaning";
  TLD["CLINIC"] = "clinic";
  TLD["CLOTHING"] = "clothing";
  TLD["CLOUD"] = "cloud";
  TLD["CLUB"] = "club";
  TLD["CO"] = "co";
  TLD["CO_UK"] = "co.uk";
  TLD["CO_IN"] = "co.in";
  TLD["CO_NZ"] = "co.nz";
  TLD["COACH"] = "coach";
  TLD["CODES"] = "codes";
  TLD["COFFEE"] = "coffee";
  TLD["COLLEGE"] = "college";
  TLD["COM"] = "com";
  TLD["COM_CO"] = "com.co";
  TLD["COMMUNITY"] = "community";
  TLD["COMPANY"] = "company";
  TLD["COMPUTER"] = "computer";
  TLD["COM_AU"] = "com.au";
  TLD["COM_MX"] = "com.mx";
  TLD["CONDOS"] = "condos";
  TLD["CONSTRUCTION"] = "construction";
  TLD["CONSULTING"] = "consulting";
  TLD["CONTACT"] = "contact";
  TLD["CONTRACTORS"] = "contractors";
  TLD["COOKING"] = "cooking";
  TLD["COOL"] = "cool";
  TLD["COUNTRY"] = "country";
  TLD["COUPONS"] = "coupons";
  TLD["COURSES"] = "courses";
  TLD["CREDIT"] = "credit";
  TLD["CREDITCARD"] = "creditcard";
  TLD["CRICKET"] = "cricket";
  TLD["CRUISES"] = "cruises";
  TLD["DAD"] = "dad";
  TLD["DANCE"] = "dance";
  TLD["DATE"] = "date";
  TLD["DATING"] = "dating";
  TLD["DAY"] = "day";
  TLD["DE"] = "de";
  TLD["DEALS"] = "deals";
  TLD["DEGREE"] = "degree";
  TLD["DELIVERY"] = "delivery";
  TLD["DEMOCRAT"] = "democrat";
  TLD["DENTAL"] = "dental";
  TLD["DENTIST"] = "dentist";
  TLD["DESI"] = "desi";
  TLD["DESIGN"] = "design";
  TLD["DEV"] = "dev";
  TLD["DIAMONDS"] = "diamonds";
  TLD["DIGITAL"] = "digital";
  TLD["DIRECT"] = "direct";
  TLD["DIRECTORY"] = "directory";
  TLD["DISCOUNT"] = "discount";
  TLD["DOCTOR"] = "doctor";
  TLD["DOG"] = "dog";
  TLD["DOMAINS"] = "domains";
  TLD["DOWNLOAD"] = "download";
  TLD["DURBAN"] = "durban";
  TLD["EARTH"] = "earth";
  TLD["ECO"] = "eco";
  TLD["EDUCATION"] = "education";
  TLD["EMAIL"] = "email";
  TLD["ENERGY"] = "energy";
  TLD["ENGINEER"] = "engineer";
  TLD["ENGINEERING"] = "engineering";
  TLD["ENTERPRISES"] = "enterprises";
  TLD["EQUIPMENT"] = "equipment";
  TLD["ESQ"] = "esq";
  TLD["ESTATE"] = "estate";
  TLD["EU"] = "eu";
  TLD["EVENTS"] = "events";
  TLD["EXCHANGE"] = "exchange";
  TLD["EXPERT"] = "expert";
  TLD["EXPOSED"] = "exposed";
  TLD["EXPRESS"] = "express";
  TLD["FAIL"] = "fail";
  TLD["FAITH"] = "faith";
  TLD["FAMILY"] = "family";
  TLD["FAN"] = "fan";
  TLD["FANS"] = "fans";
  TLD["FARM"] = "farm";
  TLD["FASHION"] = "fashion";
  TLD["FINANCE"] = "finance";
  TLD["FINANCIAL"] = "financial";
  TLD["FISH"] = "fish";
  TLD["FISHING"] = "fishing";
  TLD["FIT"] = "fit";
  TLD["FITNESS"] = "fitness";
  TLD["FLIGHTS"] = "flights";
  TLD["FLORIST"] = "florist";
  TLD["FLOWERS"] = "flowers";
  TLD["FOO"] = "foo";
  TLD["FOOTBALL"] = "football";
  TLD["FORSALE"] = "forsale";
  TLD["FOUNDATION"] = "foundation";
  TLD["FR"] = "fr";
  TLD["FUN"] = "fun";
  TLD["FUND"] = "fund";
  TLD["FURNITURE"] = "furniture";
  TLD["FUTBOL"] = "futbol";
  TLD["FYI"] = "fyi";
  TLD["GALLERY"] = "gallery";
  TLD["GAMES"] = "games";
  TLD["GAY"] = "gay";
  TLD["GIFTS"] = "gifts";
  TLD["GIVES"] = "gives";
  TLD["GLASS"] = "glass";
  TLD["GLOBAL"] = "global";
  TLD["GMBH"] = "gmbh";
  TLD["GOLD"] = "gold";
  TLD["GOLF"] = "golf";
  TLD["GRAPHICS"] = "graphics";
  TLD["GRATIS"] = "gratis";
  TLD["GREEN"] = "green";
  TLD["GRIPE"] = "gripe";
  TLD["GROUP"] = "group";
  TLD["GUIDE"] = "guide";
  TLD["GURU"] = "guru";
  TLD["HAIR"] = "hair";
  TLD["HAUS"] = "haus";
  TLD["HEALTH"] = "health";
  TLD["HEALTHCARE"] = "healthcare";
  TLD["HOCKEY"] = "hockey";
  TLD["HOLDINGS"] = "holdings";
  TLD["HOLIDAY"] = "holiday";
  TLD["HOMES"] = "homes";
  TLD["HORSE"] = "horse";
  TLD["HOSPITAL"] = "hospital";
  TLD["HOST"] = "host";
  TLD["HOUSE"] = "house";
  TLD["HOW"] = "how";
  TLD["ICU"] = "icu";
  TLD["IMMO"] = "immo";
  TLD["IMMOBILIEN"] = "immobilien";
  TLD["IN"] = "in";
  TLD["INDUSTRIES"] = "industries";
  TLD["INFO"] = "info";
  TLD["INK"] = "ink";
  TLD["INSTITUTE"] = "institute";
  TLD["INSURE"] = "insure";
  TLD["INTERNATIONAL"] = "international";
  TLD["INVESTMENTS"] = "investments";
  TLD["IO"] = "io";
  TLD["IRISH"] = "irish";
  TLD["IT"] = "it";
  TLD["JETZT"] = "jetzt";
  TLD["JEWELRY"] = "jewelry";
  TLD["JOBURG"] = "joburg";
  TLD["JP"] = "jp";
  TLD["KAUFEN"] = "kaufen";
  TLD["KIM"] = "kim";
  TLD["KITCHEN"] = "kitchen";
  TLD["KIWI"] = "kiwi";
  TLD["LAND"] = "land";
  TLD["LAT"] = "lat";
  TLD["LAWYER"] = "lawyer";
  TLD["LEASE"] = "lease";
  TLD["LEGAL"] = "legal";
  TLD["LGBT"] = "lgbt";
  TLD["LIFE"] = "life";
  TLD["LIGHTING"] = "lighting";
  TLD["LIMITED"] = "limited";
  TLD["LIMO"] = "limo";
  TLD["LIVE"] = "live";
  TLD["LLC"] = "llc";
  TLD["LOAN"] = "loan";
  TLD["LOANS"] = "loans";
  TLD["LOL"] = "lol";
  TLD["LONDON"] = "london";
  TLD["LOVE"] = "love";
  TLD["LTD"] = "ltd";
  TLD["LUXURY"] = "luxury";
  TLD["MAISON"] = "maison";
  TLD["MAKEUP"] = "makeup";
  TLD["MANAGEMENT"] = "management";
  TLD["MARKET"] = "market";
  TLD["MARKETING"] = "marketing";
  TLD["MBA"] = "mba";
  TLD["ME"] = "me";
  TLD["MEDIA"] = "media";
  TLD["MEMORIAL"] = "memorial";
  TLD["MEN"] = "men";
  TLD["MENU"] = "menu";
  TLD["MIAMI"] = "miami";
  TLD["MOBI"] = "mobi";
  TLD["MODA"] = "moda";
  TLD["MOE"] = "moe";
  TLD["MOM"] = "mom";
  TLD["MONEY"] = "money";
  TLD["MONSTER"] = "monster";
  TLD["MORTGAGE"] = "mortgage";
  TLD["MOV"] = "mov";
  TLD["MOVIE"] = "movie";
  TLD["MUSIC"] = "music";
  TLD["MX"] = "mx";
  TLD["NAGOYA"] = "nagoya";
  TLD["NAVY"] = "navy";
  TLD["NET"] = "net";
  TLD["NET_CO"] = "net.co";
  TLD["NETWORK"] = "network";
  TLD["NEW"] = "new";
  TLD["NEWS"] = "news";
  TLD["NEXUS"] = "nexus";
  TLD["NINJA"] = "ninja";
  TLD["NL"] = "nl";
  TLD["NOM_CO"] = "nom.co";
  TLD["NZ"] = "nz";
  TLD["ONE"] = "one";
  TLD["ONLINE"] = "online";
  TLD["OOO"] = "ooo";
  TLD["ORG"] = "org";
  TLD["ORG_UK"] = "org.uk";
  TLD["PAGE"] = "page";
  TLD["PARTNERS"] = "partners";
  TLD["PARTS"] = "parts";
  TLD["PARTY"] = "party";
  TLD["PET"] = "pet";
  TLD["PHD"] = "phd";
  TLD["PHOTOGRAPHY"] = "photography";
  TLD["PHOTOS"] = "photos";
  TLD["PICS"] = "pics";
  TLD["PICTURES"] = "pictures";
  TLD["PINK"] = "pink";
  TLD["PIZZA"] = "pizza";
  TLD["PL"] = "pl";
  TLD["PLACE"] = "place";
  TLD["PLUMBING"] = "plumbing";
  TLD["PLUS"] = "plus";
  TLD["POKER"] = "poker";
  TLD["PRESS"] = "press";
  TLD["PRO"] = "pro";
  TLD["PRODUCTIONS"] = "productions";
  TLD["PROF"] = "prof";
  TLD["PROMO"] = "promo";
  TLD["PROPERTIES"] = "properties";
  TLD["PUB"] = "pub";
  TLD["PW"] = "pw";
  TLD["QUEST"] = "quest";
  TLD["RACING"] = "racing";
  TLD["REALTY"] = "realty";
  TLD["RECIPES"] = "recipes";
  TLD["RED"] = "red";
  TLD["REHAB"] = "rehab";
  TLD["REISE"] = "reise";
  TLD["REISEN"] = "reisen";
  TLD["RENT"] = "rent";
  TLD["RENTALS"] = "rentals";
  TLD["REPAIR"] = "repair";
  TLD["REPORT"] = "report";
  TLD["REPUBLICAN"] = "republican";
  TLD["REST"] = "rest";
  TLD["RESTAURANT"] = "restaurant";
  TLD["REVIEW"] = "review";
  TLD["REVIEWS"] = "reviews";
  TLD["RIP"] = "rip";
  TLD["ROCKS"] = "rocks";
  TLD["RODEO"] = "rodeo";
  TLD["RSVP"] = "rsvp";
  TLD["RUN"] = "run";
  TLD["SALE"] = "sale";
  TLD["SALON"] = "salon";
  TLD["SARL"] = "sarl";
  TLD["SCHOOL"] = "school";
  TLD["SCHULE"] = "schule";
  TLD["SCIENCE"] = "science";
  TLD["SERVICES"] = "services";
  TLD["SHOES"] = "shoes";
  TLD["SHOP"] = "shop";
  TLD["SHOPPING"] = "shopping";
  TLD["SHOW"] = "show";
  TLD["SINGLES"] = "singles";
  TLD["SITE"] = "site";
  TLD["SKI"] = "ski";
  TLD["SKIN"] = "skin";
  TLD["SOCCER"] = "soccer";
  TLD["SOCIAL"] = "social";
  TLD["SOFTWARE"] = "software";
  TLD["SOLAR"] = "solar";
  TLD["SOLUTIONS"] = "solutions";
  TLD["SOY"] = "soy";
  TLD["SPACE"] = "space";
  TLD["STORE"] = "store";
  TLD["STREAM"] = "stream";
  TLD["STUDIO"] = "studio";
  TLD["STUDY"] = "study";
  TLD["STYLE"] = "style";
  TLD["SUPPLIES"] = "supplies";
  TLD["SUPPLY"] = "supply";
  TLD["SUPPORT"] = "support";
  TLD["SURF"] = "surf";
  TLD["SURGERY"] = "surgery";
  TLD["SYSTEMS"] = "systems";
  TLD["TAX"] = "tax";
  TLD["TAXI"] = "taxi";
  TLD["TEAM"] = "team";
  TLD["TECH"] = "tech";
  TLD["TECHNOLOGY"] = "technology";
  TLD["TENNIS"] = "tennis";
  TLD["THEATER"] = "theater";
  TLD["TIENDA"] = "tienda";
  TLD["TIPS"] = "tips";
  TLD["TIRES"] = "tires";
  TLD["TIROL"] = "tirol";
  TLD["TODAY"] = "today";
  TLD["TOKYO"] = "tokyo";
  TLD["TOOLS"] = "tools";
  TLD["TOP"] = "top";
  TLD["TOURS"] = "tours";
  TLD["TOWN"] = "town";
  TLD["TOYS"] = "toys";
  TLD["TRADE"] = "trade";
  TLD["TRAINING"] = "training";
  TLD["TUBE"] = "tube";
  TLD["UK"] = "uk";
  TLD["UNIVERSITY"] = "university";
  TLD["UNO"] = "uno";
  TLD["US"] = "us";
  TLD["VACATIONS"] = "vacations";
  TLD["VEGAS"] = "vegas";
  TLD["VENTURES"] = "ventures";
  TLD["VET"] = "vet";
  TLD["VIAJES"] = "viajes";
  TLD["VIDEO"] = "video";
  TLD["VILLAS"] = "villas";
  TLD["VIN"] = "vin";
  TLD["VIP"] = "vip";
  TLD["VISION"] = "vision";
  TLD["VODKA"] = "vodka";
  TLD["VOTE"] = "vote";
  TLD["VOYAGE"] = "voyage";
  TLD["WATCH"] = "watch";
  TLD["WEBCAM"] = "webcam";
  TLD["WEBSITE"] = "website";
  TLD["WEDDING"] = "wedding";
  TLD["WIKI"] = "wiki";
  TLD["WIN"] = "win";
  TLD["WINE"] = "wine";
  TLD["WORK"] = "work";
  TLD["WORKS"] = "works";
  TLD["WORLD"] = "world";
  TLD["WTF"] = "wtf";
  TLD["XYZ"] = "xyz";
  TLD["YACHTS"] = "yachts";
  TLD["YOGA"] = "yoga";
  TLD["YOKOHAMA"] = "yokohama";
  TLD["ZIP"] = "zip";
  TLD["ZONE"] = "zone";
  TLD["UNSUPPORTED"] = "unsupported";
})(TLD || (TLD = {}));
var _default = TLD;
exports["default"] = _default;
module.exports = exports.default;
