// @sqs/enums/TransferInRegistrar.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;
/* File generated by JavaScriptEnumProcessor */
/* Derived from Java class com.squarespace.v6.business.components.domains.transfer.transfererin.models.TransferInRegistrar */
/* Java class defined in repository squarespace-v6 */
var TransferInRegistrar;
(function (TransferInRegistrar) {
  TransferInRegistrar["ASCIO_TECH"] = "ASCIO_TECH";
  TransferInRegistrar["AUTOMATTIC"] = "AUTOMATTIC";
  TransferInRegistrar["CLOUD_GROUP"] = "CLOUD_GROUP";
  TransferInRegistrar["DOMAIN_ROBOT"] = "DOMAIN_ROBOT";
  TransferInRegistrar["DOMAIN_COM"] = "DOMAIN_COM";
  TransferInRegistrar["DOMAINPEOPLE"] = "DOMAINPEOPLE";
  TransferInRegistrar["DREAMHOST"] = "DREAMHOST";
  TransferInRegistrar["DREAMSCAPE"] = "DREAMSCAPE";
  TransferInRegistrar["EASYSPACE"] = "EASYSPACE";
  TransferInRegistrar["EASYSTREET"] = "EASYSTREET";
  TransferInRegistrar["ENOM"] = "ENOM";
  TransferInRegistrar["EPAG"] = "EPAG";
  TransferInRegistrar["FASTDOMAIN"] = "FASTDOMAIN";
  TransferInRegistrar["GANDI"] = "GANDI";
  TransferInRegistrar["GODADDY"] = "GODADDY";
  TransferInRegistrar["GOOGLE"] = "GOOGLE";
  TransferInRegistrar["HOSTINGER"] = "HOSTINGER";
  TransferInRegistrar["HOSTNET"] = "HOSTNET";
  TransferInRegistrar["IN2NET"] = "IN2NET";
  TransferInRegistrar["KEY_SYSTEMS"] = "KEY_SYSTEMS";
  TransferInRegistrar["LAUNCHPAD"] = "LAUNCHPAD";
  TransferInRegistrar["LIVEDOMAINS"] = "LIVEDOMAINS";
  TransferInRegistrar["MARKMONITOR"] = "MARKMONITOR";
  TransferInRegistrar["MESH_DIGITAL"] = "MESH_DIGITAL";
  TransferInRegistrar["METAREGISTRAR"] = "METAREGISTRAR";
  TransferInRegistrar["NAME"] = "NAME";
  TransferInRegistrar["NAMECHEAP"] = "NAMECHEAP";
  TransferInRegistrar["NAMESILO"] = "NAMESILO";
  TransferInRegistrar["NAMESPRO"] = "NAMESPRO";
  TransferInRegistrar["NETWORK_SOLUTIONS"] = "NETWORK_SOLUTIONS";
  TransferInRegistrar["ONEANDONE"] = "ONEANDONE";
  TransferInRegistrar["ONE_API"] = "ONE_API";
  TransferInRegistrar["ONE_TWO_THREE_REG"] = "ONE_TWO_THREE_REG";
  TransferInRegistrar["ONE_COM"] = "ONE_COM";
  TransferInRegistrar["OPEN_TLD"] = "OPEN_TLD";
  TransferInRegistrar["PARAGON_INTERNET"] = "PARAGON_INTERNET";
  TransferInRegistrar["PDR_LTD"] = "PDR_LTD";
  TransferInRegistrar["PORKBUN"] = "PORKBUN";
  TransferInRegistrar["REALTIME_REGISTER"] = "REALTIME_REGISTER";
  TransferInRegistrar["REBEL_CA"] = "REBEL_CA";
  TransferInRegistrar["REGISTER"] = "REGISTER";
  TransferInRegistrar["SQUARESPACE_DOMAINS"] = "SQUARESPACE_DOMAINS";
  TransferInRegistrar["TUCOWS"] = "TUCOWS";
  TransferInRegistrar["WHC_ONLINE"] = "WHC_ONLINE";
  TransferInRegistrar["WILD_WEST"] = "WILD_WEST";
  TransferInRegistrar["UNKNOWN"] = "UNKNOWN";
})(TransferInRegistrar || (TransferInRegistrar = {}));
var _default = TransferInRegistrar;
exports["default"] = _default;
module.exports = exports.default;
