// @sqs/enums/Tutorials.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;
/* File generated by JavaScriptEnumProcessor */
/* Derived from Java class com.squarespace.v6.utils.enums.Tutorials */
/* Java class defined in repository squarespace-v6 */
var Tutorials;
(function (Tutorials) {
  Tutorials["ANALYTICS_APP_BADGE_SHOWN"] = "analytics-app-badge-shown";
  Tutorials["ANALYTICS_UPGRADE_ABANDONED_CART"] = "analytics-upgrade-abandoned-cart";
  Tutorials["ANALYTICS_UPGRADE_CONVERSIONS"] = "analytics-upgrade-conversions";
  Tutorials["ANALYTICS_UPGRADE_PURCHASE_FUNNEL"] = "analytics-upgrade-purchase-funnel";
  Tutorials["ANALYTICS_UPGRADE_SALES_BY_PRODUCT"] = "analytics-upgrade-sales-by-product";
  Tutorials["ANALYTICS_UPGRADE_TRAFFIC_SOURCES"] = "analytics-upgrade-traffic-sources";
  Tutorials["ASSISTANT_BUSINESS_GUIDANCE"] = "assistant-business-guidance-card-in-nav";
  Tutorials["BACKGROUND_ART_ONBOARDING"] = "background_art_onboarding";
  Tutorials["BLOG_MONETIZATION_CARD_DISMISSED"] = "blog_monetization_card_dismissed";
  Tutorials["BUTTONS_UPDATE"] = "buttons_update";
  Tutorials["CAMPAIGNS_BLOG_TO_EMAIL_MODAL"] = "CAMPAIGNS_BLOG_TO_EMAIL_MODAL";
  Tutorials["COMMERCE_APP_REDIRECT_INVENTORY_PANEL"] = "commerce-app-redirect-inventory-panel";
  Tutorials["COMMERCE_APP_REDIRECT_ORDERS_PANEL"] = "commerce-app-redirect-orders-panel";
  Tutorials["COURSE_SALES_PAGE_SETUP"] = "courses-sales-page-setup";
  Tutorials["COURSE_FIND_EXPERT_CARD_DISMISSED"] = "course_find_expert_card_dismissed";
  Tutorials["CONTENT_BLOCKERS_SURVEY"] = "CONTENT_BLOCKERS_SURVEY";
  Tutorials["DEVICE_VIEW"] = "deviceview";
  Tutorials["DIGITAL_PRODUCTS_SALES_PAGE_SETUP"] = "digital-products-sales-page-setup";
  Tutorials["ENGAGING_CONTENT_ONBOARDING"] = "analytics-engaging-content-onboarding";
  Tutorials["EXPLICIT_THUMBNAILS_TUTORIAL_KEY"] = "explicit_thumbnails_tutorial_2021";
  Tutorials["FLUID_ENGINE_MOBILE_TUTORIAL"] = "has-seen-fluid-engine-mobile-changes-notification";
  Tutorials["FLUID_ENGINE_SAVED_SECTIONS"] = "fe-saved-sections";
  Tutorials["FLUID_ENGINE_TUTORIAL_ID"] = "fe-onboarding-wizard";
  Tutorials["FLYOUT_DISCOVERABILITY_AB_TEST"] = "FLYOUT_DISCOVERABILITY_AB_TEST";
  Tutorials["FONTS_PANEL_SUBPANEL_TOAST"] = "fonts-panel-subpanel-toast";
  Tutorials["FORM_IMPROVEMENTS_TUTORIAL_NAME"] = "form-improvements-promotion-banner";
  Tutorials["GENERAL_ONBOARDING"] = "analytics-general-onboarding";
  Tutorials["GLOBAL_SETTINGS_CONTEXTUAL_TOOLTIP"] = "global-settings-contextual-tooltip";
  Tutorials["GLOBAL_SETTINGS_PANEL_TOOLTIP"] = "global-settings-panel-tooltip";
  Tutorials["IN_EDITOR_SURVEY"] = "in_editor_survey";
  Tutorials["IN_PRODUCT_SURVEY"] = "IN_PRODUCT_SURVEY";
  Tutorials["INTRODUCING_SITE_STYLES"] = "tutorial-introducing-site-styles";
  Tutorials["JANUARY_2021_DATA_DELAY"] = "analytics-january-2021-data-delay";
  Tutorials["MOBILE_PREVIEW_EDITING_WARNING_TUTORIAL_ID"] = "mobile-preview-editing-warning";
  Tutorials["PRINT_ON_DEMAND_TUTORIAL"] = "analytics-print-on-demand-tutorial";
  Tutorials["SHIPPING_LABELS_TUTORIAL_NAME"] = "commerce_shipping_labels_order_panel";
  Tutorials["SITE_CONTENT_NEW_ENGAGEMENT"] = "analytics-site-content-new-engagement";
  Tutorials["STYLE_TARGETING_MODAL_ID"] = "style-targeting-modal";
  Tutorials["TEMPLATE_SWITCHING_TEST"] = "template-switching-test";
  Tutorials["TEXT_BLOCK_IMPROVEMENTS"] = "text-block-improvements";
  Tutorials["TUTORIAL_ADD_PREMIUM_BLOCK"] = "tutorial-add-premium-block";
  Tutorials["TUTORIAL_ADD_BLOCK_BUTTON"] = "layout-add-block-button";
  Tutorials["TUTORIAL_LAYOUT_ENGINE"] = "le2-welcome";
  Tutorials["TUTORIAL_ID_CHANGE_STYLES"] = "tutorial-change-styles";
  Tutorials["TUTORIAL_ID_REFRESHED_PRESETS"] = "tutorial-font-and-color-presets-refresh-2022H1";
  Tutorials["WELCOME_WIZARD_TUTORIAL_NAME"] = "commerce_customer_notifications_welcome_wizard";
  Tutorials["WELCOME_MESSAGE"] = "image-editor-1-welcome-message";
  Tutorials["DP_ADDONS_UPSELL_BANNER_DISMISSED"] = "dp-addons-upsell-banner-dismissed";
  Tutorials["DP_SELLING_DASHBOARD_BANNER_DISMISSED"] = "dp-selling-dashboard-banner-dismissed";
  Tutorials["DP_DASHBOARD_MATURE_STATE"] = "dp-dashboard-mature-state";
  Tutorials["DP_DASHBOARD_SETUP_STATE"] = "dp-dashboard-setup-state";
})(Tutorials || (Tutorials = {}));
var _default = Tutorials;
exports["default"] = _default;
module.exports = exports.default;
