// @sqs/enums/USStates.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;
/* File generated by JavaScriptEnumProcessor */
/* Derived from Java class com.squarespace.v6.utils.enums.USState */
/* Java class defined in repository squarespace-v6 */
var USStates;
(function (USStates) {
  USStates["AL"] = "AL";
  USStates["AK"] = "AK";
  USStates["AZ"] = "AZ";
  USStates["AR"] = "AR";
  USStates["CA"] = "CA";
  USStates["CO"] = "CO";
  USStates["CT"] = "CT";
  USStates["DE"] = "DE";
  USStates["FL"] = "FL";
  USStates["GA"] = "GA";
  USStates["HI"] = "HI";
  USStates["ID"] = "ID";
  USStates["IL"] = "IL";
  USStates["IN"] = "IN";
  USStates["IA"] = "IA";
  USStates["KS"] = "KS";
  USStates["KY"] = "KY";
  USStates["LA"] = "LA";
  USStates["ME"] = "ME";
  USStates["MD"] = "MD";
  USStates["MA"] = "MA";
  USStates["MI"] = "MI";
  USStates["MN"] = "MN";
  USStates["MS"] = "MS";
  USStates["MO"] = "MO";
  USStates["MT"] = "MT";
  USStates["NE"] = "NE";
  USStates["NV"] = "NV";
  USStates["NH"] = "NH";
  USStates["NJ"] = "NJ";
  USStates["NM"] = "NM";
  USStates["NY"] = "NY";
  USStates["NC"] = "NC";
  USStates["ND"] = "ND";
  USStates["OH"] = "OH";
  USStates["OK"] = "OK";
  USStates["OR"] = "OR";
  USStates["PA"] = "PA";
  USStates["RI"] = "RI";
  USStates["SC"] = "SC";
  USStates["SD"] = "SD";
  USStates["TN"] = "TN";
  USStates["TX"] = "TX";
  USStates["UT"] = "UT";
  USStates["VT"] = "VT";
  USStates["VA"] = "VA";
  USStates["WA"] = "WA";
  USStates["DC"] = "DC";
  USStates["WV"] = "WV";
  USStates["WI"] = "WI";
  USStates["WY"] = "WY";
  USStates["AA"] = "AA";
  USStates["AE"] = "AE";
  USStates["AP"] = "AP";
})(USStates || (USStates = {}));
var _default = USStates;
exports["default"] = _default;
module.exports = exports.default;
