// @sqs/track-events/lib/v0/index.js
import _objectWithoutProperties from "@babel/runtime/helpers/objectWithoutProperties";
import _defineProperty from "@babel/runtime/helpers/defineProperty";
import _classCallCheck from "@babel/runtime/helpers/classCallCheck";
import _createClass from "@babel/runtime/helpers/createClass";
import _possibleConstructorReturn from "@babel/runtime/helpers/possibleConstructorReturn";
import _getPrototypeOf from "@babel/runtime/helpers/getPrototypeOf";
import _inherits from "@babel/runtime/helpers/inherits";
var _excluded = ["websiteId", "memberAccountId", "website_locale", "member_locale", "templateId", "variantId"];
function ownKeys(e, r) { var t = Object.keys(e); if (Object.getOwnPropertySymbols) { var o = Object.getOwnPropertySymbols(e); r && (o = o.filter(function (r) { return Object.getOwnPropertyDescriptor(e, r).enumerable; })), t.push.apply(t, o); } return t; }
function _objectSpread(e) { for (var r = 1; r < arguments.length; r++) { var t = null != arguments[r] ? arguments[r] : {}; r % 2 ? ownKeys(Object(t), !0).forEach(function (r) { _defineProperty(e, r, t[r]); }) : Object.getOwnPropertyDescriptors ? Object.defineProperties(e, Object.getOwnPropertyDescriptors(t)) : ownKeys(Object(t)).forEach(function (r) { Object.defineProperty(e, r, Object.getOwnPropertyDescriptor(t, r)); }); } return e; }
function _callSuper(t, o, e) { return o = _getPrototypeOf(o), _possibleConstructorReturn(t, _isNativeReflectConstruct() ? Reflect.construct(o, e || [], _getPrototypeOf(t).constructor) : o.apply(t, e)); }
function _isNativeReflectConstruct() { try { var t = !Boolean.prototype.valueOf.call(Reflect.construct(Boolean, [], function () {})); } catch (t) {} return (_isNativeReflectConstruct = function _isNativeReflectConstruct() { return !!t; })(); }
import BaseClient from '../base';
import cookie from '@sqs/cookie-cutter';
import qs from 'querystring';
import network from '@sqs/network';
import { getDefaultConfig } from './constants/defaultConfig';

/**
 * EventsClient V0
 * A simple wrapper for sending events to the v0 Squarespace events pipeline
 */
var EventsClient = /*#__PURE__*/function (_BaseClient) {
  function EventsClient(config) {
    var _this;
    var defaultPayload = arguments.length > 1 && arguments[1] !== undefined ? arguments[1] : {};
    _classCallCheck(this, EventsClient);
    _this = _callSuper(this, EventsClient);
    var defaultConfig = getDefaultConfig();
    _this._config = _objectSpread(_objectSpread({
      expirationKey: undefined,
      storageKey: undefined
    }, defaultConfig), config);
    _this._defaultPayload = defaultPayload;
    return _this;
  }

  /**
   * Generate the event payload
   * @param {Object} event - The object with the event data
   * @returns {Object} - The fully constructed event payload
   */
  _inherits(EventsClient, _BaseClient);
  return _createClass(EventsClient, [{
    key: "_generatePayload",
    value: function _generatePayload(event) {
      var hash = window.location.hash;
      var mergedDefaultAndEventData = _objectSpread(_objectSpread({}, this._defaultPayload), event);
      var _mergedDefaultAndEven = mergedDefaultAndEventData.websiteId,
        websiteId = _mergedDefaultAndEven === void 0 ? null : _mergedDefaultAndEven,
        _mergedDefaultAndEven2 = mergedDefaultAndEventData.memberAccountId,
        memberAccountId = _mergedDefaultAndEven2 === void 0 ? null : _mergedDefaultAndEven2,
        _mergedDefaultAndEven3 = mergedDefaultAndEventData.website_locale,
        website_locale = _mergedDefaultAndEven3 === void 0 ? null : _mergedDefaultAndEven3,
        _mergedDefaultAndEven4 = mergedDefaultAndEventData.member_locale,
        member_locale = _mergedDefaultAndEven4 === void 0 ? null : _mergedDefaultAndEven4,
        _mergedDefaultAndEven5 = mergedDefaultAndEventData.templateId,
        templateId = _mergedDefaultAndEven5 === void 0 ? null : _mergedDefaultAndEven5,
        _mergedDefaultAndEven6 = mergedDefaultAndEventData.variantId,
        variantId = _mergedDefaultAndEven6 === void 0 ? null : _mergedDefaultAndEven6,
        rest = _objectWithoutProperties(mergedDefaultAndEventData, _excluded);
      var payload = _objectSpread(_objectSpread(_objectSpread({}, rest), hash && {
        pageHash: hash
      }), {}, {
        pagePath: this._getNormalizedPathname(),
        websiteId: websiteId,
        templateId: templateId,
        variantId: variantId,
        memberAccountId: memberAccountId,
        member_locale: member_locale,
        website_locale: website_locale
      });

      // log the payload
      if (this._config.logging) {
        this._log({
          payload: _objectSpread({}, payload)
        });
      }
      return payload;
    }

    /**
     * Format an event for analytics and send to their endpoint
     * @param {Object} event - The object with the event data
     * @returns {Promise} - Resolved when request is complete
     */
    // @ts-expect-error
  }, {
    key: "track",
    value: function track(eventName) {
      var event = arguments.length > 1 && arguments[1] !== undefined ? arguments[1] : {};
      var optionalTopLevelArgs = arguments.length > 2 && arguments[2] !== undefined ? arguments[2] : {};
      if (this._config.fireEvents === false) {
        return Promise.resolve();
      }
      var payload = this._generatePayload(event);
      return this._sendXhr({
        eventName: eventName,
        payload: payload
      }, optionalTopLevelArgs);
    }

    /**
     * Send an XHR to the analytics endpoint
     * @param {Object} payload - The event payload to send
     * @param {string} url - Optional: The URL to send event to, used for v2 validation endpoint
     * @returns {Promise} - Resolves when the request was completed successfully
     */
  }, {
    key: "_sendXhr",
    value: function _sendXhr(_ref, optionalTopLevelArgs, url) {
      var eventName = _ref.eventName,
        payload = _ref.payload;
      url = url || this._config.url;
      return network.post(url, qs.stringify(_objectSpread(_objectSpread({
        crumb: cookie.get('crumb'),
        event: eventName
      }, optionalTopLevelArgs), {}, {
        data: JSON.stringify(payload)
      })), {
        headers: {
          'Content-Type': 'application/x-www-form-urlencoded'
        }
      });
    }

    /**
     * Spawn a new tracker that extends the configuration and default payload of originating instance
     * @param {Object} defaults - The additional default payload to be sent by the new tracker
     * @returns {EventsClient} - The newly spawned tracker
     */
  }, {
    key: "spawnTracker",
    value: function spawnTracker() {
      var defaults = arguments.length > 0 && arguments[0] !== undefined ? arguments[0] : {};
      return new EventsClient(this._config, _objectSpread(_objectSpread({}, this._defaultPayload), defaults));
    }
  }]);
}(BaseClient);
export { EventsClient as default };
