#  Ensure an i18n translation `t()` method or `<T>` component includes notes

Ensure an i18n translation `t()` method or `<T>` component includes notes. This is to help give translators more context about the string they are translating. Adding notes to the translations is also something that (as of 2024) has been requested by the #localization team on all new and updated translations.

To use the rule, you'll want to turn this rule on in your eslint config.

> WARNING: Adding a note to an existing translation used in production (that did not have a note attribute on it before) will change the hash of the translation, and kick off a new extraction/translation for that string.This means the string will be untranslated in production. Be careful, and remember you will need to first ["stage/dark translate"](https://i18n-docs.squarespace.net/docs/translation-workflows/translation-workflow-faqs#adding-new-frontend-translations) the translation (with a new note) before you can swap it out for the translation that does not have a note.

```
  "@sqs/i18n-no-missing-notes": "error"
```

## Rule Details

Examples of **correct** code for this rule, with the above options:

```js
  <T project="my.project" notes="This is the name of a town in Texas">Hellodis</T>

  t('Hellodis.', null, { project: 'my.project', notes: 'This is the name of a town in Texas' })
```

Examples of **incorrect** code for this rule, with the above options:

```js
  <T project="my.project" notes="">Hellodis</T>

  <T project="my.project" >Hellodis</T>

  t('Hellodis.', null, { project: 'my.project', notes: '' })

  t('Hellodis.', null, { project: 'my.project' })
```


## When Not To Use It

If you do not care about `t()` strings or `T` components including notes then you do not need this rule.
