# Make sure the project attribute is right in `t()` strings and `<T>` components (i18n-valid-project)

Campaigns found that translations strings and JSX components could have the wrong "project" attribute, and it would be hard to catch in a PR because most people don't memorize the projects in the `translations` directory. This rule allows you to specify "project" names for your translation strings `t()` and components `<T>`, so that you never have to manually check again. 

To use the rule, you'll want to put the following in your `.eslintrc`. Notice that the "allow" array should have the correct project strings in your repo. For Campaigns, they are these 3: strings.campaigns.web, strings.campaigns.unsubscribe, imageEditor.ui.

```
  "@sqs/i18n-valid-project": [
    "error",
    {
      "allow": [
        "strings.campaigns.web",
        "strings.campaigns.unsubscribe",
        "imageEditor.ui"
      ]
    }
  ],
```

## Rule Details

Examples of **correct** code for this rule, with the above options:

```js
  <T project="strings.campaigns.web">Add</T>

  t('Something went wrong when sending your test email.', null, {
    project: 'strings.campaigns.web',
  }),
```

Examples of **incorrect** code for this rule, with the above options:

```js
  <T project="fjeiaofjeioafjierafioe">Add</T>

  t('Something went wrong when sending your test email.', null, {
    project: '123notrealproject',
  }),
  ```

### Options

Should be an object with an "allow" key and value is an array of strings. See above for example .eslintrc config.

## When Not To Use It

If you do not care about `t()` strings or `T` components then you do not need this plugin. 