# Disallow Locale Specific Help Center Article Link. (no-locale-in-help-center-article-link)

Squarepsace Help Center article links (e.g., https://support.squarespace.com/hc/articles/206756327) should be locale agnostic, and should not include a locale in the URL when linking to one of the articles from within another application. 

When a user clicks on a link to the Squarespace Help Center, the Help Center application will determine the correct locale to use. If a locale is specified in the URL, then that locale will be used (we want to avoid this and let the Help Center handle determining the locale).

It is _not_ fixable (via `--fix`) command.

## Rule Details

Examples of **incorrect** code for this rule:

```js
const myUrl = "https://support.squarespace.com/hc/en-us/articles/206756327";
const myOtherUrl = "/api/auth/sso/help?path=/hc/boop/articles/123"
```

Examples of **correct** code for this rule:

```js
// en-us is removed from url
const myUrl = "https://support.squarespace.com/hc/articles/206756327";
const myOtherUrl = "/api/auth/sso/help?path=/hc/articles/123"
```

### Options

This can be set to either `"warn"` or `"error"`

## When Not To Use It

If you do not care if a locale is specific in the Help Center url then don't use this rule.
