# Prefer use of the apostrophe symbol when using contractions or possessives in translated strings. (prefer-apostrophe-in-string-literal)

In March of 2021 the Rolodex team decided to use the apostrophe symbol (not to be confused with the single quote character) in contraction or possesive words.

This lint rule checks to see if a string literal that uses a single quote should actually be using an apostrophe.

It does not care about string literals used in `it` blocks, `describe` blocks, `throw` statements, or comments.

It is _not_ fixable (via `--fix`) command.
## Rule Details

Examples of **incorrect** code for this rule:

```js

myFunc("Let's go to the movies");

myOtherFunc('Let\'s go to the movies');

const myVariable = "We'll go to the movies";

```

Examples of **correct** code for this rule:

```js

myFunc("Let’s go to the movies");

const myVariable = "We’ll go to the movies";

// does not check function expressions that are part of a throw or thrown error
throw new Error("Can't reload");
throw "Can't reload";

// does not check function expressions that are named 'it'
it("doesn't check it blocks")

// does not check function expressions that are named 'describe'
describe("doesn't check describe blocks")

// does not error/warn when it is a single quote used to denote a quote/string
const quote = 'These single quotes are not used in a contraction or possessive';
```

### Options

This can be set to either `"warn"` or `"error"`

## When Not To Use It

If you do not care if a single quote is used where an apostrophe should be used, then don't use this rule.
