// @sqs/febs-build-config/lib/index.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.withManifest = exports.withImages = exports.withPublic = exports.withTypeScript = exports.withLess = exports.withI18n = exports.baseWebpackConfig = exports.getWebpackConfig = exports.rewriteFileTemplate = exports.compose = void 0;
var build_utils_1 = require("./build-utils");
Object.defineProperty(exports, "compose", { enumerable: true, get: function () { return build_utils_1.compose; } });
Object.defineProperty(exports, "rewriteFileTemplate", { enumerable: true, get: function () { return build_utils_1.rewriteFileTemplate; } });
__exportStar(require("./constants"), exports);
__exportStar(require("./i18n-utils"), exports);
var webpack_config_1 = require("./webpack.config");
Object.defineProperty(exports, "getWebpackConfig", { enumerable: true, get: function () { return __importDefault(webpack_config_1).default; } });
var webpack_base_config_1 = require("./webpack.base.config");
Object.defineProperty(exports, "baseWebpackConfig", { enumerable: true, get: function () { return __importDefault(webpack_base_config_1).default; } });
var withI18n_1 = require("./withI18n");
Object.defineProperty(exports, "withI18n", { enumerable: true, get: function () { return __importDefault(withI18n_1).default; } });
var withLess_1 = require("./withLess");
Object.defineProperty(exports, "withLess", { enumerable: true, get: function () { return __importDefault(withLess_1).default; } });
var withTypeScript_1 = require("./withTypeScript");
Object.defineProperty(exports, "withTypeScript", { enumerable: true, get: function () { return __importDefault(withTypeScript_1).default; } });
var withPublic_1 = require("./withPublic");
Object.defineProperty(exports, "withPublic", { enumerable: true, get: function () { return __importDefault(withPublic_1).default; } });
var withImages_1 = require("./withImages");
Object.defineProperty(exports, "withImages", { enumerable: true, get: function () { return __importDefault(withImages_1).default; } });
var withManifest_1 = require("./withManifest");
Object.defineProperty(exports, "withManifest", { enumerable: true, get: function () { return __importDefault(withManifest_1).default; } });
