// @sqs/febs-build-config/lib/webpack-utils.js
'use strict';
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.requireForkTsCheckerWebpackPlugin = exports.getHelpers = void 0;
const febs_logger_1 = __importDefault(require("@sqs/febs-logger"));
const helperFactory_1 = require("./helperFactory");
const prodValues = ['prod', 'production'];
const devValues = ['dev', 'development'];
const testValues = ['test', 'testing'];
const parallelValues = ['parallel'];
const envValues = {
    dev: devValues,
    parallel: parallelValues,
    prod: prodValues,
    test: testValues,
};
exports.getHelpers = helperFactory_1.getHelperFactory(envValues);
const requireForkTsCheckerWebpackPlugin = () => {
    try {
        return require('fork-ts-checker-webpack-plugin');
    }
    catch (error) {
        febs_logger_1.default.error('ForkTsCheckerWebpackPlugin failed to install, likely because typescript is not installed in this project.');
        throw error;
    }
};
exports.requireForkTsCheckerWebpackPlugin = requireForkTsCheckerWebpackPlugin;
