// @sqs/febs-build-config/lib/withPublic.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
const html_webpack_plugin_1 = __importDefault(require("html-webpack-plugin"));
const path = __importStar(require("path"));
const webpack_merge_1 = __importDefault(require("webpack-merge"));
exports.default = (argv) => (webpackConfig) => {
    const appPaths = argv.paths;
    const appIndex = path.join(appPaths.public, './index.html');
    return webpack_merge_1.default(webpackConfig, {
        module: {
            rules: [
                {
                    test: /\.ico/,
                    type: 'asset',
                    parser: {
                        dataUrlCondition: {
                            maxSize: 1024
                        }
                    },
                    generator: {
                        filename: '[name]-[hash][ext]'
                    },
                },
            ],
        },
        plugins: [
            new html_webpack_plugin_1.default({
                template: appIndex,
            }),
        ],
    });
};
