// @sqs/febs-build-config/lib/withImages.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
const webpack_merge_1 = __importDefault(require("webpack-merge"));
const withImages = (options = {}) => (webpackConfig) => {
    const resolver = options.resolver || require.resolve;
    const inline = options.inline !== undefined ? options.inline : false;
    const updateWebpackConfig = webpack_merge_1.default(webpackConfig, {
        module: {
            rules: [
                {
                    test: /\.svg$/,
                    use: [
                        {
                            loader: resolver('svg-url-loader'),
                            options: {
                                iesafe: true,
                                limit: inline ? undefined : 1024,
                                noquotes: true,
                            },
                        },
                        {
                            loader: resolver('./svg-o-loader'),
                        },
                    ],
                },
                {
                    test: /\.(png|jpe?g|gif)/,
                    type: 'asset',
                    parser: {
                        dataUrlCondition: {
                            maxSize: 1024
                        }
                    },
                    generator: {
                        filename: '[name]-[hash][ext]'
                    },
                },
            ],
        },
    });
    return updateWebpackConfig;
};
exports.default = withImages;
