// @phensley/cldr-core/lib/systems/calendars/interval.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.timePeriodFieldFlags = exports.TIME_FLAGS = exports.TIME_PERIOD_FIELDS = void 0;
/**
 * @internal
 */
exports.TIME_PERIOD_FIELDS = [
    'year',
    'month',
    'week',
    'day',
    'hour',
    'minute',
    'second',
    'millis',
];
/**
 * @internal
 */
exports.TIME_FLAGS = 16 /* TimePeriodFieldFlag.HOUR */ | 32 /* TimePeriodFieldFlag.MINUTE */ | 64 /* TimePeriodFieldFlag.SECOND */ | 128 /* TimePeriodFieldFlag.MILLIS */;
var FIELDMAP = {
    year: 1 /* TimePeriodFieldFlag.YEAR */,
    month: 2 /* TimePeriodFieldFlag.MONTH */,
    week: 4 /* TimePeriodFieldFlag.WEEK */,
    day: 8 /* TimePeriodFieldFlag.DAY */,
    hour: 16 /* TimePeriodFieldFlag.HOUR */,
    minute: 32 /* TimePeriodFieldFlag.MINUTE */,
    second: 64 /* TimePeriodFieldFlag.SECOND */,
    millis: 128 /* TimePeriodFieldFlag.MILLIS */,
};
/**
 * @internal
 */
var timePeriodFieldFlags = function (fields) {
    return fields.reduce(function (p, c) {
        p |= FIELDMAP[c];
        return p;
    }, 0);
};
exports.timePeriodFieldFlags = timePeriodFieldFlags;
