// @sqs/in-product-promo/dist/api.js
"use strict";
// tslint:disable
/**
 * in-product-promo-service
 * A service for managing the allocation of in-product promo codes
 *
 * The version of the OpenAPI document: 1.0.0
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.PromotionType = exports.PromotionSurfaceArea = exports.PromotionCampaignValidationErrorCode = void 0;
/**
 *
 * @export
 * @enum {string}
 */
var PromotionCampaignValidationErrorCode;
(function (PromotionCampaignValidationErrorCode) {
    PromotionCampaignValidationErrorCode["CAMPAIGN_ID"] = "INVALID_CAMPAIGN_ID";
    PromotionCampaignValidationErrorCode["CAMPAIGN_NAME_LENGTH"] = "INVALID_CAMPAIGN_NAME_LENGTH";
    PromotionCampaignValidationErrorCode["MEMBER_ACCOUNT_IDS_FILE"] = "INVALID_MEMBER_ACCOUNT_IDS_FILE";
    PromotionCampaignValidationErrorCode["CAMPAIGN_TIMEFRAME"] = "INVALID_CAMPAIGN_TIMEFRAME";
    PromotionCampaignValidationErrorCode["PROMO_CODE_LENGTH"] = "INVALID_PROMO_CODE_LENGTH";
    PromotionCampaignValidationErrorCode["PROMO_TYPE"] = "INVALID_PROMO_TYPE";
    PromotionCampaignValidationErrorCode["TERMS_URL"] = "INVALID_TERMS_URL";
})(PromotionCampaignValidationErrorCode = exports.PromotionCampaignValidationErrorCode || (exports.PromotionCampaignValidationErrorCode = {}));
/**
 *
 * @export
 * @enum {string}
 */
var PromotionSurfaceArea;
(function (PromotionSurfaceArea) {
    PromotionSurfaceArea["CMS_TRIAL_BANNER"] = "CMS_TRIAL_BANNER";
})(PromotionSurfaceArea = exports.PromotionSurfaceArea || (exports.PromotionSurfaceArea = {}));
/**
 *
 * @export
 * @enum {string}
 */
var PromotionType;
(function (PromotionType) {
    PromotionType["TEN_PERCENT_OFF"] = "TEN_PERCENT_OFF";
    PromotionType["FIFTEEN_PERCENT_OFF"] = "FIFTEEN_PERCENT_OFF";
    PromotionType["TWENTY_PERCENT_OFF"] = "TWENTY_PERCENT_OFF";
    PromotionType["TWENTY_FIVE_PERCENT_OFF"] = "TWENTY_FIVE_PERCENT_OFF";
    PromotionType["THIRTY_PERCENT_OFF"] = "THIRTY_PERCENT_OFF";
    PromotionType["THIRTY_FIVE_PERCENT_OFF"] = "THIRTY_FIVE_PERCENT_OFF";
    PromotionType["FORTY_PERCENT_OFF"] = "FORTY_PERCENT_OFF";
    PromotionType["FORTY_FIVE_PERCENT_OFF"] = "FORTY_FIVE_PERCENT_OFF";
    PromotionType["FIFTY_PERCENT_OFF"] = "FIFTY_PERCENT_OFF";
})(PromotionType = exports.PromotionType || (exports.PromotionType = {}));
