// @sqs/eslint-config-core/rules/react.js
module.exports = {
  rules: {
    'react/boolean-prop-naming': 'off',
    'react/button-has-type': ['off'],
    'react/default-props-match-prop-types': 'off',
    'react/display-name': 'off',
    'react/destructuring-assignment': ['off'],
    'react/forbid-component-props': 'off',
    'react/forbid-dom-props': 'off',
    'react/forbid-elements': 'off',
    'react/forbid-prop-types': 'off',
    'react/forbid-foreign-prop-types': 'off',
    'react/no-access-state-in-setstate': 'off',
    'react/no-array-index-key': 'warn',
    'react/no-children-prop': 'warn',
    'react/no-danger': 'off',
    'react/no-danger-with-children': 'off',
    'react/no-deprecated': 'off',
    'react/no-did-mount-set-state': 'off',
    'react/no-did-update-set-state': ['warn'],
    'react/no-direct-mutation-state': 'warn',
    'react/no-find-dom-node': 'warn',
    'react/no-is-mounted': 'warn',
    'react/no-multi-comp': ['warn', { ignoreStateless: true }],
    'react/no-redundant-should-component-update': 'warn',
    'react/no-render-return-value': 'warn',
    'react/no-set-state': 'off',
    'react/no-typos': 'off',
    'react/no-string-refs': 'warn',
    'react/no-this-in-sfc': 'warn',
    'react/no-unescaped-entities': 'warn',
    'react/no-unknown-property': ['warn'],
    'react/no-unsafe': 'off',
    'react/no-unused-prop-types': ['off'],
    'react/no-unused-state': 'off',
    'react/no-will-update-set-state': ['off'],
    'react/prefer-es6-class': 'off',
    'react/prefer-read-only-props': 'warn',
    'react/prefer-stateless-function': 'off',
    'react/prop-types': [
      'warn',
      {
        ignore: ['className', 'children', 'style'],
      },
    ],
    'react/react-in-jsx-scope': 'warn',
    'react/require-default-props': 'off',
    'react/require-optimization': 'off',
    'react/require-render-return': 'off',
    'react/self-closing-comp': 'warn',
    'react/sort-comp': [
      'warn',
      {
        order: [
          'displayName',
          'propTypes',
          'contextTypes',
          'childContextTypes',
          'mixins',
          'statics',
          'defaultProps',
          'constructor',
          'getDefaultProps',
          'getInitialState',
          'getChildContext',
          'componentWillMount',
          'componentDidMount',
          'componentWillReceiveProps',
          'shouldComponentUpdate',
          'componentWillUpdate',
          'componentDidUpdate',
          'componentWillUnmount',
          '/^on.+$/',
          '/^handle.+$/',
          '/^get.+$/',
          '/^(?!render).+$/',
          '/^render.+$/',
          'render',
        ],
      },
    ],
    'react/sort-prop-types': 'off',
    'react/state-in-constructor': 'off',
    'react/static-property-placement': 'warn',
    'react/style-prop-object': 'off',
    'react/void-dom-elements-no-children': 'off',

    //JSX-specific rules
    'react/jsx-boolean-value': ['off'],
    'react/jsx-child-element-spacing': 'off',
    'react/jsx-closing-bracket-location': [1, 'tag-aligned'],
    'react/jsx-closing-tag-location': 'warn',
    'react/jsx-curly-brace-presence': ['off'],
    'react/jsx-curly-newline': 'off',
    'react/jsx-curly-spacing': 'off',
    'react/jsx-equals-spacing': ['warn', 'never'],
    'react/jsx-filename-extension': 'off',
    'react/jsx-first-prop-new-line': 'off',
    'react/jsx-handler-names': 'off',
    'react/jsx-indent': 'off',
    'react/jsx-indent-props': [1, 2],
    'react/jsx-key': 'warn',
    'react/jsx-max-depth': ['warn', { max: 9 }],
    'react/jsx-max-props-per-line': 'off',
    'react/jsx-no-bind': 'off',
    'react/jsx-no-comment-textnodes': 'off',
    'react/jsx-no-duplicate-props': ['off'],
    'react/jsx-no-literals': 'off',
    'react/jsx-no-target-blank': 'error',
    'react/jsx-no-undef': 'error',
    'react/jsx-one-expression-per-line': 'off',
    'react/jsx-fragments': 'off',
    'react/jsx-pascal-case': 'warn',
    'react/jsx-props-no-multi-spaces': 'warn',
    'react/jsx-props-no-spreading': 'off',
    'react/sort-default-props': 'warn',
    'react/jsx-sort-props': 'off',
    'react/jsx-space-before-closing': 'off',
    'react/jsx-tag-spacing': 'off',
    'react/jsx-uses-react': 'warn',
    'react/jsx-uses-vars': 'warn',
    'react/jsx-wrap-multilines': [
      'warn',
      {
        declaration: 'parens-new-line',
        assignment: 'parens-new-line',
        return: 'parens-new-line',
        arrow: 'parens-new-line',
        condition: 'parens-new-line',
        logical: 'parens-new-line',
        prop: 'parens-new-line',
      },
    ],

    // react-hooks
    'react-hooks/rules-of-hooks': 'error',
    'react-hooks/exhaustive-deps': 'warn',
  },
};
