// @sqs/i18n-build/lib/ast-utils/index.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.T_FUNCTION_NAME = exports.T_COMPONENT_NAME = exports.TRANSLATION_ID_TAG = exports.PROJECT_TAG = exports.PLURAL_COMPONENT_NUMBER_TAG = exports.PLURAL_COMPONENT_NAME = exports.PLURALIZE_FUNCTION_NAME = exports.PLURALIZATION_KEYS = exports.PARSER_PLUGINS = exports.NOTES_TAG = exports.FORMS_TAG = exports.I18nFunction = exports.I18nComponent = exports.transformPluralizeFunction = exports.transformPluralComponent = exports.transformTFunction = exports.transformTComponent = exports.AutoConfigImporter = exports.TranslationsReplacer = exports.InvalidUsageError = exports.I18nValidate = exports.shouldWrapChildren = void 0;
const tslib_1 = require("tslib");
var detect_1 = require("./find/detect");
Object.defineProperty(exports, "shouldWrapChildren", { enumerable: true, get: function () { return tslib_1.__importDefault(detect_1).default; } });
var validate_1 = require("./validate");
Object.defineProperty(exports, "I18nValidate", { enumerable: true, get: function () { return tslib_1.__importDefault(validate_1).default; } });
var errors_1 = require("./errors");
Object.defineProperty(exports, "InvalidUsageError", { enumerable: true, get: function () { return errors_1.InvalidUsageError; } });
var TranslationsReplacer_1 = require("./visitors/TranslationsReplacer");
Object.defineProperty(exports, "TranslationsReplacer", { enumerable: true, get: function () { return TranslationsReplacer_1.TranslationsReplacer; } });
var AutoConfigImporter_1 = require("./visitors/AutoConfigImporter");
Object.defineProperty(exports, "AutoConfigImporter", { enumerable: true, get: function () { return AutoConfigImporter_1.AutoConfigImporter; } });
var generate_1 = require("./generate");
Object.defineProperty(exports, "transformTComponent", { enumerable: true, get: function () { return generate_1.transformTComponent; } });
Object.defineProperty(exports, "transformTFunction", { enumerable: true, get: function () { return generate_1.transformTFunction; } });
Object.defineProperty(exports, "transformPluralComponent", { enumerable: true, get: function () { return generate_1.transformPluralComponent; } });
Object.defineProperty(exports, "transformPluralizeFunction", { enumerable: true, get: function () { return generate_1.transformPluralizeFunction; } });
exports.I18nComponent = tslib_1.__importStar(require("./parse/components"));
exports.I18nFunction = tslib_1.__importStar(require("./parse/functions"));
const constants_1 = tslib_1.__importDefault(require("./utils/constants"));
const { FORMS_TAG, NOTES_TAG, PARSER_PLUGINS, PLURALIZATION_KEYS, PLURALIZE_FUNCTION_NAME, PLURAL_COMPONENT_NAME, PLURAL_COMPONENT_NUMBER_TAG, PROJECT_TAG, TRANSLATION_ID_TAG, T_COMPONENT_NAME, T_FUNCTION_NAME, } = constants_1.default;
exports.FORMS_TAG = FORMS_TAG;
exports.NOTES_TAG = NOTES_TAG;
exports.PARSER_PLUGINS = PARSER_PLUGINS;
exports.PLURALIZATION_KEYS = PLURALIZATION_KEYS;
exports.PLURALIZE_FUNCTION_NAME = PLURALIZE_FUNCTION_NAME;
exports.PLURAL_COMPONENT_NAME = PLURAL_COMPONENT_NAME;
exports.PLURAL_COMPONENT_NUMBER_TAG = PLURAL_COMPONENT_NUMBER_TAG;
exports.PROJECT_TAG = PROJECT_TAG;
exports.TRANSLATION_ID_TAG = TRANSLATION_ID_TAG;
exports.T_COMPONENT_NAME = T_COMPONENT_NAME;
exports.T_FUNCTION_NAME = T_FUNCTION_NAME;
