// @sqs/i18n-ui/lib/functions/date/customMomentLocales/de.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;
var DAYS_IN_WEEK = 7;
var ZERO_SECONDS_OUTPUT = 'gerade eben';
var _default = exports.default = {
  parentLocale: 'de',
  months: ['Januar', 'Februar', 'März', 'April', 'Mai', 'Juni', 'Juli', 'August', 'September', 'Oktober', 'November', 'Dezember'],
  monthsShort: ['Jan.', 'Feb.', 'März', 'Apr.', 'Mai', 'Juni', 'Juli', 'Aug.', 'Sept.', 'Okt.', 'Nov.', 'Dez.'],
  weekdays: ['Sonntag', 'Montag', 'Dienstag', 'Mittwoch', 'Donnerstag', 'Freitag', 'Samstag'],
  weekdaysShort: ['So.', 'Mo.', 'Di.', 'Mi.', 'Do.', 'Fr.', 'Sa.'],
  longDateFormat: {
    LT: 'H:mm',
    LTS: 'H:mm:ss',
    L: 'D.M.YYYY',
    LLT: 'L [um] LT',
    LLTS: 'L [um] LTS',
    l: 'M.YYYY',
    LL: 'D. MMMM',
    LLdddd: 'dddd, LL',
    LLLT: 'LL [um] LT',
    LLddddLT: 'dddd, LL LT',
    ll: 'D. MMM',
    llddd: 'ddd, ll',
    llLT: 'll [um] LT',
    lldddLT: 'ddd, ll LT',
    LLL: 'LL YYYY',
    LLLLT: 'LLL [um] LT',
    lll: 'll YYYY',
    lllLT: 'lll [um] LT',
    LLLL: 'dddd, LLL',
    llll: 'ddd, lll',
    LLLLLT: 'LLLL [um] LT',
    llllLT: 'llll [um] LT'
  },
  relativeTime: {
    future: function future(number) {
      return number === ZERO_SECONDS_OUTPUT ? number : 'in ' + number;
    },
    past: function past(number) {
      return number === ZERO_SECONDS_OUTPUT ? number : 'vor ' + number;
    },
    s: ZERO_SECONDS_OUTPUT,
    ss: function ss(number) {
      if (number === 1) {
        return '1 Sekunde';
      }
      return number + ' Sekunden';
    },
    m: '1 Minute',
    mm: '%d Minuten',
    h: '1 Stunde',
    hh: '%d Stunden',
    d: '1 Tag',
    dd: function dd(number) {
      if (number < 7) {
        return number + ' Tagen';
      }
      var nbOfWeeks = Math.floor(number / DAYS_IN_WEEK);
      return nbOfWeeks === 1 ? '1 Woche' : nbOfWeeks + ' Wochen';
    },
    M: '1 Monat',
    MM: '%d Monaten',
    y: '1 Jahr',
    yy: '%d Jahren'
  },
  calendar: {
    lastDay: '[Gestern] LT',
    sameDay: '[Heute] LT',
    nextDay: '[Morgen] LT',
    lastWeek: '[Letzter] dddd LT',
    nextWeek: 'dddd LT',
    sameYear: 'll',
    sameElse: 'lll'
  }
};
module.exports = exports.default;
