// @sqs/i18n-ui/lib/functions/date/customMomentLocales/en.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;
var DAYS_IN_WEEK = 7;
var ZERO_SECONDS_OUTPUT = 'just now';
var _default = exports.default = {
  longDateFormat: {
    LT: 'h:mma',
    LTS: 'h:mm:ssa',
    L: 'M/D/YYYY',
    LLT: 'L [at] LT',
    LLTS: 'L [at] LTS',
    l: 'M/YYYY',
    LL: 'MMMM D',
    LLdddd: 'dddd, LL',
    LLLT: 'LL [at] LT',
    LLddddLT: 'dddd, LL [at] LT',
    ll: 'MMM D',
    llddd: 'ddd, ll',
    llLT: 'll [at] LT',
    lldddLT: 'ddd, ll [at] LT',
    LLL: 'LL, YYYY',
    LLLLT: 'LLL [at] LT',
    lll: 'll, YYYY',
    lllLT: 'lll [at] LT',
    LLLL: 'dddd, LLL',
    llll: 'ddd, lll',
    LLLLLT: 'LLLL [at] LT',
    llllLT: 'llll [at] LT'
  },
  relativeTime: {
    future: function future(number) {
      return number === ZERO_SECONDS_OUTPUT ? number : 'in ' + number;
    },
    past: function past(number) {
      return number === ZERO_SECONDS_OUTPUT ? number : number + ' ago';
    },
    s: ZERO_SECONDS_OUTPUT,
    ss: function ss(number) {
      if (number === 1) {
        return 'a second';
      }
      return number + ' seconds';
    },
    m: 'a minute',
    mm: '%d minutes',
    h: 'an hour',
    hh: '%d hours',
    d: 'a day',
    dd: function dd(number) {
      if (number < 7) {
        return number + ' days';
      }
      var nbOfWeeks = Math.floor(number / DAYS_IN_WEEK);
      return nbOfWeeks === 1 ? 'a week' : nbOfWeeks + ' weeks';
    },
    M: 'a month',
    MM: '%d months',
    y: 'a year',
    yy: '%d years'
  },
  meridiem: {
    am: 'am',
    AM: 'AM',
    pm: 'pm',
    PM: 'PM'
  },
  calendar: {
    lastDay: '[Yesterday] LT',
    sameDay: '[Today] LT',
    nextDay: '[Tomorrow] LT',
    lastWeek: '[last] dddd LT',
    sameWeek: 'dddd LT',
    lastYear: 'll',
    sameElse: 'lll'
  }
};
module.exports = exports.default;
