// @sqs/i18n-ui/lib/functions/date/customMomentLocales/es.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;
var DAYS_IN_WEEK = 7;
var ZERO_SECONDS_OUTPUT = 'reciente';
var _default = exports.default = {
  parentLocale: 'es',
  months: ['enero', 'febrero', 'marzo', 'abril', 'mayo', 'junio', 'julio', 'agosto', 'septiembre', 'octubre', 'noviembre', 'diciembre'],
  monthsShort: ['ene.', 'feb.', 'mar.', 'abr.', 'may.', 'jun.', 'jul.', 'ago.', 'set.', 'oct.', 'nov.', 'dic.'],
  weekdays: ['domingo', 'lunes', 'martes', 'miércoles', 'jueves', 'viernes', 'sábado'],
  weekdaysShort: ['do.', 'lu.', 'ma.', 'mi.', 'ju.', 'vi.', 'sa.'],
  longDateFormat: {
    LT: 'H:mm',
    LTS: 'H:mm:ss',
    L: 'D/M/YYYY',
    LLT: 'L LT',
    LLTS: 'L LTS',
    l: 'M/YYYY',
    LL: 'D [de] MMMM',
    LLdddd: 'dddd, LL',
    LLLT: 'LL LT',
    LLddddLT: 'dddd, LL LT',
    ll: 'D [de] MMM',
    llddd: 'ddd, ll',
    llLT: 'll LT',
    lldddLT: 'ddd, ll LT',
    LLL: 'LL [de] YYYY',
    LLLLT: 'LLL LT',
    lll: 'll, YYYY',
    lllLT: 'lll LT',
    LLLL: 'dddd LLL',
    llll: 'ddd, lll',
    LLLLLT: 'LLLL LT',
    llllLT: 'llll LT'
  },
  relativeTime: {
    future: function future(number) {
      return number === ZERO_SECONDS_OUTPUT ? number : 'en ' + number;
    },
    past: function past(number) {
      return number === ZERO_SECONDS_OUTPUT ? number : 'hace ' + number;
    },
    s: ZERO_SECONDS_OUTPUT,
    ss: function ss(number) {
      if (number === 1) {
        return 'un segundo';
      }
      return number + ' segundos';
    },
    m: 'un minuto',
    mm: '%d minutos',
    h: 'una hora',
    hh: '%d horas',
    d: 'un día',
    dd: function dd(number) {
      if (number < 7) {
        return number + ' días';
      }
      var nbOfWeeks = Math.floor(number / DAYS_IN_WEEK);
      return nbOfWeeks === 1 ? 'una semana' : nbOfWeeks + ' semanas';
    },
    M: 'un mes',
    MM: '%d meses',
    y: 'un año',
    yy: '%d años'
  },
  calendar: {
    lastDay: '[Ayer] LT',
    sameDay: '[Hoy] LT',
    nextDay: '[Mañana] LT',
    lastWeek: 'dddd [pasado] LT',
    nextWeek: 'dddd LT',
    sameYear: 'll',
    sameElse: 'lll'
  }
};
module.exports = exports.default;
