// @sqs/i18n-ui/lib/functions/date/customMomentLocales/fr.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;
var DAYS_IN_WEEK = 7;
var ZERO_SECONDS_OUTPUT = "à l'instant";
var _default = exports.default = {
  parentLocale: 'fr',
  months: ['janvier', 'février', 'mars', 'avril', 'mai', 'juin', 'juillet', 'août', 'septembre', 'octobre', 'novembre', 'décembre'],
  monthsShort: ['janv.', 'févr.', 'mars', 'avr.', 'mai', 'juin', 'juill.', 'août', 'sept.', 'oct.', 'nov.', 'déc.'],
  weekdays: ['dimanche', 'lundi', 'mardi', 'mercredi', 'jeudi', 'vendredi', 'samedi'],
  weekdaysShort: ['dim.', 'lun.', 'mar.', 'mer.', 'jeu.', 'ven.', 'sam.'],
  longDateFormat: {
    LT: 'H:mm',
    LTS: 'H:mm:ss',
    L: 'D/M/YYYY',
    LLT: 'L [à] LT',
    LLTS: 'L [à] LTS',
    l: 'M/YYYY',
    LL: 'D MMMM',
    LLdddd: 'dddd LL',
    LLLT: 'LL [à] LT',
    LLddddLT: 'dddd LL LT',
    ll: 'D MMM',
    llddd: 'ddd ll',
    llLT: 'll [à] LT',
    lldddLT: 'ddd ll LT',
    LLL: 'LL YYYY',
    LLLLT: 'LLL [à] LT',
    lll: 'll YYYY',
    lllLT: 'lll [à] LT',
    LLLL: 'dddd, LLL',
    llll: 'ddd, lll',
    LLLLLT: 'LLLL [à] LT',
    llllLT: 'llll [à] LT'
  },
  relativeTime: {
    future: function future(number) {
      return number === ZERO_SECONDS_OUTPUT ? number : 'dans ' + number;
    },
    past: function past(number) {
      return number === ZERO_SECONDS_OUTPUT ? number : 'il y a ' + number;
    },
    s: ZERO_SECONDS_OUTPUT,
    ss: function ss(number) {
      if (number === 1) {
        return 'une seconde';
      }
      return number + ' secondes';
    },
    m: 'une minute',
    mm: '%d minutes',
    h: 'une heure',
    hh: '%d heures',
    d: 'un jour',
    dd: function dd(number) {
      if (number < 7) {
        return number + ' jours';
      }
      var nbOfWeeks = Math.floor(number / DAYS_IN_WEEK);
      return nbOfWeeks === 1 ? 'une semaine' : nbOfWeeks + ' semaines';
    },
    M: 'un mois',
    MM: '%d mois',
    y: 'an',
    yy: '%d ans'
  },
  calendar: {
    lastDay: '[Hier] LT',
    sameDay: "[Aujourd'hui] LT",
    nextDay: '[Demain] LT',
    lastWeek: 'dddd [dernier] LT',
    nextWeek: 'dddd LT',
    sameYear: 'll',
    sameElse: 'lll'
  }
};
module.exports = exports.default;
