// @sqs/i18n-ui/lib/functions/date/customMomentLocales/it.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;
var DAYS_IN_WEEK = 7;
var ZERO_SECONDS_OUTPUT = 'poco fa';
var _default = exports.default = {
  parentLocale: 'it',
  months: ['gennaio', 'febbraio', 'marzo', 'aprile', 'maggio', 'giugno', 'luglio', 'agosto', 'settembre', 'ottobre', 'novembre', 'dicembre'],
  monthsShort: ['gen.', 'feb.', 'mar.', 'apr.', 'mag.', 'giu.', 'lug.', 'ago.', 'set.', 'ott.', 'nov.', 'dic.'],
  weekdays: ['domenica', 'lunedì', 'martedì', 'mercoledì', 'giovedì', 'venerdì', 'sabato'],
  weekdaysShort: ['do.', 'lun.', 'mar.', 'mer.', 'gio.', 'ven.', 'sab.'],
  longDateFormat: {
    LT: 'H:mm',
    LTS: 'H:mm:ss',
    L: 'D/M/YYYY',
    LLT: 'L [alle] LT',
    LLTS: 'L [alle] LTS',
    l: 'M/YYYY',
    LL: 'D MMMM',
    LLdddd: 'dddd, LL',
    LLLT: 'LL [alle] LT',
    LLddddLT: 'dddd, LL LT',
    ll: 'D MMM',
    llddd: 'ddd, ll',
    llLT: 'll [alle] LT',
    lldddLT: 'ddd, ll LT',
    LLL: 'LL YYYY',
    LLLLT: 'LLL [alle] LT',
    lll: 'll YYYY',
    lllLT: 'lll [alle] LT',
    LLLL: 'dddd, LLL',
    llll: 'ddd, lll',
    LLLLLT: 'LLLL [alle] LT',
    llllLT: 'llll [alle] LT'
  },
  relativeTime: {
    future: function future(number) {
      return number === ZERO_SECONDS_OUTPUT ? number : 'in ' + number;
    },
    past: function past(number) {
      return number === ZERO_SECONDS_OUTPUT ? number : number + ' fa';
    },
    s: ZERO_SECONDS_OUTPUT,
    ss: function ss(number) {
      if (number === 1) {
        return 'un secondo';
      }
      return number + ' secondi';
    },
    m: 'un minuto',
    mm: '%d minuti',
    h: 'un ora',
    hh: '%d ore',
    d: 'un giorno',
    dd: function dd(number) {
      if (number < 7) {
        return number + ' giorni';
      }
      var nbOfWeeks = Math.floor(number / DAYS_IN_WEEK);
      return nbOfWeeks === 1 ? 'un settimana' : nbOfWeeks + ' settimane';
    },
    M: 'un mese',
    MM: '%d mesi',
    y: 'un ano',
    yy: '%d anos'
  },
  calendar: {
    lastDay: '[Ieri] LT',
    sameDay: '[Oggi] LT',
    nextDay: '[Domani] LT',
    lastWeek: '[lo scorso] dddd LT',
    nextWeek: 'dddd LT',
    sameYear: 'll',
    sameElse: 'lll'
  }
};
module.exports = exports.default;
