// @sqs/i18n-ui/lib/functions/date/customMomentLocales/pt.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;
require("core-js/modules/es.array.includes.js");
var DAYS_IN_WEEK = 7;
var ZERO_SECONDS_OUTPUT = 'agora mesmo';
var _default = exports.default = {
  parentLocale: 'pt',
  months: ['janeiro', 'fevereiro', 'março', 'abril', 'maio', 'junho', 'julho', 'agosto', 'setembro', 'outubro', 'novembro', 'dezembro'],
  monthsShort: ['jan.', 'fev.', 'mar.', 'abr.', 'maio', 'jun.', 'jul.', 'ago.', 'set.', 'out.', 'nov.', 'dez.'],
  weekdays: ['domingo', 'segunda-feira', 'terça-feira', 'quarta-feira', 'quinta-feira', 'sexta-feira', 'sábado'],
  weekdaysShort: ['dom.', 'seg.', 'ter.', 'qua.', 'qui.', 'sex.', 'sab.'],
  longDateFormat: {
    LT: 'H:mm',
    LTS: 'H:mm:ss',
    L: 'D/M/YYYY',
    LLT: 'L [às] LT',
    LLTS: 'L [às] LTS',
    l: 'M/YYYY',
    LL: 'D [de] MMMM',
    LLdddd: 'dddd, LL',
    LLLT: 'LL [às] LT',
    LLddddLT: 'dddd, LL LT',
    ll: 'D [de] MMM',
    llddd: 'ddd, ll',
    llLT: 'll [às] LT',
    lldddLT: 'ddd, ll LT',
    LLL: 'LL [de] YYYY',
    LLLLT: 'LLL [às] LT',
    lll: 'll YYYY',
    lllLT: 'lll [às] LT',
    LLLL: 'dddd, LLL',
    llll: 'ddd, lll',
    LLLLLT: 'LLLL [às] LT',
    llllLT: 'llll [às] LT'
  },
  relativeTime: {
    future: function future(number) {
      return number === ZERO_SECONDS_OUTPUT ? number : 'em ' + number;
    },
    past: function past(number) {
      return number === ZERO_SECONDS_OUTPUT ? number : number + ' atras';
    },
    s: ZERO_SECONDS_OUTPUT,
    ss: function ss(number) {
      if (number === 1) {
        return 'um segundo';
      }
      return number + ' segundos';
    },
    m: 'um minuto',
    mm: '%d minutos',
    h: 'uma hora',
    hh: '%d horas',
    d: 'um dia',
    dd: function dd(number) {
      if (number < 7) {
        return number + ' dias';
      }
      var nbOfWeeks = Math.floor(number / DAYS_IN_WEEK);
      return nbOfWeeks === 1 ? 'uma semana' : nbOfWeeks + ' semanas';
    },
    M: 'um mês',
    MM: '%d meses',
    y: 'um ano',
    yy: '%d anos'
  },
  calendar: {
    lastDay: '[Ontem] LT',
    sameDay: '[Hoje] LT',
    nextDay: '[Amanhã] LT',
    lastWeek: function lastWeek() {
      return '[' + ([0, 6].includes(this.day()) ? 'último' : 'última') + '] dddd LT';
    },
    nextWeek: 'dddd LT',
    sameYear: 'll',
    sameElse: 'lll'
  }
};
module.exports = exports.default;
