// @sqs/universal-state/lib/transport/Bridge.js
import _defineProperty from "@babel/runtime/helpers/defineProperty";
import { sendEvent, listenForEvents, UniversalStateEventType } from './protocol';
/**
 * Bridge sets up communication with a 'consumer' (non-root) window.
 *
 * It subscribes to the 'real' store, posts state updates to the consumer window
 * whenever state changes, and relays dispatched actions from the consumer window
 * back to the store.
 *
 * When a REGISTER event is received from a consumer window, Bridge responds
 * immediately with a STATE_UPDATE event to populate the window's state.
 */
export default class Bridge {
  constructor(store, consumerWindow) {
    _defineProperty(this, "handleEvent", event => {
      switch (event.type) {
        case UniversalStateEventType.REGISTER:
          this.sendStateUpdate();
          break;
        case UniversalStateEventType.DISPATCH:
          this.store.dispatch(event.payload.action);
          break;
        default:
          break;
      }
    });
    _defineProperty(this, "sendStateUpdate", () => {
      var state = this.store.getState();
      sendEvent(this.consumerWindow, {
        type: UniversalStateEventType.STATE_UPDATE,
        payload: {
          state
        }
      });
    });
    this.store = store;
    this.consumerWindow = consumerWindow;
    this.stopListeningForEvents = listenForEvents(this.consumerWindow, this.handleEvent);
    this.unsubscribeFromStore = this.store.subscribe(this.sendStateUpdate);
  }
  teardown() {
    this.unsubscribeFromStore();
    this.stopListeningForEvents();
  }
}
