// @sqs/febs-test-config/lib/createJestConfig.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const createJestConfig = cb => {
    const locale = 'en-US';
    const language = locale.split('-')[0];
    const baseConfig = {
        coverageDirectory: '<rootDir>/coverage',
        coverageReporters: ['text', 'html'],
        globals: {
            __BRANCH_NAME__: 'jestBranch',
            __BUILD_NUMBER__: 'jest',
            __DEV__: false,
            __TEST__: true,
            __TRANSLATION_LANGUAGE__: language,
            __TRANSLATION_LOCALE__: locale,
        },
        moduleNameMapper: {
            '\\.(css|less)$': require.resolve('identity-obj-proxy'),
            '\\.(jpg|jpeg|png|gif|eot|otf|webp|svg|ttf|woff|woff2)$': require.resolve('./mocks/fileMock'),
            '\\.yaml$': require.resolve('identity-obj-proxy'),
        },
        setupFiles: [
            require.resolve('regenerator-runtime'),
            require.resolve('./setup/polyfills'),
            require.resolve('./setup/shim'),
        ],
        snapshotSerializers: [require.resolve('jest-serializer-path')],
        testMatch: [
            '<rootDir>/**/__tests__/**/*.test.[jt]s?(x)',
            '<rootDir>/**/?(*.)test.[jt]s?(x)',
        ],
        transform: {
            '^.+\\.[jt]sx?$': require.resolve('./babelTransform'),
        },
        transformIgnorePatterns: ['/node_modules/'],
        watchPathIgnorePatterns: ['/node_modules/'],
    };
    return typeof cb === 'function' ? cb(baseConfig) : baseConfig;
};
exports.default = createJestConfig;
