// @sqs/track-events/cjs/index.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
var _typeof = require("@babel/runtime/helpers/typeof");
Object.defineProperty(exports, "__esModule", {
  value: true
});
Object.defineProperty(exports, "ClientConfig", {
  enumerable: true,
  get: function get() {
    return _types.ClientConfig;
  }
});
Object.defineProperty(exports, "DefaultPayload", {
  enumerable: true,
  get: function get() {
    return _types.DefaultPayload;
  }
});
Object.defineProperty(exports, "EventSource", {
  enumerable: true,
  get: function get() {
    return _types.EventSource;
  }
});
Object.defineProperty(exports, "SiteVisitorClient", {
  enumerable: true,
  get: function get() {
    return _v3.SiteVisitorClient;
  }
});
Object.defineProperty(exports, "SourceEnvironment", {
  enumerable: true,
  get: function get() {
    return _types.SourceEnvironment;
  }
});
Object.defineProperty(exports, "v0", {
  enumerable: true,
  get: function get() {
    return _v["default"];
  }
});
Object.defineProperty(exports, "v1", {
  enumerable: true,
  get: function get() {
    return _v2["default"];
  }
});
Object.defineProperty(exports, "v2", {
  enumerable: true,
  get: function get() {
    return _v3["default"];
  }
});
var _v = _interopRequireDefault(require("./v0"));
var _v2 = _interopRequireDefault(require("./v1"));
var _v3 = _interopRequireWildcard(require("./v2"));
var _types = require("./types");
function _getRequireWildcardCache(e) { if ("function" != typeof WeakMap) return null; var r = new WeakMap(), t = new WeakMap(); return (_getRequireWildcardCache = function _getRequireWildcardCache(e) { return e ? t : r; })(e); }
function _interopRequireWildcard(e, r) { if (!r && e && e.__esModule) return e; if (null === e || "object" != _typeof(e) && "function" != typeof e) return { "default": e }; var t = _getRequireWildcardCache(r); if (t && t.has(e)) return t.get(e); var n = { __proto__: null }, a = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var u in e) if ("default" !== u && {}.hasOwnProperty.call(e, u)) { var i = a ? Object.getOwnPropertyDescriptor(e, u) : null; i && (i.get || i.set) ? Object.defineProperty(n, u, i) : n[u] = e[u]; } return n["default"] = e, t && t.set(e, n), n; }
