// @sqs/i18n-locale-list/lib/locales.js
import "core-js/modules/es.array.concat.js";
/**
 * Locales are highly tailored to v6, since v6's functionality still acts as
 * the source of truth for how we internationalize our products.
 *
 * At the time of the creation of this file, we currently have two separate
 * notions of locales we support within v6. One set of locales is for members -
 * i.e. logged in users of /config - and the other set of locales is for site
 * visitors, or people who visit Squarespace sites.
 *
 * These diverge because memberLocales are set from the Member Accounts app and
 * we do not allow users to specify their locale beyond our six supported languages.
 * If a member selects Spanish they are set to es-419 (Latin American Spanish),
 * even if they would are from Mexico and expect Mexican formatting.
 *
 * Visitor locales are set on a site-basis within v6 directly. Users navigate
 * Website -> Regional to set this locale from a large dropdown, the contents
 * of which is populated below.
 *
 * Locales live in two states:
 * - `beta`: Historically what we've used to designate locales that we haven't
 *   launched yet. They should be shipped to production but only visible behind
 *   beta flags or open to specific users.
 * - `enabled`: Locales that are shipped to all users in production.
 *
 * The `all` list is a union of `beta` and `enabled`.
 *
 * `translationFiles` specifies the locales for which we generate translations.
 * Even if we allow users to specify `es-MX` and render Mexican-specific
 * formatters, we do not currently provide translations that cater specifically
 * to Mexico. Instead we would display translations for `es-419`.
 * Additionally it does not have the concept of "beta" locales because
 * its intended usage is for build configurations (the number of language available to build).
 * There is some validation logic in FEBs that ensures you only build what’s available inside `translationFiles`.
 * The actual updating of builds is independent of that list so it should be safe to update
 * as long as there are no unintended usages (which should be checked prior to updating)
 *
 * `default` specifies the default, or fallback, locale. It should always be
 * en-US.
 */

/**
 * List of supported translation locales
 */
var supportedTranslationLocales = ['en-US', 'es-419', 'fr-FR', 'de-DE', 'pt-BR', 'it-IT', 'nl-NL', 'sv-SE', 'no-NO', 'fi-FI', 'da-DK', 'ja-JP', 'tr-TR'];
var supportedLanguageTags = ['en', 'es', 'fr', 'de', 'pt', 'it', 'nl', 'sv', 'no', 'fi', 'da', 'ja', 'tr'];
var memberEnabledLocales = ['en-US', 'de-DE', 'fr-FR', 'es-419', 'pt-BR', 'it-IT', 'ja-JP'];
var memberBetaLocales = ['tr-TR'];
var visitorEnabledLocales = ['da-DK', 'da-GL', 'de-AT', 'de-CH', 'de-DE', 'de-LU', 'en-AU', 'en-CA', 'en-GB', 'en-IE', 'en-IN', 'en-NZ', 'en-US', 'en-ZA', 'es-AR', 'es-BO', 'es-CL', 'es-CO', 'es-CR', 'es-DO', 'es-EC', 'es-ES', 'es-GT', 'es-HN', 'es-MX', 'es-NI', 'es-PA', 'es-PE', 'es-PR', 'es-PY', 'es-SV', 'es-UY', 'es-VE', 'fi-FI', 'fr-BE', 'fr-CA', 'fr-CH', 'fr-FR', 'fr-LU', 'it-CH', 'it-IT', 'ja-JP', 'nl-BE', 'nl-NL', 'no-NO', 'pt-BR', 'pt-PT', 'sv-SE'];
var visitorBetaLocales = ['tr-TR'];
// There is a typescript limitation that prevents us from typing default exports,
// so let's create a typed Locales and set it as the default export
// https://github.com/microsoft/TypeScript/issues/13626
var LOCALES = {
  member: {
    all: [].concat(memberEnabledLocales, memberBetaLocales),
    beta: [].concat(memberBetaLocales),
    enabled: [].concat(memberEnabledLocales)
  },
  visitor: {
    all: [].concat(visitorEnabledLocales, visitorBetaLocales),
    beta: [].concat(visitorBetaLocales),
    enabled: [].concat(visitorEnabledLocales)
  },
  supportedLanguageTags: [].concat(supportedLanguageTags),
  translationFiles: [].concat(supportedTranslationLocales),
  defaultLocale: 'en-US'
};
export default LOCALES;
