// @sqs/i18n-ui/lib/functions/translationDictionary/MessageStore.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var utils_1 = require("./utils");
var MessageStore = (function () {
    function MessageStore(strs) {
        if (strs === void 0) { strs = {}; }
        this.__messages = strs;
    }
    MessageStore.prototype.get = function (key) {
        if ((0, utils_1.isString)(this.__messages[key])) {
            return this.__messages[key];
        }
        return (0, utils_1.getObj)(this.__messages, key.split('.'), null);
    };
    MessageStore.prototype.has = function (key) {
        return (this.__messages[key] !== undefined || (0, utils_1.getObj)(this.__messages, key.split('.'), null) !== null);
    };
    MessageStore.prototype.set = function (key, value) {
        if ((0, utils_1.isString)(key)) {
            this.setMessage(key, value);
        }
        else if ((0, utils_1.isPOJO)(key)) {
            this.extend(key);
        }
        return this;
    };
    MessageStore.prototype.setMessage = function (key, value) {
        this.__messages[key] = value;
        return this;
    };
    MessageStore.prototype.replace = function (strMessages) {
        this.__messages = strMessages;
        return this;
    };
    MessageStore.prototype.extend = function (strMessages) {
        Object.assign(this.__messages, strMessages);
        return this;
    };
    MessageStore.prototype.reset = function () {
        this.__messages = {};
        return this;
    };
    return MessageStore;
}());
exports.default = MessageStore;
