// @phensley/cldr-core/lib/internals/numbers/util.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.getCurrencyForRegion = exports.getCurrencyFractions = void 0;
var autogen_currencies_1 = require("./autogen.currencies");
var string_1 = require("../../utils/string");
var currencyFractions = (function () {
    var map = {};
    autogen_currencies_1.currencyFractionsRaw.split('|').forEach(function (r) {
        var parts = r.split(':');
        var code = parts[0];
        var values = (0, string_1.numarray)(parts[1]);
        map[code] = {
            digits: values[0],
            rounding: values[1],
            cashDigits: values[2],
            cashRounding: values[3],
        };
    });
    return map;
})();
var defaultCurrencyFractions = {
    digits: 2,
    rounding: 0,
    cashDigits: 2,
    cashRounding: 0,
};
var currencyRegions = (0, string_1.stringToObject)(autogen_currencies_1.currencyRegionsRaw, '|', ':');
/**
 * @internal
 */
var getCurrencyFractions = function (code) {
    return currencyFractions[code] || defaultCurrencyFractions;
};
exports.getCurrencyFractions = getCurrencyFractions;
/**
 * @internal
 */
var getCurrencyForRegion = function (region) {
    return currencyRegions[region] || 'USD';
};
exports.getCurrencyForRegion = getCurrencyForRegion;
