// @sqs/i18n-ui/lib/functions/cldr/framework.native.js
"use strict";

require("core-js/modules/es.symbol.js");
require("core-js/modules/es.array.filter.js");
require("core-js/modules/es.object.get-own-property-descriptor.js");
require("core-js/modules/es.object.get-own-property-descriptors.js");
require("core-js/modules/es.object.keys.js");
require("core-js/modules/es.object.to-string.js");
require("core-js/modules/web.dom-collections.for-each.js");
var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;
var _defineProperty2 = _interopRequireDefault(require("@babel/runtime/helpers/defineProperty"));
var _cldrCore = require("@phensley/cldr-core");
var _i18nCldr = require("@sqs/i18n-cldr");
function ownKeys(e, r) { var t = Object.keys(e); if (Object.getOwnPropertySymbols) { var o = Object.getOwnPropertySymbols(e); r && (o = o.filter(function (r) { return Object.getOwnPropertyDescriptor(e, r).enumerable; })), t.push.apply(t, o); } return t; }
function _objectSpread(e) { for (var r = 1; r < arguments.length; r++) { var t = null != arguments[r] ? arguments[r] : {}; r % 2 ? ownKeys(Object(t), !0).forEach(function (r) { (0, _defineProperty2.default)(e, r, t[r]); }) : Object.getOwnPropertyDescriptors ? Object.defineProperties(e, Object.getOwnPropertyDescriptors(t)) : ownKeys(Object(t)).forEach(function (r) { Object.defineProperty(e, r, Object.getOwnPropertyDescriptor(t, r)); }); } return e; }
var languages = {
  af: require('@sqs/i18n-cldr/packs/af.json'),
  am: require('@sqs/i18n-cldr/packs/am.json'),
  ar: require('@sqs/i18n-cldr/packs/ar.json'),
  as: require('@sqs/i18n-cldr/packs/as.json'),
  az: require('@sqs/i18n-cldr/packs/az.json'),
  be: require('@sqs/i18n-cldr/packs/be.json'),
  bg: require('@sqs/i18n-cldr/packs/bg.json'),
  bn: require('@sqs/i18n-cldr/packs/bn.json'),
  bs: require('@sqs/i18n-cldr/packs/bs.json'),
  ca: require('@sqs/i18n-cldr/packs/ca.json'),
  cs: require('@sqs/i18n-cldr/packs/cs.json'),
  cy: require('@sqs/i18n-cldr/packs/cy.json'),
  da: require('@sqs/i18n-cldr/packs/da.json'),
  de: require('@sqs/i18n-cldr/packs/de.json'),
  el: require('@sqs/i18n-cldr/packs/el.json'),
  en: require('@sqs/i18n-cldr/packs/en.json'),
  es: require('@sqs/i18n-cldr/packs/es.json'),
  et: require('@sqs/i18n-cldr/packs/et.json'),
  eu: require('@sqs/i18n-cldr/packs/eu.json'),
  fa: require('@sqs/i18n-cldr/packs/fa.json'),
  fi: require('@sqs/i18n-cldr/packs/fi.json'),
  fil: require('@sqs/i18n-cldr/packs/fil.json'),
  fr: require('@sqs/i18n-cldr/packs/fr.json'),
  ga: require('@sqs/i18n-cldr/packs/ga.json'),
  gl: require('@sqs/i18n-cldr/packs/gl.json'),
  gu: require('@sqs/i18n-cldr/packs/gu.json'),
  he: require('@sqs/i18n-cldr/packs/he.json'),
  hi: require('@sqs/i18n-cldr/packs/hi.json'),
  hr: require('@sqs/i18n-cldr/packs/hr.json'),
  hu: require('@sqs/i18n-cldr/packs/hu.json'),
  hy: require('@sqs/i18n-cldr/packs/hy.json'),
  id: require('@sqs/i18n-cldr/packs/id.json'),
  is: require('@sqs/i18n-cldr/packs/is.json'),
  it: require('@sqs/i18n-cldr/packs/it.json'),
  ja: require('@sqs/i18n-cldr/packs/ja.json'),
  jv: require('@sqs/i18n-cldr/packs/jv.json'),
  ka: require('@sqs/i18n-cldr/packs/ka.json'),
  kk: require('@sqs/i18n-cldr/packs/kk.json'),
  km: require('@sqs/i18n-cldr/packs/km.json'),
  kn: require('@sqs/i18n-cldr/packs/kn.json'),
  ko: require('@sqs/i18n-cldr/packs/ko.json'),
  ky: require('@sqs/i18n-cldr/packs/ky.json'),
  lo: require('@sqs/i18n-cldr/packs/lo.json'),
  lt: require('@sqs/i18n-cldr/packs/lt.json'),
  lv: require('@sqs/i18n-cldr/packs/lv.json'),
  mk: require('@sqs/i18n-cldr/packs/mk.json'),
  ml: require('@sqs/i18n-cldr/packs/ml.json'),
  mn: require('@sqs/i18n-cldr/packs/mn.json'),
  mr: require('@sqs/i18n-cldr/packs/mr.json'),
  ms: require('@sqs/i18n-cldr/packs/ms.json'),
  my: require('@sqs/i18n-cldr/packs/my.json'),
  nb: require('@sqs/i18n-cldr/packs/nb.json'),
  ne: require('@sqs/i18n-cldr/packs/ne.json'),
  nl: require('@sqs/i18n-cldr/packs/nl.json'),
  or: require('@sqs/i18n-cldr/packs/or.json'),
  pa: require('@sqs/i18n-cldr/packs/pa.json'),
  pl: require('@sqs/i18n-cldr/packs/pl.json'),
  ps: require('@sqs/i18n-cldr/packs/ps.json'),
  pt: require('@sqs/i18n-cldr/packs/pt.json'),
  ro: require('@sqs/i18n-cldr/packs/ro.json'),
  ru: require('@sqs/i18n-cldr/packs/ru.json'),
  sd: require('@sqs/i18n-cldr/packs/sd.json'),
  si: require('@sqs/i18n-cldr/packs/si.json'),
  sk: require('@sqs/i18n-cldr/packs/sk.json'),
  sl: require('@sqs/i18n-cldr/packs/sl.json'),
  so: require('@sqs/i18n-cldr/packs/so.json'),
  sq: require('@sqs/i18n-cldr/packs/sq.json'),
  sr: require('@sqs/i18n-cldr/packs/sr.json'),
  sv: require('@sqs/i18n-cldr/packs/sv.json'),
  sw: require('@sqs/i18n-cldr/packs/sw.json'),
  ta: require('@sqs/i18n-cldr/packs/ta.json'),
  te: require('@sqs/i18n-cldr/packs/te.json'),
  th: require('@sqs/i18n-cldr/packs/th.json'),
  tk: require('@sqs/i18n-cldr/packs/tk.json'),
  tr: require('@sqs/i18n-cldr/packs/tr.json'),
  uk: require('@sqs/i18n-cldr/packs/uk.json'),
  ur: require('@sqs/i18n-cldr/packs/ur.json'),
  uz: require('@sqs/i18n-cldr/packs/uz.json'),
  vi: require('@sqs/i18n-cldr/packs/vi.json'),
  yue: require('@sqs/i18n-cldr/packs/yue.json'),
  zh: require('@sqs/i18n-cldr/packs/zh.json'),
  zu: require('@sqs/i18n-cldr/packs/zu.json')
};
var loader = function loader(language) {
  return languages[language];
};
var getCLDRFramework = function getCLDRFramework(cldrOptions) {
  return new _cldrCore.CLDRFramework(_objectSpread(_objectSpread({
    loader: loader
  }, cldrOptions), {}, {
    config: _i18nCldr.config
  }));
};
var _default = exports.default = getCLDRFramework;
module.exports = exports.default;
