// @sqs/universal-state/lib/index.js
/*
 * Re-export names so consumers don't have to reach into ./lib
 */

export * from './types';

// functions
import * as _actions from './actions';
export { _actions as actions };
import * as _selectors from './selectors';
export { _selectors as selectors };
/*
 * Set up the store
 */
import createStore from './createStore';
import { createContext } from './transport';
var store = createStore();

// Set up React context components and utility functions

var _createContext = createContext(store),
  Provider = _createContext.Provider,
  Consumer = _createContext.Consumer,
  FrameBridge = _createContext.FrameBridge,
  useSelector = _createContext.useSelector,
  useDispatch = _createContext.useDispatch,
  connect = _createContext.connect;

// Export the store itself, for code outside of a React tree
export { Provider, Consumer, FrameBridge, useSelector, useDispatch, connect };
export { store };

// Export a default object with all utilities

export default {
  Provider,
  Consumer,
  FrameBridge,
  useSelector,
  useDispatch,
  connect,
  store
};
