// @phensley/cldr-core/lib/internals/init.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.InternalsImpl = void 0;
var schema_1 = require("../schema");
var schema_2 = require("./schema");
var calendars_1 = require("./calendars");
var datefields_1 = require("./datefields");
var general_1 = require("./general");
var numbers_1 = require("./numbers");
var units_1 = require("./units");
var checksum_1 = require("../resource/checksum");
/**
 * @internal
 */
var InternalsImpl = /** @class */ (function () {
    function InternalsImpl(config, version, debug, skipChecksum, patternCacheSize) {
        if (debug === void 0) { debug = false; }
        if (skipChecksum === void 0) { skipChecksum = false; }
        if (patternCacheSize === void 0) { patternCacheSize = 50; }
        // TODO: may move this up depending on how integration evolves
        this.config = config;
        var code = new schema_1.CodeBuilder(config);
        var origin = code.origin();
        this.indices = origin.indices;
        this.schema = (0, schema_2.buildSchema)(origin, debug);
        this.checksum = skipChecksum ? '' : (0, checksum_1.checksumIndices)(version, origin.indices);
        this.calendars = new calendars_1.CalendarInternalsImpl(this, patternCacheSize);
        this.dateFields = new datefields_1.DateFieldInternalsImpl(this);
        this.general = new general_1.GeneralInternalsImpl(this, patternCacheSize);
        this.numbers = new numbers_1.NumberInternalsImpl(this, patternCacheSize);
        this.units = new units_1.UnitsInternalImpl(this);
    }
    return InternalsImpl;
}());
exports.InternalsImpl = InternalsImpl;
