// @sqs/praetor/build/main/lib/static_client.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var VARIANT_TRUE = 'true';
var DEFAULT_SEGMENT = 'default';
var ExperimentType;
(function (ExperimentType) {
    ExperimentType["FEATURE_TOGGLE"] = "FEATURE_TOGGLE";
    ExperimentType["AB_TEST"] = "AB_TEST";
})(ExperimentType = exports.ExperimentType || (exports.ExperimentType = {}));
var Configuration = /** @class */ (function () {
    function Configuration(variant) {
        var _this = this;
        this.experiments = {};
        this.isConfigurationLoaded = Boolean(variant.isConfigurationLoaded);
        if (Array.isArray(variant.experimentContextList)) {
            variant.experimentContextList.forEach(function (context) {
                if (context !== null &&
                    typeof context === 'object' &&
                    context.hasOwnProperty('name')) {
                    _this.experiments[context.name] = context;
                }
            });
        }
    }
    Configuration.prototype.isValid = function () {
        return this.isConfigurationLoaded;
    };
    Configuration.prototype.getContext = function (experimentName) {
        return this.experiments[experimentName];
    };
    return Configuration;
}());
exports.Configuration = Configuration;
var StaticPraetorClient = /** @class */ (function () {
    function StaticPraetorClient(variant) {
        this.configuration = new Configuration(variant);
    }
    StaticPraetorClient.prototype.getFeatureToggle = function (feature, defaultIfError) {
        var _a = this.getContextValidity(feature, ExperimentType.FEATURE_TOGGLE), context = _a.context, error = _a.error;
        if (error || context === null) {
            return {
                enabled: defaultIfError,
                error: error
            };
        }
        if (context.containsError) {
            return {
                enabled: this.isFeatureToggleEnabled(context),
                error: 'The specified feature has an invalid server-side definition'
            };
        }
        return {
            enabled: this.isFeatureToggleEnabled(context)
        };
    };
    StaticPraetorClient.prototype.getABTestVariant = function (feature, defaultVariantIfError) {
        var _a = this.getContextValidity(feature, ExperimentType.AB_TEST), context = _a.context, error = _a.error;
        if (error || context === null) {
            return {
                error: error,
                segment: DEFAULT_SEGMENT,
                variant: defaultVariantIfError
            };
        }
        if (context.containsError) {
            return {
                error: 'The specified feature has an invalid server-side definition',
                segment: context.segmentName,
                variant: context.variant
            };
        }
        return {
            segment: context.segmentName,
            variant: context.variant
        };
    };
    StaticPraetorClient.prototype.getAllExperiments = function () {
        return this.configuration;
    };
    StaticPraetorClient.prototype.getContextValidity = function (feature, expectedType) {
        if (!this.configuration.isValid()) {
            return {
                context: null,
                error: 'The underlying Praetor configuration is not loaded'
            };
        }
        var context = this.configuration.getContext(feature);
        if (context === undefined) {
            return {
                context: null,
                error: 'The specified feature does not exist'
            };
        }
        else if (context.experimentType !== expectedType) {
            return {
                context: null,
                error: "The specified feature is not a " + expectedType
            };
        }
        return {
            context: context
        };
    };
    StaticPraetorClient.prototype.isFeatureToggleEnabled = function (context) {
        return context.variant === VARIANT_TRUE;
    };
    return StaticPraetorClient;
}());
exports.default = StaticPraetorClient;
