// @sqs/praetor/build/main/lib/static_client.test.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (Object.hasOwnProperty.call(mod, k)) result[k] = mod[k];
    result["default"] = mod;
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
var ava_1 = __importDefault(require("ava"));
var static_client_1 = __importStar(require("./static_client"));
var goodConfig = {
    isConfigurationLoaded: true,
    experimentContextList: [
        {
            name: 'toggle-1',
            description: '',
            experimentType: static_client_1.ExperimentType.FEATURE_TOGGLE,
            segmentName: 'default',
            variant: 'true',
            containsError: false,
            status: 'ACTIVE'
        },
        {
            name: 'abtest-1',
            description: '',
            experimentType: static_client_1.ExperimentType.AB_TEST,
            segmentName: 'special',
            variant: 'blue',
            containsError: false,
            status: 'ACTIVE'
        }
    ]
};
var badConfig = {
    isConfigurationLoaded: false,
    experimentContextList: []
};
var emptyConfig = {};
ava_1.default('can retrieve feature toggle details', function (t) {
    var client = new static_client_1.default(goodConfig);
    t.is(client.getFeatureToggle('toggle-1', false).enabled, true);
});
ava_1.default("returns default if feature doesn't exist", function (t) {
    var client = new static_client_1.default(goodConfig);
    t.is(client.getFeatureToggle('toggle-nope', false).enabled, false);
});
ava_1.default('can retrieve AB test variants', function (t) {
    var client = new static_client_1.default(goodConfig);
    var _a = client.getABTestVariant('abtest-1', 'red'), error = _a.error, variant = _a.variant, segment = _a.segment;
    t.is(error, undefined);
    t.is(variant, 'blue');
    t.is(segment, 'special');
});
ava_1.default('returns default if AB test does not exist', function (t) {
    var client = new static_client_1.default(goodConfig);
    var _a = client.getABTestVariant('abtest-nope', 'red'), error = _a.error, variant = _a.variant, segment = _a.segment;
    t.not(error, undefined);
    t.is(variant, 'red');
    t.is(segment, 'default');
});
ava_1.default('returns default for an AB test if the configuration is bad', function (t) {
    var client = new static_client_1.default(badConfig);
    var _a = client.getABTestVariant('abtest-nope', 'red'), error = _a.error, variant = _a.variant, segment = _a.segment;
    t.not(error, undefined);
    t.is(variant, 'red');
    t.is(segment, 'default');
});
ava_1.default('returns default for a feature toggle if the configuration is bad', function (t) {
    var client = new static_client_1.default(badConfig);
    t.is(client.getFeatureToggle('toggle-nope', false).enabled, false);
});
ava_1.default('constructs an praetor client with invalid config if the configuration is empty', function (t) {
    t.notThrows(function () { return new static_client_1.default(emptyConfig); });
    var client = new static_client_1.default(emptyConfig);
    var config = client.getAllExperiments();
    t.is(config.isValid(), false);
});
