// @phensley/cldr-core/lib/schema/schema/calendar.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.WeekdayValues = exports.QuarterValues = exports.FormatWidthValues = exports.FieldWidthValues = exports.EraWidthValues = exports.DayPeriodValues = exports.DateTimePatternFieldValues = exports.GregorianMonthsIndex = exports.PersianEraIndex = exports.JapaneseEraIndex = exports.GregorianEraIndex = exports.BuddhistEraIndex = void 0;
var instructions_1 = require("../instructions");
/**
 * Generate a key index containing numeric keys from start to end
 * inclusive.
 */
var rangeindex = function (start, end) {
    var r = [];
    for (var i = start; i <= end; i++) {
        r.push(String(i));
    }
    return new instructions_1.KeyIndexImpl(r);
};
/**
 * @public
 */
exports.BuddhistEraIndex = rangeindex(0, 0);
/**
 * @public
 */
exports.GregorianEraIndex = rangeindex(0, 1);
/**
 * @public
 */
exports.JapaneseEraIndex = rangeindex(0, 236);
/**
 * @public
 */
exports.PersianEraIndex = exports.BuddhistEraIndex;
/**
 * @public
 */
exports.GregorianMonthsIndex = rangeindex(1, 12);
/**
 * @public
 */
exports.DateTimePatternFieldValues = ['y', 'M', 'd', 'a', 'H', 'm', 's'];
/**
 * @public
 */
exports.DayPeriodValues = [
    'noon',
    'midnight',
    'am',
    'pm',
    'morning1',
    'morning2',
    'afternoon1',
    'afternoon2',
    'evening1',
    'evening2',
    'night1',
    'night2',
];
/**
 * @public
 */
exports.EraWidthValues = ['names', 'abbr', 'narrow'];
/**
 * @public
 */
exports.FieldWidthValues = ['abbreviated', 'narrow', 'short', 'wide'];
/**
 * @public
 */
exports.FormatWidthValues = ['short', 'medium', 'long', 'full'];
/**
 * @public
 */
exports.QuarterValues = ['1', '2', '3', '4'];
/**
 * @public
 */
exports.WeekdayValues = ['1', '2', '3', '4', '5', '6', '7'];
