// @phensley/cldr-core/lib/schema/schema/datefields.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.RelativeTimeFieldIndex = exports.DateFieldWidthIndex = exports.DateFieldIndex = exports.RelativeTimeFieldValues = exports.DateFieldWidthValues = exports.DateFieldValues = void 0;
var instructions_1 = require("../instructions");
/**
 * @public
 */
exports.DateFieldValues = [
    'era',
    'year',
    'quarter',
    'month',
    'week',
    'weekday',
    'weekdayOfMonth',
    'sun',
    'mon',
    'tue',
    'wed',
    'thu',
    'fri',
    'sat',
    'day',
    'dayperiod',
    'hour',
    'minute',
    'second',
    'zone',
];
/**
 * @public
 */
exports.DateFieldWidthValues = ['short', 'narrow', 'wide'];
/**
 * @public
 */
exports.RelativeTimeFieldValues = [
    'year',
    'quarter',
    'month',
    'week',
    'day',
    'sun',
    'mon',
    'tue',
    'wed',
    'thu',
    'fri',
    'sat',
    'hour',
    'minute',
    'second',
];
/**
 * @public
 */
exports.DateFieldIndex = new instructions_1.KeyIndexImpl(exports.DateFieldValues);
/**
 * @public
 */
exports.DateFieldWidthIndex = new instructions_1.KeyIndexImpl(exports.DateFieldWidthValues);
/**
 * @public
 */
exports.RelativeTimeFieldIndex = new instructions_1.KeyIndexImpl(exports.RelativeTimeFieldValues);
