// @phensley/cldr-core/lib/schema/schema/numbers.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.NumberSymbolIndex = exports.NumberSystemsIndex = exports.NumberMiscPatternIndex = exports.NumberMiscPatternValues = exports.NumberSystemCategoryValues = exports.NumberSymbolValues = exports.CurrencySpacingPatternIndex = exports.CurrencySpacingPosIndex = exports.PluralDigitValues = void 0;
var instructions_1 = require("../instructions");
/**
 * @public
 */
exports.PluralDigitValues = [1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15];
/**
 * @public
 */
exports.CurrencySpacingPosIndex = new instructions_1.KeyIndexImpl(['before', 'after']);
/**
 * @public
 */
exports.CurrencySpacingPatternIndex = new instructions_1.KeyIndexImpl([
    'currencyMatch',
    'surroundingMatch',
    'insertBetween',
]);
/**
 * @public
 */
exports.NumberSymbolValues = [
    'approximatelySign',
    'currencyDecimal',
    'currencyGroup',
    'decimal',
    'exponential',
    'group',
    'infinity',
    'list',
    'minusSign',
    'nan',
    'perMille',
    'percentSign',
    'plusSign',
    'superscriptingExponent',
    'timeSeparator',
];
/**
 * @public
 */
exports.NumberSystemCategoryValues = ['default', 'native', 'finance', 'traditional'];
/**
 * @public
 */
exports.NumberMiscPatternValues = ['at-least', 'at-most', 'approx', 'range'];
/**
 * @public
 */
exports.NumberMiscPatternIndex = new instructions_1.KeyIndexImpl(exports.NumberMiscPatternValues);
/**
 * @public
 */
exports.NumberSystemsIndex = new instructions_1.KeyIndexImpl(exports.NumberSystemCategoryValues);
/**
 * @public
 */
exports.NumberSymbolIndex = new instructions_1.KeyIndexImpl(exports.NumberSymbolValues);
