// @sqs/universal-state/cjs/slices/flyout.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.selectors = exports.reducer = exports.initialState = exports.actions = void 0;
var _toolkit = require("@reduxjs/toolkit");
var initialState = exports.initialState = {
  isOpen: false,
  focusedProduct: null,
  isTemplatePickerOpen: false
};
var slice = (0, _toolkit.createSlice)({
  name: 'flyout',
  initialState,
  reducers: {
    open(state) {
      state.isOpen = true;
      state.focusedProduct = null;
    },
    close(state) {
      state.isOpen = false;
      state.focusedProduct = null;
    },
    openSubpanel(state, action) {
      state.isOpen = true;
      state.focusedProduct = action.payload;
    },
    closeSubpanel(state) {
      // Don't mutate state.isOpen.
      // This guards against race conditions where closeSubpanel() is called after close().
      state.focusedProduct = null;
    },
    openTemplatePicker(state) {
      state.isTemplatePickerOpen = true;
    },
    closeTemplatePicker(state) {
      state.isTemplatePickerOpen = false;
    }
  }
});
var reducer = exports.reducer = slice.reducer;
var actions = exports.actions = {
  openFlyout: slice.actions.open,
  closeFlyout: slice.actions.close,
  openFlyoutSubpanel: slice.actions.openSubpanel,
  closeFlyoutSubpanel: slice.actions.closeSubpanel,
  openTemplatePicker: slice.actions.openTemplatePicker,
  closeTemplatePicker: slice.actions.closeTemplatePicker
};
var selectors = exports.selectors = {
  selectIsOpen: _ref => {
    var flyout = _ref.flyout;
    return flyout.isOpen;
  },
  selectFocusedProduct: _ref2 => {
    var flyout = _ref2.flyout;
    return flyout.focusedProduct;
  },
  selectIsTemplatePickerOpen: _ref3 => {
    var flyout = _ref3.flyout;
    return flyout.isTemplatePickerOpen;
  }
};
