// @sqs/universal-state/cjs/slices/project.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.selectors = exports.reducer = exports.initialState = exports.actions = void 0;
var _toolkit = require("@reduxjs/toolkit");
var _bootstrap = require("./bootstrap");
var initialState = exports.initialState = null;
var slice = (0, _toolkit.createSlice)({
  name: 'project',
  initialState: initialState,
  reducers: {
    updateProjectTitle(state, action) {
      if (!state) {
        throw new Error("Received an action of type \"".concat(action.type, "\", but project state hasn't been bootstrapped. ") + 'You probably need to dispatch the "bootstrap/bootstrap" action first.');
      }
      state.projectTitle = action.payload;
    }
  },
  extraReducers: builder => {
    builder.addCase(_bootstrap.actions.hardBootstrap, (_, action) => action.payload.project);
    builder.addCase(_bootstrap.actions.softBootstrap, (state, action) => {
      return state || action.payload.project;
    });
  }
});
var reducer = exports.reducer = slice.reducer;
var actions = exports.actions = {
  updateProjectTitle: slice.actions.updateProjectTitle
};
var selectors = exports.selectors = {
  selectProjectTitle: _ref => {
    var _project$projectTitle;
    var project = _ref.project;
    return (_project$projectTitle = project === null || project === void 0 ? void 0 : project.projectTitle) !== null && _project$projectTitle !== void 0 ? _project$projectTitle : null;
  }
};
