// @sqs/universal-state/cjs/index.js
"use strict";

require("core-js/modules/es.array.iterator.js");
require("core-js/modules/es.object.to-string.js");
require("core-js/modules/es.weak-map.js");
require("core-js/modules/web.dom-collections.iterator.js");
require("core-js/modules/web.dom-collections.for-each.js");
var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
Object.defineProperty(exports, "__esModule", {
  value: true
});
var _exportNames = {
  Provider: true,
  Consumer: true,
  FrameBridge: true,
  useSelector: true,
  useDispatch: true,
  connect: true,
  store: true,
  actions: true,
  selectors: true
};
exports.useSelector = exports.useDispatch = exports.store = exports.selectors = exports.default = exports.connect = exports.actions = exports.Provider = exports.FrameBridge = exports.Consumer = void 0;
var _types = require("./types");
Object.keys(_types).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  if (key in exports && exports[key] === _types[key]) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function get() {
      return _types[key];
    }
  });
});
var _actions = _interopRequireWildcard(require("./actions"));
exports.actions = _actions;
var _selectors = _interopRequireWildcard(require("./selectors"));
exports.selectors = _selectors;
var _createStore = _interopRequireDefault(require("./createStore"));
var _transport = require("./transport");
function _getRequireWildcardCache(e) { if ("function" != typeof WeakMap) return null; var r = new WeakMap(), t = new WeakMap(); return (_getRequireWildcardCache = function _getRequireWildcardCache(e) { return e ? t : r; })(e); }
function _interopRequireWildcard(e, r) { if (!r && e && e.__esModule) return e; if (null === e || "object" != typeof e && "function" != typeof e) return { default: e }; var t = _getRequireWildcardCache(r); if (t && t.has(e)) return t.get(e); var n = { __proto__: null }, a = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var u in e) if ("default" !== u && Object.prototype.hasOwnProperty.call(e, u)) { var i = a ? Object.getOwnPropertyDescriptor(e, u) : null; i && (i.get || i.set) ? Object.defineProperty(n, u, i) : n[u] = e[u]; } return n.default = e, t && t.set(e, n), n; }
/*
 * Re-export names so consumers don't have to reach into ./lib
 */

// functions

/*
 * Set up the store
 */

var store = exports.store = (0, _createStore.default)();

// Set up React context components and utility functions

var _createContext = (0, _transport.createContext)(store),
  Provider = exports.Provider = _createContext.Provider,
  Consumer = exports.Consumer = _createContext.Consumer,
  FrameBridge = exports.FrameBridge = _createContext.FrameBridge,
  useSelector = exports.useSelector = _createContext.useSelector,
  useDispatch = exports.useDispatch = _createContext.useDispatch,
  connect = exports.connect = _createContext.connect;

// Export the store itself, for code outside of a React tree
// Export a default object with all utilities
var _default = exports.default = {
  Provider,
  Consumer,
  FrameBridge,
  useSelector,
  useDispatch,
  connect,
  store
};
