// @sqs/eslint-plugin/lib/rules/no-locale-in-help-center-article-link.js
/**
 * @fileoverview Disallow Locale Specific Help Center Article link.
 * @author Stephanie Manwaring
 */

module.exports = {
  meta: {
    docs: {
      description:
        'Disallow Locale Specific Help Center Article Link.',
      category: 'Stylistic Issues',
      recommended: false,
    },
    fixable: null,
    schema: [],
  },

  create: function(context) {
    /**
     * checks for the following pattens:
     * /hc/en-us/articles
     * /hc/boop-1/articles
     * /hc/123/articles
     */
    const IS_LOCALE_SPECIFIED_IN_HC_LINK_REGEX = /\/hc\/([a-zA-Z0-9\-])*\/articles/;
    const ERROR_MESSAGE_NO_LOCALE_IN_HC_LINK =
    'Help Center link should not specify a locale in the URL.';

    return {
      Literal(node) {
        if (
          node.value &&
          IS_LOCALE_SPECIFIED_IN_HC_LINK_REGEX.test(node.value)
        ) {
          context.report({
            node,
            loc: node.loc,
            message: ERROR_MESSAGE_NO_LOCALE_IN_HC_LINK,
          });
        }
      },
    };
  },
};
