// @sqs/eslint-plugin/lib/rules/no-window-delete.js
/**
 * @fileoverview Rule to detect deleting protected window properties
 */

module.exports = {
  meta: {
    docs: {
      description: 'Disallow deleting window properties (Safari 9.x)',
      category: 'Browser Compatibility Issues',
      recommended: false,
    },
  },

  create(context) {
    return {
      'UnaryExpression:exit'(node) {
        // Check if we're inside a try/catch
        // UnaryExpression -> ExpressionStatement -> BlockStatement -> TryStatement
        if (
          node.parent.parent.type === 'BlockStatement' &&
          node.parent.parent.parent &&
          node.parent.parent.parent.type === 'TryStatement'
        ) {
          return;
        }

        if (
          node.operator === 'delete' &&
          node.argument.type === 'MemberExpression' &&
          node.argument.object.type === 'Identifier' &&
          node.argument.object.name === 'window'
        ) {
          context.report(
            node,
            'Deleting protected window properties in Safari 9.x throws a TypeError. Wrap statement in try/catch.'
          );
        }
      },
    };
  },
};
