// @sqs/i18n-locale-list/lib/locales.js
import _toConsumableArray from "@babel/runtime/helpers/toConsumableArray";
import "core-js/modules/es.array.concat.js";

/**
 * Locales are highly tailored to v6, since v6's functionality still acts as
 * the source of truth for how we internationalize our products.
 *
 * At the time of the creation of this file, we currently have two separate
 * notions of locales we support within v6. One set of locales is for members -
 * i.e. logged in users of /config - and the other set of locales is for site
 * visitors, or people who visit Squarespace sites.
 *
 * These diverge because memberLocales are set from the Member Accounts app and
 * we do not allow users to specify their locale beyond our six supported languages.
 * If a member selects Spanish they are set to es-419 (Latin American Spanish),
 * even if they would are from Mexico and expect Mexican formatting.
 *
 * Visitor locales are set on a site-basis within v6 directly. Users navigate
 * Website -> Regional to set this locale from a large dropdown, the contents
 * of which is populated below.
 *
 * Locales live in two states:
 * - `beta`: Historically what we've used to designate locales that we haven't
 *   launched yet. They should be shipped to production but only visible behind
 *   beta flags or open to specific users.
 * - `enabled`: Locales that are shipped to all users in production.
 *
 * The `all` list is a union of `beta` and `enabled`.
 *
 * `translationFiles` specifies the locales for which we generate translations.
 * Even if we allow users to specify `es-MX` and render Mexican-specific
 * formatters, we do not currently provide translations that cater specifically
 * to Mexico. Instead we would display translations for `es-419`.
 *
 * `default` specifies the default, or fallback, locale. It should always be
 * en-US.
 */

/**
 * List of supported translation locales
 */
var supportedTranslationLocales = ['en-US', 'es-419', 'fr-FR', 'de-DE', 'pt-BR', 'it-IT', 'nl-NL', 'sv-SE', 'no-NO', 'fi-FI', 'da-DK'];
/**
 * List of supported locales
 */

var supportedLocales = [].concat(supportedTranslationLocales, [// Arabic
'ar-AE', 'ar-BH', 'ar-DZ', 'ar-EG', 'ar-IQ', 'ar-JO', 'ar-KW', 'ar-LB', 'ar-LY', 'ar-MA', 'ar-OM', 'ar-QA', 'ar-SA', 'ar-SD', 'ar-SY', 'ar-TN', 'ar-YE', // Belarusian
'be-BY', // Bulgarian
'bg-BG', // Catalan
'ca-ES', // Czech
'cs-CZ', // Danish (Greenland)
'da-GL', // German
'de-AT', 'de-CH', 'de-LU', // Greek
'el-GR', // English
'en-AU', 'en-CA', 'en-GB', 'en-HK', 'en-IE', 'en-IN', 'en-NZ', 'en-SG', 'en-ZA', // Spanish
'es-AR', 'es-BO', 'es-CL', 'es-CO', 'es-CR', 'es-DO', 'es-EC', 'es-ES', 'es-GT', 'es-HN', 'es-MX', 'es-NI', 'es-PA', 'es-PE', 'es-PR', 'es-PY', 'es-SV', 'es-UY', 'es-VE', // Estonian
'et-EE', // French
'fr-BE', 'fr-CA', 'fr-CH', 'fr-LU', // Hindi
'hi-IN', // Croatian
'hr-HR', // Hungarian
'hu-HU', // Icelandic
'is-IS', // Italian
'it-CH', // Hebrew
'iw-IL', // Japanese
'ja-JP', // Korean
'ko-KR', // Lithuanian
'lt-LT', // Latvian
'lv-LV', // Macedonian
'mk-MK', // Dutch
'nl-BE', // Polish
'pl-PL', // Portuguese
'pt-PT', // Romanian
'ro-RO', // Russian
'ru-RU', // Serbian (Latin)
'sh-YU', // Serbian (Cyrillic)
'sr-YU', // Slovak
'sk-SK', // Slovenian
'sl-SI', // Albanian
'sq-AL', // Thai
'th-TH', // Turkish
'tr-TR', // Ukrainian (Cyrillic)
'uk-UA', // Chinese
'zh-CN', 'zh-HK']);
var supportedLanguageTags = ['en', 'es', 'fr', 'de', 'pt', 'it', 'nl', 'sv'];
// There is a typescript limitation that prevents us from typing default exports,
// so let's create a typed Locales and set it as the default export
// https://github.com/microsoft/TypeScript/issues/13626
var LOCALES = {
  member: {
    all: ['en-US', 'en-AU', 'en-CA', 'en-IE', 'en-HK', 'en-GB', 'en-NZ', 'en-SG', 'de-DE', 'de-AT', 'de-CH', 'fr-FR', 'fr-CA', 'fr-BE', 'es-419', 'es-ES', 'es-AR', 'es-CO', 'es-MX', 'pt-BR', 'pt-PT', 'it-IT'],
    beta: [],
    enabled: ['en-US', 'en-AU', 'en-CA', 'en-IE', 'en-HK', 'en-GB', 'en-NZ', 'en-SG', 'de-DE', 'de-AT', 'de-CH', 'fr-FR', 'fr-CA', 'fr-BE', 'es-419', 'es-ES', 'es-AR', 'es-CO', 'es-MX', 'pt-BR', 'pt-PT', 'it-IT']
  },
  visitor: {
    all: _toConsumableArray(supportedLocales),
    beta: [],
    enabled: _toConsumableArray(supportedLocales)
  },
  supportedLanguageTags: [].concat(supportedLanguageTags),
  translationFiles: [].concat(supportedTranslationLocales),
  defaultLocale: 'en-US'
};
export default LOCALES;
