// @sqs/track-events/cjs/v2/constants/defaultConfig.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.getDefaultConfig = void 0;
var _util = require("../../util");
var _types = require("../../types");
var getDefaultConfig = exports.getDefaultConfig = function getDefaultConfig() {
  var _maybeWindow;
  return {
    devBaseDomainUrl: 'https://clanker-events.stage.sqsp.net',
    cookie: {
      daysToStore: 390,
      domains: ['squarespace.com', 'squarespace.net', 'sqsp.net', 'acuityscheduling.com', 'acuityscheduling.net'],
      path: '/',
      secure: ((_maybeWindow = (0, _util.maybeWindow)()) === null || _maybeWindow === void 0 ? void 0 : _maybeWindow.location.protocol.indexOf('https')) > -1
    },
    expirationKey: 'SS_ANALYTICS_EXPIRATION',
    fireEvents: true,
    logging: false,
    sourceEnvironment: _types.SourceEnvironment.DEV,
    storageKey: 'SS_ANALYTICS_ID',
    url: 'https://clanker-events.stage.sqsp.net/api/v1/clanker/events',
    useBeacon: true,
    validateMode: false,
    setAnalyticsCookie: true
  };
};
