// @sqs/track-events/cjs/v1/constants/common-schema-fields.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;
var _default = exports["default"] = {
  // Required fields
  actor: {
    type: 'string',
    nullable: false
  },
  action: {
    type: 'string',
    nullable: false
  },
  event_owner_team: {
    type: 'string',
    nullable: false
  },
  event_source: {
    type: 'string',
    nullable: false
  },
  object_type: {
    type: 'string',
    nullable: false
  },
  // Optional fields
  context_website_identifier: {
    type: 'string',
    nullable: true
  },
  destination_url: {
    type: 'string',
    nullable: true
  },
  object_identifier: {
    type: 'string',
    nullable: true
  },
  object_id: {
    type: 'string',
    nullable: true
  },
  object_display_name: {
    type: 'string',
    nullable: true
  },
  object_value: {
    type: 'string',
    nullable: true
  },
  indirect_object_identifier: {
    type: 'string',
    nullable: true
  },
  indirect_object_id: {
    type: 'string',
    nullable: true
  },
  indirect_object_display_name: {
    type: 'string',
    nullable: true
  },
  indirect_object_value: {
    type: 'string',
    nullable: true
  },
  indirect_object_type: {
    type: 'string',
    nullable: true
  },
  product_area: {
    type: 'string',
    nullable: true
  },
  product_page: {
    type: 'string',
    nullable: true
  },
  product_section: {
    type: 'string',
    nullable: true
  },
  product_design_identifier: {
    type: 'string',
    nullable: true
  },
  user_member_account_id: {
    type: 'string',
    nullable: true
  },
  user_session_id: {
    type: 'string',
    nullable: true
  },
  browser_locale: {
    type: 'string',
    nullable: true
  },
  member_account_locale: {
    type: 'string',
    nullable: true
  },
  website_locale: {
    type: 'string',
    nullable: true
  },
  frontsite_language: {
    type: 'string',
    nullable: true
  }
};
module.exports = exports.default;
