// @sqs/track-events/cjs/v1/constants/defaultConfig.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.getDefaultConfig = void 0;
var _util = require("../../util");
var getDefaultConfig = exports.getDefaultConfig = function getDefaultConfig() {
  var _maybeWindow, _maybeWindow2;
  var HOST = ((_maybeWindow = (0, _util.maybeWindow)()) === null || _maybeWindow === void 0 ? void 0 : _maybeWindow.location.hostname.indexOf('squarespace.com')) > -1 ? 'https://events.squarespace.com' : 'https://events.stage.sqsp.net';
  return {
    customSchemaName: null,
    fireEvents: true,
    logging: false,
    url: "".concat(HOST, "/api/v1/events"),
    useBeacon: true,
    storageKey: 'SS_ANALYTICS_ID',
    expirationKey: 'SS_ANALYTICS_EXPIRATION',
    cookie: {
      daysToStore: 390,
      domains: ['squarespace.com', 'squarespace.net', 'sqsp.net'],
      path: '/',
      secure: ((_maybeWindow2 = (0, _util.maybeWindow)()) === null || _maybeWindow2 === void 0 ? void 0 : _maybeWindow2.location.protocol.indexOf('https')) > -1
    },
    setAnalyticsCookie: true
  };
};
